<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Pranav Sampara" />


<title>SIPmg-vignette</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<style type="text/css">.pagedtable {
overflow: auto;
padding-left: 8px;
padding-right: 8px;
}
.pagedtable-wrapper {
border: 1px solid #ccc;
border-radius: 4px;
margin-bottom: 10px;
}
.pagedtable table {
width: 100%;
max-width: 100%;
margin: 0;
}
.pagedtable th {
padding: 0 5px 0 5px;
border: none;
border-bottom: 2px solid #dddddd;
min-width: 45px;
}
.pagedtable-empty th {
display: none;
}
.pagedtable td {
padding: 0 4px 0 4px;
}
.pagedtable .even {
background-color: rgba(140, 140, 140, 0.1);
}
.pagedtable-padding-col {
display: none;
}
.pagedtable a {
-webkit-touch-callout: none;
-webkit-user-select: none;
-khtml-user-select: none;
-moz-user-select: none;
-ms-user-select: none;
user-select: none;
}
.pagedtable-index-nav {
cursor: pointer;
padding: 0 5px 0 5px;
float: right;
border: 0;
}
.pagedtable-index-nav-disabled {
cursor: default;
text-decoration: none;
color: #999;
}
a.pagedtable-index-nav-disabled:hover {
text-decoration: none;
color: #999;
}
.pagedtable-indexes {
cursor: pointer;
float: right;
border: 0;
}
.pagedtable-index-current {
cursor: default;
text-decoration: none;
font-weight: bold;
color: #333;
border: 0;
}
a.pagedtable-index-current:hover {
text-decoration: none;
font-weight: bold;
color: #333;
}
.pagedtable-index {
width: 30px;
display: inline-block;
text-align: center;
border: 0;
}
.pagedtable-index-separator-left {
display: inline-block;
color: #333;
font-size: 9px;
padding: 0 0 0 0;
cursor: default;
}
.pagedtable-index-separator-right {
display: inline-block;
color: #333;
font-size: 9px;
padding: 0 4px 0 0;
cursor: default;
}
.pagedtable-footer {
padding-top: 4px;
padding-bottom: 5px;
}
.pagedtable-not-empty .pagedtable-footer {
border-top: 2px solid #dddddd;
}
.pagedtable-info {
overflow: hidden;
color: #999;
white-space: nowrap;
text-overflow: ellipsis;
}
.pagedtable-header-name {
overflow: hidden;
text-overflow: ellipsis;
}
.pagedtable-header-type {
color: #999;
font-weight: 400;
}
.pagedtable-na-cell {
font-style: italic;
opacity: 0.3;
}
</style>
<script>// Production steps of ECMA-262, Edition 5, 15.4.4.18
// Reference: http://es5.github.io/#x15.4.4.18
if (!Array.prototype.forEach) {

  Array.prototype.forEach = function(callback, thisArg) {

    var T, k;

    if (this === null) {
      throw new TypeError(' this is null or not defined');
    }

    // 1. Let O be the result of calling toObject() passing the
    // |this| value as the argument.
    var O = Object(this);

    // 2. Let lenValue be the result of calling the Get() internal
    // method of O with the argument "length".
    // 3. Let len be toUint32(lenValue).
    var len = O.length >>> 0;

    // 4. If isCallable(callback) is false, throw a TypeError exception.
    // See: http://es5.github.com/#x9.11
    if (typeof callback !== "function") {
      throw new TypeError(callback + ' is not a function');
    }

    // 5. If thisArg was supplied, let T be thisArg; else let
    // T be undefined.
    if (arguments.length > 1) {
      T = thisArg;
    }

    // 6. Let k be 0
    k = 0;

    // 7. Repeat, while k < len
    while (k < len) {

      var kValue;

      // a. Let Pk be ToString(k).
      //    This is implicit for LHS operands of the in operator
      // b. Let kPresent be the result of calling the HasProperty
      //    internal method of O with argument Pk.
      //    This step can be combined with c
      // c. If kPresent is true, then
      if (k in O) {

        // i. Let kValue be the result of calling the Get internal
        // method of O with argument Pk.
        kValue = O[k];

        // ii. Call the Call internal method of callback with T as
        // the this value and argument list containing kValue, k, and O.
        callback.call(T, kValue, k, O);
      }
      // d. Increase k by 1.
      k++;
    }
    // 8. return undefined
  };
}

// Production steps of ECMA-262, Edition 5, 15.4.4.19
// Reference: http://es5.github.io/#x15.4.4.19
if (!Array.prototype.map) {

  Array.prototype.map = function(callback, thisArg) {

    var T, A, k;

    if (this == null) {
      throw new TypeError(' this is null or not defined');
    }

    // 1. Let O be the result of calling ToObject passing the |this|
    //    value as the argument.
    var O = Object(this);

    // 2. Let lenValue be the result of calling the Get internal
    //    method of O with the argument "length".
    // 3. Let len be ToUint32(lenValue).
    var len = O.length >>> 0;

    // 4. If IsCallable(callback) is false, throw a TypeError exception.
    // See: http://es5.github.com/#x9.11
    if (typeof callback !== 'function') {
      throw new TypeError(callback + ' is not a function');
    }

    // 5. If thisArg was supplied, let T be thisArg; else let T be undefined.
    if (arguments.length > 1) {
      T = thisArg;
    }

    // 6. Let A be a new array created as if by the expression new Array(len)
    //    where Array is the standard built-in constructor with that name and
    //    len is the value of len.
    A = new Array(len);

    // 7. Let k be 0
    k = 0;

    // 8. Repeat, while k < len
    while (k < len) {

      var kValue, mappedValue;

      // a. Let Pk be ToString(k).
      //   This is implicit for LHS operands of the in operator
      // b. Let kPresent be the result of calling the HasProperty internal
      //    method of O with argument Pk.
      //   This step can be combined with c
      // c. If kPresent is true, then
      if (k in O) {

        // i. Let kValue be the result of calling the Get internal
        //    method of O with argument Pk.
        kValue = O[k];

        // ii. Let mappedValue be the result of calling the Call internal
        //     method of callback with T as the this value and argument
        //     list containing kValue, k, and O.
        mappedValue = callback.call(T, kValue, k, O);

        // iii. Call the DefineOwnProperty internal method of A with arguments
        // Pk, Property Descriptor
        // { Value: mappedValue,
        //   Writable: true,
        //   Enumerable: true,
        //   Configurable: true },
        // and false.

        // In browsers that support Object.defineProperty, use the following:
        // Object.defineProperty(A, k, {
        //   value: mappedValue,
        //   writable: true,
        //   enumerable: true,
        //   configurable: true
        // });

        // For best browser support, use the following:
        A[k] = mappedValue;
      }
      // d. Increase k by 1.
      k++;
    }

    // 9. return A
    return A;
  };
}

var PagedTable = function (pagedTable) {
  var me = this;

  var source = function(pagedTable) {
    var sourceElems = [].slice.call(pagedTable.children).filter(function(e) {
      return e.hasAttribute("data-pagedtable-source");
    });

    if (sourceElems === null || sourceElems.length !== 1) {
      throw("A single data-pagedtable-source was not found");
    }

    return JSON.parse(sourceElems[0].innerHTML);
  }(pagedTable);

  var options = function(source) {
    var options = typeof(source.options) !== "undefined" &&
      source.options !== null ? source.options : {};

    var columns = typeof(options.columns) !== "undefined" ? options.columns : {};
    var rows = typeof(options.rows) !== "undefined" ? options.rows : {};

    var positiveIntOrNull = function(value) {
      return parseInt(value) >= 0 ? parseInt(value) : null;
    };

    return {
      pages: positiveIntOrNull(options.pages),
      rows: {
        min: positiveIntOrNull(rows.min),
        max: positiveIntOrNull(rows.max),
        total: positiveIntOrNull(rows.total)
      },
      columns: {
        min: positiveIntOrNull(columns.min),
        max: positiveIntOrNull(columns.max),
        total: positiveIntOrNull(columns.total)
      }
    };
  }(source);

  var Measurer = function() {

    // set some default initial values that will get adjusted in runtime
    me.measures = {
      padding: 12,
      character: 8,
      height: 15,
      defaults: true
    };

    me.calculate = function(measuresCell) {
      if (!me.measures.defaults)
        return;

      var measuresCellStyle = window.getComputedStyle(measuresCell, null);

      var newPadding = parsePadding(measuresCellStyle.paddingLeft) +
            parsePadding(measuresCellStyle.paddingRight);

      var sampleString = "ABCDEFGHIJ0123456789";
      var newCharacter = Math.ceil(measuresCell.clientWidth / sampleString.length);

      if (newPadding <= 0 || newCharacter <= 0)
        return;

      me.measures.padding = newPadding;
      me.measures.character = newCharacter;
      me.measures.height = measuresCell.clientHeight;
      me.measures.defaults = false;
    };

    return me;
  };

  var Page = function(data, options) {
    var me = this;

    var defaults = {
      max: 7,
      rows: 10
    };

    var totalPages = function() {
      return Math.ceil(data.length / me.rows);
    };

    me.number = 0;
    me.max = options.pages !== null ? options.pages : defaults.max;
    me.visible = me.max;
    me.rows = options.rows.min !== null ? options.rows.min : defaults.rows;
    me.total = totalPages();

    me.setRows = function(newRows) {
      me.rows = newRows;
      me.total = totalPages();
    };

    me.setPageNumber = function(newPageNumber) {
      if (newPageNumber < 0) newPageNumber = 0;
      if (newPageNumber >= me.total) newPageNumber = me.total - 1;

      me.number = newPageNumber;
    };

    me.setVisiblePages = function(visiblePages) {
      me.visible = Math.min(me.max, visiblePages);
      me.setPageNumber(me.number);
    };

    me.getVisiblePageRange = function() {
      var start = me.number - Math.max(Math.floor((me.visible - 1) / 2), 0);
      var end = me.number + Math.floor(me.visible / 2) + 1;
      var pageCount = me.total;

      if (start < 0) {
        var diffToStart = 0 - start;
        start += diffToStart;
        end += diffToStart;
      }

      if (end > pageCount) {
        var diffToEnd = end - pageCount;
        start -= diffToEnd;
        end -= diffToEnd;
      }

      start = start < 0 ? 0 : start;
      end = end >= pageCount ? pageCount : end;

      var first = false;
      var last = false;

      if (start > 0 && me.visible > 1) {
        start = start + 1;
        first = true;
      }

      if (end < pageCount && me.visible > 2) {
        end = end - 1;
        last = true;
      }

      return {
        first: first,
        start: start,
        end: end,
        last: last
      };
    };

    me.getRowStart = function() {
      var rowStart = page.number * page.rows;
      if (rowStart < 0)
        rowStart = 0;

      return rowStart;
    };

    me.getRowEnd = function() {
      var rowStart = me.getRowStart();
      return Math.min(rowStart + me.rows, data.length);
    };

    me.getPaddingRows = function() {
      var rowStart = me.getRowStart();
      var rowEnd = me.getRowEnd();
      return data.length > me.rows ? me.rows - (rowEnd - rowStart) : 0;
    };
  };

  var Columns = function(data, columns, options) {
    var me = this;

    me.defaults = {
      min: 5
    };

    me.number = 0;
    me.visible = 0;
    me.total = columns.length;
    me.subset = [];
    me.padding = 0;
    me.min = options.columns.min !== null ? options.columns.min : me.defaults.min;
    me.max = options.columns.max !== null ? options.columns.max : null;
    me.widths = {};

    var widthsLookAhead = Math.max(100, options.rows.min);
    var paddingColChars = 10;

    me.emptyNames = function() {
      columns.forEach(function(column) {
        if (columns.label !== null && columns.label !== "")
          return false;
      });

      return true;
    };

    var parsePadding = function(value) {
      return parseInt(value) >= 0 ? parseInt(value) : 0;
    };

    me.calculateWidths = function(measures) {
      columns.forEach(function(column) {
        var maxChars = Math.max(
          column.label.toString().length,
          column.type.toString().length
        );

        for (var idxRow = 0; idxRow < Math.min(widthsLookAhead, data.length); idxRow++) {
          maxChars = Math.max(maxChars, data[idxRow][column.name.toString()].length);
        }

        me.widths[column.name] = {
          // width in characters
          chars: maxChars,
          // width for the inner html columns
          inner: maxChars * measures.character,
          // width adding outer styles like padding
          outer: maxChars * measures.character + measures.padding
        };
      });
    };

    me.getWidth = function() {
      var widthOuter = 0;
      for (var idxCol = 0; idxCol < me.subset.length; idxCol++) {
        var columnName = me.subset[idxCol].name;
        widthOuter = widthOuter + me.widths[columnName].outer;
      }

      widthOuter = widthOuter + me.padding * paddingColChars * measurer.measures.character;

      if (me.hasMoreLeftColumns()) {
        widthOuter = widthOuter + columnNavigationWidthPX + measurer.measures.padding;
      }

      if (me.hasMoreRightColumns()) {
        widthOuter = widthOuter + columnNavigationWidthPX + measurer.measures.padding;
      }

      return widthOuter;
    };

    me.updateSlice = function() {
      if (me.number + me.visible >= me.total)
        me.number = me.total - me.visible;

      if (me.number < 0) me.number = 0;

      me.subset = columns.slice(me.number, Math.min(me.number + me.visible, me.total));

      me.subset = me.subset.map(function(column) {
        Object.keys(column).forEach(function(colKey) {
          column[colKey] = column[colKey] === null ? "" : column[colKey].toString();
        });

        column.width = null;
        return column;
      });
    };

    me.setVisibleColumns = function(columnNumber, newVisibleColumns, paddingCount) {
      me.number = columnNumber;
      me.visible = newVisibleColumns;
      me.padding = paddingCount;

      me.updateSlice();
    };

    me.incColumnNumber = function(increment) {
      me.number = me.number + increment;
    };

    me.setColumnNumber = function(newNumber) {
      me.number = newNumber;
    };

    me.setPaddingCount = function(newPadding) {
      me.padding = newPadding;
    };

    me.getPaddingCount = function() {
      return me.padding;
    };

    me.hasMoreLeftColumns = function() {
      return me.number > 0;
    };

    me.hasMoreRightColumns = function() {
      return me.number + me.visible < me.total;
    };

    me.updateSlice(0);
    return me;
  };

  var data = source.data;
  var page = new Page(data, options);
  var measurer = new Measurer(data, options);
  var columns = new Columns(data, source.columns, options);

  var table = null;
  var tableDiv = null;
  var header = null;
  var footer = null;
  var tbody = null;

  // Caches pagedTable.clientWidth, specially for webkit
  var cachedPagedTableClientWidth = null;

  var onChangeCallbacks = [];

  var clearSelection = function() {
    if(document.selection && document.selection.empty) {
      document.selection.empty();
    } else if(window.getSelection) {
      var sel = window.getSelection();
      sel.removeAllRanges();
    }
  };

  var columnNavigationWidthPX = 5;

  var renderColumnNavigation = function(increment, backwards) {
    var arrow = document.createElement("div");
    arrow.setAttribute("style",
      "border-top: " + columnNavigationWidthPX + "px solid transparent;" +
      "border-bottom: " + columnNavigationWidthPX + "px solid transparent;" +
      "border-" + (backwards ? "right" : "left") + ": " + columnNavigationWidthPX + "px solid;");

    var header = document.createElement("th");
    header.appendChild(arrow);
    header.setAttribute("style",
      "cursor: pointer;" +
      "vertical-align: middle;" +
      "min-width: " + columnNavigationWidthPX + "px;" +
      "width: " + columnNavigationWidthPX + "px;");

    header.onclick = function() {
      columns.incColumnNumber(backwards ? -1 : increment);

      me.animateColumns(backwards);
      renderFooter();

      clearSelection();
      triggerOnChange();
    };

    return header;
  };

  var maxColumnWidth = function(width) {
    var padding = 80;
    var columnMax = Math.max(cachedPagedTableClientWidth - padding, 0);

    return parseInt(width) > 0 ?
      Math.min(columnMax, parseInt(width)) + "px" :
      columnMax + "px";
  };

  var clearHeader = function() {
    var thead = pagedTable.querySelectorAll("thead")[0];
    thead.innerHTML = "";
  };

  var renderHeader = function(clear) {
    cachedPagedTableClientWidth = pagedTable.clientWidth;

    var fragment = document.createDocumentFragment();

    header = document.createElement("tr");
    fragment.appendChild(header);

    if (columns.number > 0)
      header.appendChild(renderColumnNavigation(-columns.visible, true));

    columns.subset = columns.subset.map(function(columnData) {
      var column = document.createElement("th");
      column.setAttribute("align", columnData.align);
      column.style.textAlign = columnData.align;

      column.style.maxWidth = maxColumnWidth(null);
      if (columnData.width) {
        column.style.minWidth =
          column.style.maxWidth = maxColumnWidth(columnData.width);
      }

      var columnName = document.createElement("div");
      columnName.setAttribute("class", "pagedtable-header-name");
      if (columnData.label === "") {
        columnName.innerHTML = "&nbsp;";
      }
      else {
        columnName.appendChild(document.createTextNode(columnData.label));
      }
      column.appendChild(columnName);

      var columnType = document.createElement("div");
      columnType.setAttribute("class", "pagedtable-header-type");
      if (columnData.type === "") {
        columnType.innerHTML = "&nbsp;";
      }
      else {
        columnType.appendChild(document.createTextNode("<" + columnData.type + ">"));
      }
      column.appendChild(columnType);

      header.appendChild(column);

      columnData.element = column;

      return columnData;
    });

    for (var idx = 0; idx < columns.getPaddingCount(); idx++) {
      var paddingCol = document.createElement("th");
      paddingCol.setAttribute("class", "pagedtable-padding-col");
      header.appendChild(paddingCol);
    }

    if (columns.number + columns.visible < columns.total)
      header.appendChild(renderColumnNavigation(columns.visible, false));

    if (typeof(clear) == "undefined" || clear) clearHeader();
    var thead = pagedTable.querySelectorAll("thead")[0];
    thead.appendChild(fragment);
  };

  me.animateColumns = function(backwards) {
    var thead = pagedTable.querySelectorAll("thead")[0];

    var headerOld = thead.querySelectorAll("tr")[0];
    var tbodyOld = table.querySelectorAll("tbody")[0];

    me.fitColumns(backwards);

    renderHeader(false);

    header.style.opacity = "0";
    header.style.transform = backwards ? "translateX(-30px)" : "translateX(30px)";
    header.style.transition = "transform 200ms linear, opacity 200ms";
    header.style.transitionDelay = "0";

    renderBody(false);

    if (headerOld) {
      headerOld.style.position = "absolute";
      headerOld.style.transform = "translateX(0px)";
      headerOld.style.opacity = "1";
      headerOld.style.transition = "transform 100ms linear, opacity 100ms";
      headerOld.setAttribute("class", "pagedtable-remove-head");
      if (headerOld.style.transitionEnd) {
        headerOld.addEventListener("transitionend", function() {
          var headerOldByClass = thead.querySelector(".pagedtable-remove-head");
          if (headerOldByClass) thead.removeChild(headerOldByClass);
        });
      }
      else {
        thead.removeChild(headerOld);
      }
    }

    if (tbodyOld) table.removeChild(tbodyOld);

    tbody.style.opacity = "0";
    tbody.style.transition = "transform 200ms linear, opacity 200ms";
    tbody.style.transitionDelay = "0ms";

    // force relayout
    window.getComputedStyle(header).opacity;
    window.getComputedStyle(tbody).opacity;

    if (headerOld) {
      headerOld.style.transform = backwards ? "translateX(20px)" : "translateX(-30px)";
      headerOld.style.opacity = "0";
    }

    header.style.transform = "translateX(0px)";
    header.style.opacity = "1";

    tbody.style.opacity = "1";
  }

  me.onChange = function(callback) {
    onChangeCallbacks.push(callback);
  };

  var triggerOnChange = function() {
    onChangeCallbacks.forEach(function(onChange) {
      onChange();
    });
  };

  var clearBody = function() {
    if (tbody) {
      table.removeChild(tbody);
      tbody = null;
    }
  };

  var renderBody = function(clear) {
    cachedPagedTableClientWidth = pagedTable.clientWidth

    var fragment = document.createDocumentFragment();

    var pageData = data.slice(page.getRowStart(), page.getRowEnd());

    pageData.forEach(function(dataRow, idxRow) {
      var htmlRow = document.createElement("tr");
      htmlRow.setAttribute("class", (idxRow % 2 !==0) ? "even" : "odd");

      if (columns.hasMoreLeftColumns())
        htmlRow.appendChild(document.createElement("td"));

      columns.subset.forEach(function(columnData) {
        var cellName = columnData.name;
        var dataCell = dataRow[cellName];
        var htmlCell = document.createElement("td");

        if (dataCell === "NA") htmlCell.setAttribute("class", "pagedtable-na-cell");
        if (dataCell === "__NA__") dataCell = "NA";

        var cellText = document.createTextNode(dataCell);
        htmlCell.appendChild(cellText);
        if (dataCell.length > 50) {
          htmlCell.setAttribute("title", dataCell);
        }
        htmlCell.setAttribute("align", columnData.align);
        htmlCell.style.textAlign = columnData.align;
        htmlCell.style.maxWidth = maxColumnWidth(null);
        if (columnData.width) {
          htmlCell.style.minWidth = htmlCell.style.maxWidth = maxColumnWidth(columnData.width);
        }
        htmlRow.appendChild(htmlCell);
      });

      for (var idx = 0; idx < columns.getPaddingCount(); idx++) {
        var paddingCol = document.createElement("td");
        paddingCol.setAttribute("class", "pagedtable-padding-col");
        htmlRow.appendChild(paddingCol);
      }

      if (columns.hasMoreRightColumns())
        htmlRow.appendChild(document.createElement("td"));

      fragment.appendChild(htmlRow);
    });

    for (var idxPadding = 0; idxPadding < page.getPaddingRows(); idxPadding++) {
      var paddingRow = document.createElement("tr");

      var paddingCellRow = document.createElement("td");
      paddingCellRow.innerHTML = "&nbsp;";
      paddingCellRow.setAttribute("colspan", "100%");
      paddingRow.appendChild(paddingCellRow);

      fragment.appendChild(paddingRow);
    }

    if (typeof(clear) == "undefined" || clear) clearBody();
    tbody = document.createElement("tbody");
    tbody.appendChild(fragment);

    table.appendChild(tbody);
  };

  var getLabelInfo = function() {
    var pageStart = page.getRowStart();
    var pageEnd = page.getRowEnd();
    var totalRows = data.length;

    var totalRowsLabel = options.rows.total ? options.rows.total : totalRows;
    var totalRowsLabelFormat = totalRowsLabel.toString().replace(/(\d)(?=(\d\d\d)+(?!\d))/g, '$1,');

    var infoText = (pageStart + 1) + "-" + pageEnd + " of " + totalRowsLabelFormat + " rows";
    if (totalRows < page.rows) {
      infoText = totalRowsLabel + " row" + (totalRows != 1 ? "s" : "");
    }
    if (columns.total > columns.visible) {
      var totalColumnsLabel = options.columns.total ? options.columns.total : columns.total;

      infoText = infoText + " | " + (columns.number + 1) + "-" +
        (Math.min(columns.number + columns.visible, columns.total)) +
        " of " + totalColumnsLabel + " columns";
    }

    return infoText;
  };

  var clearFooter = function() {
    footer = pagedTable.querySelectorAll("div.pagedtable-footer")[0];
    footer.innerHTML = "";

    return footer;
  };

  var createPageLink = function(idxPage) {
    var pageLink = document.createElement("a");
    pageLinkClass = idxPage === page.number ? "pagedtable-index pagedtable-index-current" : "pagedtable-index";
    pageLink.setAttribute("class", pageLinkClass);
    pageLink.setAttribute("data-page-index", idxPage);
    pageLink.onclick = function() {
      page.setPageNumber(parseInt(this.getAttribute("data-page-index")));
      renderBody();
      renderFooter();

      triggerOnChange();
    };

    pageLink.appendChild(document.createTextNode(idxPage + 1));

    return pageLink;
  }

  var renderFooter = function() {
    footer = clearFooter();

    var next = document.createElement("a");
    next.appendChild(document.createTextNode("Next"));
    next.onclick = function() {
      page.setPageNumber(page.number + 1);
      renderBody();
      renderFooter();

      triggerOnChange();
    };
    if (data.length > page.rows) footer.appendChild(next);

    var pageNumbers = document.createElement("div");
    pageNumbers.setAttribute("class", "pagedtable-indexes");

    var pageRange = page.getVisiblePageRange();

    if (pageRange.first) {
      var pageLink = createPageLink(0);
      pageNumbers.appendChild(pageLink);

      var pageSeparator = document.createElement("div");
      pageSeparator.setAttribute("class", "pagedtable-index-separator-left");
      pageSeparator.appendChild(document.createTextNode("..."))
      pageNumbers.appendChild(pageSeparator);
    }

    for (var idxPage = pageRange.start; idxPage < pageRange.end; idxPage++) {
      var pageLink = createPageLink(idxPage);

      pageNumbers.appendChild(pageLink);
    }

    if (pageRange.last) {
      var pageSeparator = document.createElement("div");
      pageSeparator.setAttribute("class", "pagedtable-index-separator-right");
      pageSeparator.appendChild(document.createTextNode("..."))
      pageNumbers.appendChild(pageSeparator);

      var pageLink = createPageLink(page.total - 1);
      pageNumbers.appendChild(pageLink);
    }

    if (data.length > page.rows) footer.appendChild(pageNumbers);

    var previous = document.createElement("a");
    previous.appendChild(document.createTextNode("Previous"));
    previous.onclick = function() {
      page.setPageNumber(page.number - 1);
      renderBody();
      renderFooter();

      triggerOnChange();
    };
    if (data.length > page.rows) footer.appendChild(previous);

    var infoLabel = document.createElement("div");
    infoLabel.setAttribute("class", "pagedtable-info");
    infoLabel.setAttribute("title", getLabelInfo());
    infoLabel.appendChild(document.createTextNode(getLabelInfo()));
    footer.appendChild(infoLabel);

    var enabledClass = "pagedtable-index-nav";
    var disabledClass = "pagedtable-index-nav pagedtable-index-nav-disabled";
    previous.setAttribute("class", page.number <= 0 ? disabledClass : enabledClass);
    next.setAttribute("class", (page.number + 1) * page.rows >= data.length ? disabledClass : enabledClass);
  };

  var measuresCell = null;

  var renderMeasures = function() {
    var measuresTable = document.createElement("table");
    measuresTable.style.visibility = "hidden";
    measuresTable.style.position = "absolute";
    measuresTable.style.whiteSpace = "nowrap";
    measuresTable.style.height = "auto";
    measuresTable.style.width = "auto";

    var measuresRow = document.createElement("tr");
    measuresTable.appendChild(measuresRow);

    measuresCell = document.createElement("td");
    var sampleString = "ABCDEFGHIJ0123456789";
    measuresCell.appendChild(document.createTextNode(sampleString));

    measuresRow.appendChild(measuresCell);

    tableDiv.appendChild(measuresTable);
  }

  me.init = function() {
    tableDiv = document.createElement("div");
    pagedTable.appendChild(tableDiv);
    var pagedTableClass = data.length > 0 ?
      "pagedtable pagedtable-not-empty" :
      "pagedtable pagedtable-empty";

    if (columns.total == 0 || (columns.emptyNames() && data.length == 0)) {
      pagedTableClass = pagedTableClass + " pagedtable-empty-columns";
    }

    tableDiv.setAttribute("class", pagedTableClass);

    renderMeasures();
    measurer.calculate(measuresCell);
    columns.calculateWidths(measurer.measures);

    table = document.createElement("table");
    table.setAttribute("cellspacing", "0");
    table.setAttribute("class", "table table-condensed");
    tableDiv.appendChild(table);

    table.appendChild(document.createElement("thead"));

    var footerDiv = document.createElement("div");
    footerDiv.setAttribute("class", "pagedtable-footer");
    tableDiv.appendChild(footerDiv);

    // if the host has not yet provided horizontal space, render hidden
    if (tableDiv.clientWidth <= 0) {
      tableDiv.style.opacity = "0";
    }

    me.render();

    // retry seizing columns later if the host has not provided space
    function retryFit() {
      if (tableDiv.clientWidth <= 0) {
        setTimeout(retryFit, 100);
      } else {
        me.render();
        triggerOnChange();
      }
    }
    if (tableDiv.clientWidth <= 0) {
      retryFit();
    }
  };

  var registerWidths = function() {
    columns.subset = columns.subset.map(function(column) {
      column.width = columns.widths[column.name].inner;
      return column;
    });
  };

  var parsePadding = function(value) {
    return parseInt(value) >= 0 ? parseInt(value) : 0;
  };

  me.fixedHeight = function() {
    return options.rows.max != null;
  }

  me.fitRows = function() {
    if (me.fixedHeight())
      return;

    measurer.calculate(measuresCell);

    var rows = options.rows.min !== null ? options.rows.min : 0;
    var headerHeight = header !== null && header.offsetHeight > 0 ? header.offsetHeight : 0;
    var footerHeight = footer !== null && footer.offsetHeight > 0 ? footer.offsetHeight : 0;

    if (pagedTable.offsetHeight > 0) {
      var availableHeight = pagedTable.offsetHeight - headerHeight - footerHeight;
      rows = Math.floor((availableHeight) / measurer.measures.height);
    }

    rows = options.rows.min !== null ? Math.max(options.rows.min, rows) : rows;

    page.setRows(rows);
  }

  // The goal of this function is to add as many columns as possible
  // starting from left-to-right, when the right most limit is reached
  // it tries to add columns from the left as well.
  //
  // When startBackwards is true columns are added from right-to-left
  me.fitColumns = function(startBackwards) {
    measurer.calculate(measuresCell);
    columns.calculateWidths(measurer.measures);

    if (tableDiv.clientWidth > 0) {
      tableDiv.style.opacity = 1;
    }

    var visibleColumns = tableDiv.clientWidth <= 0 ? Math.max(columns.min, 1) : 1;
    var columnNumber = columns.number;
    var paddingCount = 0;

    // track a list of added columns as we build the visible ones to allow us
    // to remove columns when they don't fit anymore.
    var columnHistory = [];

    var lastTableHeight = 0;
    var backwards = startBackwards;

    var tableDivStyle = window.getComputedStyle(tableDiv, null);
    var tableDivPadding = parsePadding(tableDivStyle.paddingLeft) +
      parsePadding(tableDivStyle.paddingRight);

    var addPaddingCol = false;
    var currentWidth = 0;

    while (true) {
      columns.setVisibleColumns(columnNumber, visibleColumns, paddingCount);
      currentWidth = columns.getWidth();

      if (tableDiv.clientWidth - tableDivPadding < currentWidth) {
        break;
      }

      columnHistory.push({
        columnNumber: columnNumber,
        visibleColumns: visibleColumns,
        paddingCount: paddingCount
      });

      if (columnHistory.length > 100) {
        console.error("More than 100 tries to fit columns, aborting");
        break;
      }

      if (columns.max !== null &&
        columns.visible + columns.getPaddingCount() >= columns.max) {
        break;
      }

      // if we run out of right-columns
      if (!backwards && columnNumber + columns.visible >= columns.total) {
        // if we started adding right-columns, try adding left-columns
        if (!startBackwards && columnNumber > 0) {
          backwards = true;
        }
        else if (columns.min === null || visibleColumns + columns.getPaddingCount() >= columns.min) {
          break;
        }
        else {
          paddingCount = paddingCount + 1;
        }
      }

      // if we run out of left-columns
      if (backwards && columnNumber == 0) {
        // if we started adding left-columns, try adding right-columns
        if (startBackwards && columnNumber + columns.visible < columns.total) {
          backwards = false;
        }
        else if (columns.min === null || visibleColumns + columns.getPaddingCount() >= columns.min) {
          break;
        }
        else {
          paddingCount = paddingCount + 1;
        }
      }

      // when moving backwards try fitting left columns first
      if (backwards && columnNumber > 0) {
        columnNumber = columnNumber - 1;
      }

      if (columnNumber + visibleColumns < columns.total) {
        visibleColumns = visibleColumns + 1;
      }
    }

    var lastRenderableColumn = {
        columnNumber: columnNumber,
        visibleColumns: visibleColumns,
        paddingCount: paddingCount
    };

    if (columnHistory.length > 0) {
      lastRenderableColumn = columnHistory[columnHistory.length - 1];
    }

    columns.setVisibleColumns(
      lastRenderableColumn.columnNumber,
      lastRenderableColumn.visibleColumns,
      lastRenderableColumn.paddingCount);

    if (pagedTable.offsetWidth > 0) {
      page.setVisiblePages(Math.max(Math.ceil(1.0 * (pagedTable.offsetWidth - 250) / 40), 2));
    }

    registerWidths();
  };

  me.fit = function(startBackwards) {
    me.fitRows();
    me.fitColumns(startBackwards);
  }

  me.render = function() {
    me.fitColumns(false);

    // render header/footer to measure height accurately
    renderHeader();
    renderFooter();

    me.fitRows();
    renderBody();

    // re-render footer to match new rows
    renderFooter();
  }

  var resizeLastWidth = -1;
  var resizeLastHeight = -1;
  var resizeNewWidth = -1;
  var resizeNewHeight = -1;
  var resizePending = false;

  me.resize = function(newWidth, newHeight) {

    function resizeDelayed() {
      resizePending = false;

      if (
        (resizeNewWidth !== resizeLastWidth) ||
        (!me.fixedHeight() && resizeNewHeight !== resizeLastHeight)
      ) {
        resizeLastWidth = resizeNewWidth;
        resizeLastHeight = resizeNewHeight;

        setTimeout(resizeDelayed, 200);
        resizePending = true;
      } else {
        me.render();
        triggerOnChange();

        resizeLastWidth = -1;
        resizeLastHeight = -1;
      }
    }

    resizeNewWidth = newWidth;
    resizeNewHeight = newHeight;

    if (!resizePending) resizeDelayed();
  };
};

var PagedTableDoc;
(function (PagedTableDoc) {
  var allPagedTables = [];

  PagedTableDoc.initAll = function() {
    allPagedTables = [];

    var pagedTables = [].slice.call(document.querySelectorAll('[data-pagedtable="false"],[data-pagedtable=""]'));
    pagedTables.forEach(function(pagedTable, idx) {
      pagedTable.setAttribute("data-pagedtable", "true");
      pagedTable.setAttribute("pagedtable-page", 0);
      pagedTable.setAttribute("class", "pagedtable-wrapper");

      var pagedTableInstance = new PagedTable(pagedTable);
      pagedTableInstance.init();

      allPagedTables.push(pagedTableInstance);
    });
  };

  PagedTableDoc.resizeAll = function() {
    allPagedTables.forEach(function(pagedTable) {
      pagedTable.render();
    });
  };

  window.addEventListener("resize", PagedTableDoc.resizeAll);

  return PagedTableDoc;
})(PagedTableDoc || (PagedTableDoc = {}));

window.onload = function() {
  PagedTableDoc.initAll();
};
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">SIPmg-vignette</h1>
<h4 class="author">Pranav Sampara</h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The SIPmg R package was designed to enable the exploration and
statistical analysis of qSIP metagenomics data. Particularly, this
package allows the identification of isotope incorporators and
quantifies isotopic enrichment.</p>
<p>Please check out this introductory vignette to qSIP metagenomics,
before checking out vignettes on:</p>
<p>[Coverage normalization and scaling based on linear regression
model][<a href="https://zielslab.github.io/SIPmg.github.io/scaling-coverage-data.html" class="uri">https://zielslab.github.io/SIPmg.github.io/scaling-coverage-data.html</a>]</p>
<p>[Identification of incorporators based on qSIP, ΔBD, or DESeq2
methods][<a href="https://zielslab.github.io/SIPmg.github.io/identifying-incorporators.html" class="uri">https://zielslab.github.io/SIPmg.github.io/identifying-incorporators.html</a>]</p>
</div>
<div id="background" class="section level1">
<h1>Background</h1>
<p>This section introduces a bit of a background on the methodologies
used in this pipeline, such as stable isotope probing, quantitative
metagenomics, and quantitative stable isotope probing. References as a
point of departure for more reading are also provided.</p>
<div id="stable-isotope-probing" class="section level2">
<h2>Stable isotope probing</h2>
<p>Stable isotope probing (SIP) targets active populations within
complex microbiomes by providing growth substrates enriched in a heavy
stable isotope of carbon, nitrogen, or oxygen. In DNA-SIP, the DNA of
microbes actively incorporating the labelled substrate into their DNA
during cell division and growth would become increasingly labelled and
heavier compared to the microbes not incorporating the substrate into
their DNA. Heavier DNA can be physically separated via isopycnic
centrifugation on a CsCl gradient, followed by fractionation. Sequencing
effort on the “heavier density fractions” could inform the identity and
function of microbes that actively incorporated the isotopic substrate
into their DNA. Thus, DNA-SIP is a powerful technique to link the
identity and function of environmental microbes.</p>
<p>However, traditional DNA-SIP is only a qualitative technique to
identify active microbial populations and no quantitative estimates of
isotopic enrichment of DNA are provided. For instance, the distinction
between labelled and unlabelled microbes is defined by density gradient
regions, limiting the resolution of taxon-specific responses to labelled
or unlabelled. Moreover, there exists a guanine-cytosine (GC) bias in
conventional SIP studies. The native buoyant density of DNA can differ
by as much as 0.05 g/mL over the range of GC contents that can occur in
complex communities. Lighter GC-content genomes may not shift into the
researcher-assigned “heavier density fractions”, while high GC-content
genomes may fall in the “heavier density fractions” without any isotopic
substrate assimilation. Thus, a qualitative analysis may mask the true
incorporators and, worse, falsely identify non-incorporators as
incorporators, skewing the analysis of the study.</p>
</div>
<div id="quantitative-metagenomics" class="section level2">
<h2>Quantitative metagenomics</h2>
<p>Although metagenomics offers a comprehensive account of the metabolic
repertoire in an environment, typically, the data is compositional.
Relative abundance of genes or taxa is estimated to determine microbial
community dynamics, which is directly influenced by the dynamics of rest
of the community. Even if a certain gene remains in the same
concentration, its relative abundance changes if a certain other gene
increases/decreases in concentration. Absolute abundance of gene/genome
can be critical, for instance, in the surveillance of a pathogenic
gene/genome in an environment. For quantitative measurements of
microbial taxon absolute abundance in a given metagenomic sample, <a href="https://www.nature.com/articles/s41467-018-05555-0">Hardwick et
al. (2018)</a> proposed synthetic spike-ins, or sequins, as internal
reference standards. Reference standards in metagenomics can act as
scaling factors to evaluate quantitative estimates of individual
biological features, in this case, genes or genomes.</p>
</div>
<div id="quantitative-sip-qsip" class="section level2">
<h2>Quantitative SIP (qSIP)</h2>
<p>In quantitative SIP (qSIP), quantitative estimates of isotope
incorporation, expressed as atom fraction excess (AFE), are calculated
based on a mathematical model. Quantitative estimates of isotopic
enrichment of DNA facilitate accurate and sensitive estimates of
substrate uptake measurements, and growth and mortality rates of
individual taxa in a complex community. Moreover, qSIP analysis is less
susceptible to GC and enrichment biases, and taxon abundance, making it
amenable to quantify isotopic incorporations in a complex community such
as the activated sludge microbiome, with in-situ conditions. Integration
of qSIP and quantitative metagenomics (qSIP metagenomics) facilitates
the exploration of active microbial populations’ taxonomic and metabolic
diversity with a quantitative estimate of their abundance and isotope
incorporation.</p>
</div>
<div id="for-details-on-sip-quantitative-metagenomics-and-qsip-please-refer-to-the-following-works" class="section level2">
<h2>For details on SIP, quantitative metagenomics, and qSIP, please
refer to the following works:</h2>
<div id="sip" class="section level3">
<h3>SIP</h3>
<p><a href="https://www.nature.com/articles/nprot.2007.109">Neufeld,
Josh D., Jyotsna Vohra, Marc G. Dumont, Tillmann Lueders, Mike
Manefield, Michael W. Friedrich, and J. Colin Murrell. 2007. “DNA
Stable-Isotope Probing.” Nature Protocols 2 (4): 860–66.</a></p>
<p><a href="https://link.springer.com/article/10.1007/s00248-006-9125-x">Neufeld
JD, Dumont MG, Vohra J, Murrell JC. Methodological considerations for
the use of stable isotope probing in microbial ecology. Microbial
Ecology 2007; 53: 435–442.</a></p>
<p><a href="https://journals.asm.org/doi/10.1128/AEM.02609-06">Buckley
DH, Huangyutitham V, Hsu S-F, Nelson TA. Stable isotope probing with 15N
achieved by disentangling the effects of genome G+ C content and isotope
enrichment on DNA density. Applied and environmental microbiology 2007;
73: 3189–3195.</a></p>
</div>
<div id="quantitative-metagenomics-1" class="section level3">
<h3>Quantitative metagenomics</h3>
<p><a href="https://www.nature.com/articles/s41467-018-05555-0">Hardwick
SA, Chen WY, Wong T, Kanakamedala BS, Deveson IW, Ongley SE, et
al. Synthetic microbe communities provide internal reference standards
for metagenome sequencing and analysis. Nature Communications 2018; 9:
3096.</a></p>
</div>
<div id="qsip" class="section level3">
<h3>qSIP</h3>
<p><a href="https://journals.asm.org/doi/10.1128/AEM.02280-15">Hungate
BA, Mau RL, Schwartz E, Caporaso JG, Dijkstra P, Gestel N van, et
al. Quantitative Microbial Ecology through Stable Isotope Probing. Appl
Environ Microbiol 2015; 81: 7570–7581.</a></p>
<p><a href="https://journals.asm.org/doi/10.1128/mSystems.00151-20">Sieradzki
ET, Koch BJ, Greenlon A, Sachdeva R, Malmstrom RR, Mau RL, et
al. Measurement error and resolution in quantitative stable isotope
probing: implications for experimental design. Msystems 2020; 5:
e00151-20.</a></p>
<p><a href="https://journals.asm.org/doi/10.1128/msystems.01280-22">Vyshenska
D* and Sampara P* et al. A standardized quantitative analysis strategy
for stable isotope probing metagenomics. Msystems 2023;
e01280-22</a></p>
</div>
<div id="htssip-r-package" class="section level3">
<h3>HTSSIP R package</h3>
<p><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0189616">Youngblut
ND, Barnett SE, Buckley DH. HTSSIP: An R package for analysis of high
throughput sequencing data from nucleic acid stable isotope probing
(SIP) experiments. PLOS ONE 2018; 13: e0189616.</a></p>
</div>
</div>
</div>
<div id="using-sipmg-r-package" class="section level1">
<h1>Using SIPmg R package</h1>
<p>This R package was designed to facilitate statistical analyses using
SIP metagenomic data. Particularly, coverage and taxonomic
classification of metagenome-assembled genomes and metadata of
fractionated DNA is used to identify incorporators and quantify isotopic
enrichment.</p>
<p>Overall, the package can perform the following:</p>
<ol style="list-style-type: decimal">
<li><p>Coverage normalization to absolute concentrations or relative
coverage. Scaling of coverages to absolute abundances based on
post-fractionation reference standards, called sequins.</p></li>
<li><p>Statistical analysis based on either of</p></li>
</ol>
<ul>
<li>AFE mathematical model by <a href="https://journals.asm.org/doi/10.1128/AEM.02280-15">Hungate et
al. (2015)</a> or</li>
<li>Delta buoyant density <span class="math inline">\(AFE_{ΔBD}\)</span>
= <span class="math inline">\(\frac{W_{Lab}-
W_{Light}}{I_{max}}\)</span>. Where Imax is the maximum linear shift in
DNA BD (upon 100% labeling), as discussed by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2595786/">Birnie and
Rickwood (1978)</a></li>
<li>Analyses as offered by <a href="https://CRAN.R-project.org/package=HTSSIP">HTS-SIP</a></li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Data visualization for identification of isotope incorporators.</li>
</ol>
<div id="example-workflow" class="section level2">
<h2>Example workflow</h2>
<p>qSIP analysis is typically performed using sequins, MAG coverage, and
MAG taxonomy is performed using this R markdown. Briefly, coverages are
normalized, MAG coverages are then scaled based on linear regression
models from sequin coverage and concentrations, a phyloseq object is
created from MAG absolute concentrations and taxonomy (GTDB taxonomy
output format is required) data. Note: Sequins that were spiked in the
DNA-SIP fractions will be used in scaling and creating linear regression
models for evaluating absolute MAG concentrations. Please see
<code>sequin_scaling.R</code> For normalizing coverage values please see
<code>pooling_funs.R</code> </p>
<p>This R markdown primarily uses <code>Tidyverse</code>,
<code>phyloseq</code>, and <code>HTSSIP</code> packages This markdown
uses functions from <code>functions_qSIP_MAGs.R</code> </p>
<p>For this data, MetaSIPSim (developed by <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-3372-6">Barnett
et al, 2020</a>) was used to simulate a data set from a hypothetical SIP
workflow performed on the MBARC-26 mock community (as defined in <a href="https://www.nature.com/articles/sdata201681">Singer et al,
2016</a>). The data set was further augmented with simulated
post-fractionation sequin spike-ins and the data set was simulated to
have all 26 organisms as incorporators. Some of the 26 organisms also
had plasmids, which were assembled as distinct genomic units, adding the
total genomic features in the simulated study to be 38. The example
below will also cover a case considering the same data but assuming the
isotopes were 15N and 14N, instead of 13C and 12C</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co">#Load required libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co">#list.of.packages &lt;- c(&quot;tidyverse&quot;, &quot;HTSSIP&quot;, &quot;ggpubr&quot;,&quot;data.table&quot;)</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#new.packages &lt;- list.of.packages[!(list.of.packages %in% #installed.packages()[,&quot;Package&quot;])]</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co">#if(length(new.packages)) install.packages(new.packages, quiet = TRUE, #dependencies = TRUE, repos = &quot;http://cran.us.r-project.org&quot;)</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co">#if (!require(&quot;BiocManager&quot;, quietly = TRUE))</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co">#    install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;phyloseq&quot;)</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;EBImage&quot;)</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="co">#library(tidyverse)</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="fu">library</span>(HTSSIP)</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;HTSSIP&#39;</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:SIPmg&#39;:</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="co">#&gt;     DESeq2_l2fc, HRSIP</span></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="co">#&gt; Loading required package: ggplot2</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="co">#&gt; Warning: package &#39;ggplot2&#39; was built under R version 4.3.2</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="fu">library</span>(SIPmg)</span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="at">seed =</span> <span class="dv">1000</span>)</span></code></pre></div>
</div>
<div id="load-required-data" class="section level2">
<h2>Load required data</h2>
<p>The following files are required: </p>
<div id="mag-coverage-data" class="section level3">
<h3>MAG coverage data</h3>
<p>Pooled coverages data as a comma separated file (.csv file) across
samples for <code>Features</code> including sequins that were used as
spike-ins. The followings columns are required: </p>
<ol style="list-style-type: decimal">
<li><p>Feature: A character string describing the <code>Feature</code>
label</p></li>
<li><p>Sample: The label for these n number of columns should be in the
format of “‘isotope’_rep_#_fraction_#”. For instance,
“12C_rep_1_fraction_1”. The number of sample columns should be the same
as the product of (replicates, fractions, isotopes)</p></li>
</ol>
</div>
<div id="sequin-metadata" class="section level3">
<h3>Sequin metadata</h3>
<p>Load the sequins metadata as a comma separated file (.csv file) which
has the following columns: </p>
<ol style="list-style-type: decimal">
<li><p>Feature: As described above</p></li>
<li><p>Length: Length of the sequin in bp</p></li>
<li><p>GC_content: GC content of the sequin</p></li>
<li><p>Sequence: Sequin nucleotide sequence</p></li>
<li><p>Concentration: Concentration of the sequin in
attamoles/uL</p></li>
</ol>
</div>
<div id="dilutions-data" class="section level3">
<h3>Dilutions data</h3>
<p>Load dilutions data as a comma separated file (.csv file) that
contains the following columns </p>
<ol style="list-style-type: decimal">
<li><p>Sample: Similar to the sample name as described above.</p></li>
<li><p>Dilution: Dilution of sequins added to the fraction before
sequencing.</p></li>
</ol>
</div>
<div id="fractions-metadata" class="section level3">
<h3>Fractions metadata</h3>
<p>A fractions file as a comma separated file (.csv file) with the
following columns: </p>
<ol style="list-style-type: decimal">
<li><p>Replicate: Depends on how many replicates the study has </p></li>
<li><p>Fractions: Typically in the range of 1-24 </p></li>
<li><p>Buoyant_density: As calculated from the refractometer for each
fraction and replicate </p></li>
<li><p>Isotope - “12C”, “13C”, “14N”, “15N” etc. </p></li>
<li><p>DNA_concentration </p></li>
<li><p>Sample - In the format “‘isotope’_rep_#_fraction_#”. For instance
12C_rep_1_fraction_1 </p></li>
</ol>
</div>
<div id="gtdb-style-taxonomy-data" class="section level3">
<h3>GTDB style taxonomy data</h3>
<p>A taxonomy file in the GTDB output format (.tsv format). Load the
bacteria and archaea taxonomy outputs separately. The markdown requires
loading the standard output files from GTDB-Tk separately for bacteria
and archaea</p>
</div>
<div id="mag-absolute-concentrations" class="section level3">
<h3>MAG absolute concentrations</h3>
<p>MAG absolute concentrations. <code>mag_tab</code> object obtained
from the above script is to be used as the input here</p>
</div>
<div id="gc-content" class="section level3">
<h3>GC content</h3>
<p>GC content of the MAGs as a comma separated file (.csv file). The
table should contain the following columns: </p>
<ol style="list-style-type: decimal">
<li><p>OTU: MAG identifier such as the <code>Feature</code> label from
the <code>sequin_scaling.R</code> script </p></li>
<li><p>GC_content: GC content of the <code>Feature</code> in the range
of 0-1 </p></li>
</ol>
</div>
<div id="log-scale-boolean" class="section level3">
<h3>Log scale BOOLEAN</h3>
<p>True or False depending on how you would want the MAG coverages to be
scaled. Select TRUE if you need MAG concentrations scaled on the log
scale </p>
</div>
<div id="coefficient-of-variation" class="section level3">
<h3>Coefficient of variation</h3>
<p>Acceptable coefficient of variation for coverage and detection (eg.
20 for 20 % threshold of coefficient of variation)</p>
<p>Coverages above the threshold value will be flagged in the plots.
Here a value of 20 is used. </p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="do">## Load data</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="co">#Coverage metadata</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#Uncomment if your coverage data is in the format mentioned above for this file. Remains commented if you are using the output from `checkm coverage`</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>f_tibble <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">&quot;mock_input_data/coverage_metadata.csv&quot;</span>)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; Rows: 124 Columns: 61</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; Delimiter: &quot;,&quot;</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; chr  (1): Feature</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt; dbl (60): 12C_rep_1_fraction_10, 12C_rep_1_fraction_1, 12C_rep_1_fraction_2,...</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#Sequins metadata</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>sequins <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file=</span><span class="st">&quot;mock_input_data/sequins_metadata.csv&quot;</span>)</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="co">#&gt; Rows: 115 Columns: 5</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="co">#&gt; Delimiter: &quot;,&quot;</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co">#&gt; chr (2): Feature, Sequence</span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="co">#&gt; dbl (3): Length, GC_content, Concentration</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="co">#Dilutions data</span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a>seq_dil <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="st">&quot;mock_input_data/dilutions_data.csv&quot;</span>)</span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="co">#&gt; Rows: 60 Columns: 2</span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="co">#&gt; Delimiter: &quot;,&quot;</span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="co">#&gt; chr (1): Sample</span></span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a><span class="co">#&gt; dbl (1): Dilution</span></span>
<span id="cb2-32"><a href="#cb2-32" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-33"><a href="#cb2-33" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb2-34"><a href="#cb2-34" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb2-35"><a href="#cb2-35" tabindex="-1"></a></span>
<span id="cb2-36"><a href="#cb2-36" tabindex="-1"></a><span class="co">#Log scale BOOLEAN. True or False depending on how you would want the MAG coverages to be scaled. Select TRUE if you need MAG concentrations scaled on the log scale</span></span>
<span id="cb2-37"><a href="#cb2-37" tabindex="-1"></a>log_scale <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb2-38"><a href="#cb2-38" tabindex="-1"></a></span>
<span id="cb2-39"><a href="#cb2-39" tabindex="-1"></a><span class="co">#coe_of_variation. Acceptable coefficient of variation for coverage and detection (eg. 20 - for 20 % threshold of coefficient of variation) (Coverages above the threshold value will be flagged in the plots)</span></span>
<span id="cb2-40"><a href="#cb2-40" tabindex="-1"></a>coe_of_variation <span class="ot">=</span> <span class="dv">20</span> </span>
<span id="cb2-41"><a href="#cb2-41" tabindex="-1"></a></span>
<span id="cb2-42"><a href="#cb2-42" tabindex="-1"></a><span class="co">#Taxonomy</span></span>
<span id="cb2-43"><a href="#cb2-43" tabindex="-1"></a>gtdbtk_bac_summary <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_delim</span>(<span class="st">&quot;mock_input_data/gtdbtk.bac120.summary.tsv&quot;</span>, </span>
<span id="cb2-44"><a href="#cb2-44" tabindex="-1"></a>                                 <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-45"><a href="#cb2-45" tabindex="-1"></a><span class="co">#&gt; Rows: 33 Columns: 19</span></span>
<span id="cb2-46"><a href="#cb2-46" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb2-47"><a href="#cb2-47" tabindex="-1"></a><span class="co">#&gt; Delimiter: &quot;\t&quot;</span></span>
<span id="cb2-48"><a href="#cb2-48" tabindex="-1"></a><span class="co">#&gt; chr  (2): user_genome, classification</span></span>
<span id="cb2-49"><a href="#cb2-49" tabindex="-1"></a><span class="co">#&gt; lgl (17): fastani_reference, fastani_reference_radius, fastani_taxonomy, fas...</span></span>
<span id="cb2-50"><a href="#cb2-50" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-51"><a href="#cb2-51" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb2-52"><a href="#cb2-52" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb2-53"><a href="#cb2-53" tabindex="-1"></a>gtdbtk_archaea <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_delim</span>(<span class="st">&quot;mock_input_data/gtdbtk.ar122.summary.tsv&quot;</span>, </span>
<span id="cb2-54"><a href="#cb2-54" tabindex="-1"></a>                             <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-55"><a href="#cb2-55" tabindex="-1"></a><span class="co">#&gt; Rows: 5 Columns: 19</span></span>
<span id="cb2-56"><a href="#cb2-56" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb2-57"><a href="#cb2-57" tabindex="-1"></a><span class="co">#&gt; Delimiter: &quot;\t&quot;</span></span>
<span id="cb2-58"><a href="#cb2-58" tabindex="-1"></a><span class="co">#&gt; chr  (2): user_genome, classification</span></span>
<span id="cb2-59"><a href="#cb2-59" tabindex="-1"></a><span class="co">#&gt; lgl (17): fastani_reference, fastani_reference_radius, fastani_taxonomy, fas...</span></span>
<span id="cb2-60"><a href="#cb2-60" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-61"><a href="#cb2-61" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb2-62"><a href="#cb2-62" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb2-63"><a href="#cb2-63" tabindex="-1"></a><span class="co">#GC content</span></span>
<span id="cb2-64"><a href="#cb2-64" tabindex="-1"></a>GC_content <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="st">&quot;mock_input_data/GC_content.csv&quot;</span>)</span>
<span id="cb2-65"><a href="#cb2-65" tabindex="-1"></a><span class="co">#&gt; Rows: 38 Columns: 2</span></span>
<span id="cb2-66"><a href="#cb2-66" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb2-67"><a href="#cb2-67" tabindex="-1"></a><span class="co">#&gt; Delimiter: &quot;,&quot;</span></span>
<span id="cb2-68"><a href="#cb2-68" tabindex="-1"></a><span class="co">#&gt; chr (1): OTU</span></span>
<span id="cb2-69"><a href="#cb2-69" tabindex="-1"></a><span class="co">#&gt; dbl (1): Gi</span></span>
<span id="cb2-70"><a href="#cb2-70" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-71"><a href="#cb2-71" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb2-72"><a href="#cb2-72" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb2-73"><a href="#cb2-73" tabindex="-1"></a></span>
<span id="cb2-74"><a href="#cb2-74" tabindex="-1"></a><span class="co">#Fractions</span></span>
<span id="cb2-75"><a href="#cb2-75" tabindex="-1"></a>fractions <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">&quot;mock_input_data/fractions.csv&quot;</span>)</span>
<span id="cb2-76"><a href="#cb2-76" tabindex="-1"></a><span class="co">#&gt; Rows: 60 Columns: 6</span></span>
<span id="cb2-77"><a href="#cb2-77" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb2-78"><a href="#cb2-78" tabindex="-1"></a><span class="co">#&gt; Delimiter: &quot;,&quot;</span></span>
<span id="cb2-79"><a href="#cb2-79" tabindex="-1"></a><span class="co">#&gt; chr (2): Isotope, Sample</span></span>
<span id="cb2-80"><a href="#cb2-80" tabindex="-1"></a><span class="co">#&gt; dbl (4): Replicate, Fraction, Buoyant_density, DNA_concentration(ng/uL)</span></span>
<span id="cb2-81"><a href="#cb2-81" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-82"><a href="#cb2-82" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb2-83"><a href="#cb2-83" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb2-84"><a href="#cb2-84" tabindex="-1"></a>fractions<span class="fl">.15</span>N <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">&quot;mock_input_data/fractions_15N.csv&quot;</span>)</span>
<span id="cb2-85"><a href="#cb2-85" tabindex="-1"></a><span class="co">#&gt; Rows: 60 Columns: 6</span></span>
<span id="cb2-86"><a href="#cb2-86" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb2-87"><a href="#cb2-87" tabindex="-1"></a><span class="co">#&gt; Delimiter: &quot;,&quot;</span></span>
<span id="cb2-88"><a href="#cb2-88" tabindex="-1"></a><span class="co">#&gt; chr (2): Isotope, Sample</span></span>
<span id="cb2-89"><a href="#cb2-89" tabindex="-1"></a><span class="co">#&gt; dbl (4): Replicate, Fraction, Buoyant_density, DNA_concentration(ng/uL)</span></span>
<span id="cb2-90"><a href="#cb2-90" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-91"><a href="#cb2-91" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb2-92"><a href="#cb2-92" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span></code></pre></div>
</div>
</div>
<div id="coverage-normalization-and-scaling" class="section level2">
<h2>Coverage normalization and scaling</h2>
<p>Coverage data of MAGs (or features of interest) and sequins is scaled
using sequin concentration to obtain absolute concentration of MAGs (or
features of interest). For this step, either linear regression or robust
linear regression can be used.</p>
<p>If linear regression is used, the function
<code>scale_features_lm</code> is used. The functions in
<code>sequin_scaling_lm.R</code> are used for this step.</p>
<p>If robust linear regression is used, the outliers impact on the
regression models is minimized. For robust linear regression,
<code>scale_features_rlm</code> is used. The functions in
<code>sequin_scaling_rlm.R</code> are used for this step. In this
example workflow, robust linear regression is used to scale coverage
data in log10 scale.</p>
<p>For more discussion on the coice of regression method and comparison
of the two methods please see this [vignette][<a href="https://zielslab.github.io/SIPmg.github.io/scaling-coverage-data.html" class="uri">https://zielslab.github.io/SIPmg.github.io/scaling-coverage-data.html</a>]</p>
<div id="this-function-uses-the-following-datasets" class="section level3">
<h3>This function uses the following datasets:</h3>
<ol style="list-style-type: decimal">
<li>Coverage data (<code>f_tibble</code>)</li>
<li>Sequins metadata (<code>sequins</code>)</li>
<li>Dilution of sequins used to add into the fractions
(<code>seq_dil</code>)</li>
</ol>
<p>The regression scaling plots are saved in the project folder within a
subfolder <code>sequin_scaling_plots</code></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>taxonomy_tibble <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(gtdbtk_bac_summary, gtdbtk_archaea) <span class="co">#Combine bacteria and archaea taxonomy files if it has not been done yet</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co">#mag_tab is a tibble with absolute concentrations of MAGs obtained by scaling MAG coverages using linear regression models on sequin coverages and concentration</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="do">##Scale MAG coverages to obtain MAG absolute concentrations and save scaling plots in the working directory</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#For rlm scaling using scale_features_rlm</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#For rlm scaling using scale_features_lm</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>mag_tab_scaled <span class="ot">&lt;-</span> SIPmg<span class="sc">::</span><span class="fu">scale_features_rlm</span>(f_tibble, sequins, seq_dil, log_scale, <span class="at">coe_of_variation =</span> coe_of_variation, <span class="at">save_plots =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt; Warning: There was 1 warning in `dplyr::mutate()`.</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; ℹ In argument: `fit = ifelse(...)`.</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt; Caused by warning in `rlm.default()`:</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt; ! &#39;rlm&#39; failed to converge in 20 steps</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt; Warning in dir.create(plot_dir):</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; &#39;C:\Users\psampara\AppData\Local\Temp\RtmpKKb1hw&#39; already exists</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>mag_tab <span class="ot">=</span> <span class="fu">as.matrix</span>(mag_tab_scaled<span class="sc">$</span>mag_tab) <span class="co">#Extract absolute abundances as a matrix</span></span></code></pre></div>
<div id="example-of-a-scaling-plot" class="section level4">
<h4>Example of a scaling plot</h4>
<p>An example of a scaling plot is as below</p>
<pre><code>#&gt; Warning: The dot-dot notation (`..rr.label..`) was deprecated in ggplot2 3.4.0.
#&gt; ℹ Please use `after_stat(rr.label)` instead.
#&gt; ℹ The deprecated feature was likely used in the SIPmg package.
#&gt;   Please report the issue at &lt;https://github.com/ZielsLab/SIPmg&gt;.
#&gt; This warning is displayed once every 8 hours.
#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
#&gt; generated.
#&gt; `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABQVBMVEUAAAAAADUAADoAAF4AAGYANYQAOjoAOmYAOpAAXqgAZrYLAAAzMzMzWJAzZv81AAA1ADU1hMk6AAA6ADo6AGY6OmY6OpA6ZrY6kNtNTU1NTW5NTY5NbqtNjsheAABeADVeqOtmAABmADpmAGZmOgBmZgBmZmZmkJBmtttmtv9uTU1uTW5uTY5ubo5ubqtuq+SENQCEyeuOTU2OTW6OTY6Obk2OyP+QOgCQOjqQkDqQkGaQtpCQ27aQ2/+oXgCo6+urbk2rbm6rbo6ryKur5OSr5P+2ZgC2Zjq2Zma2kDq225C22/+2/7a2///Ijk3I///JhDXJqF7J6+vbkDrbtpDb/7bb/9vb///kq27k///rqF7ryYTr66jr68nr6+v/tmb/trb/yI7/25D/27b/29v/5Kv//7b//8j//9v//+T////OCGf/AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAO4klEQVR4nO2djX/cNhnH3SNsmemOkQS468rg0jG4jS20sGVrd1sLqwft0paZ0bSFO3pbron+/z8APY9ebcuW7Qt5kZ7f5xOfLcuK/bXe9UhOmKU8QY0YSSvRe6uJIpMng9k53c7Fkwa0unFgXAsHcSvxe4lbBMgjk8QmiaUNSmJSFqCpcV0QIKW2gK520Bnd+tnIkQc5ow9/6jn8FjcOJzYnQASIABkRII8IkEcEyCNXRZEAWWrb1CBAHhEgZlKZTGKH4/H47cfyHAHSWu3K3rJHn1iuLQEB39ABscXmU/g5uXffcmwHKEmGMQASSezVLZ7EMBJhC3TeRhzQMHxAmYhBRx/ct2JR6xgUMiCZSds99jofapkHDYMG5FB3QMGWYquJSFlaL68/Yyd/6VjMBwyIseVWccyQ14Ou6YKMAAnlSc2gIQHSOt5fo7EaAyC3CBBoNZnyot455EOAQFnK8o2DPHX4awdoGDYgHoGO99N1ehQ5n8ABrSYjAlSQDeh4f7TghTxPaFURIBCvKqaqrVoSAfKIAHlEgFC8pTHNKYnZKtaDNr8TJX1VrQANAweExfx0jWIe+BAgFi8glkMSg7piVS0AYYd02IDYotbQ3g8oQUBJ2IDqRYA8IkDrm7/EkAfVq00pNgy+FGtQO0C4FzKgfI1SLAZAOeQ+vetB4QOSsxF616QJEIscECUxhyiT9ugUi/khAWIeQGIvYEBgANO/qRE+IHdnqxABYqVZmSURIAYxqAGQ3waYA0KFC6hpsjPFIBBl0lWVjBdq/fkBDcMHtF4mDbZTuBcuoMZMmgBxLXf6Z9LQIR06oNKEuoJ8gOSQBgsaUJMIkEcECLROPSiGPIjXg3hB5i7svYCGEZRigCYbqTmrRbUAxKIAlPc1JI8BEFhIczpOI0UCBILGWOaeMEaAPCJAbM2BQwIUPaBcG1ClDn8+QMPwAa3XHyRsp/AweEC9khgB8uZBLHRA6+ZBLHRAa+VB0m4BN+ECahIBAvXvD4oD0Br9QXEAWqM/KB5AvfqD5CSE4AH17Q+CfAt77IMH1LM/SM3zCR9Qg5oBAZ/QAfkW/W/Mg+SYGAGqK8WU5QtuggXU+OERAuTSyV293i0BculwTIBc5i+qw+Pow48JkHO2j7AoP7n31d2WqwErC+DAzYBNjyKWYod7lAc1ATr66FlbQMNIAJkJddAcgyXtx+M9ecoDiMUBSKzdwZv0sjnWPgbFAqgkAnRKfdLxAOo5ZzUaQH1nPbcElCc1q1XDl2/5351tfXK5XfX3/cPqVTX/yb6cX+YKra1OZWS1FaDjfeipzJPqkAD836YxOaHyU/qvcF7WVacBaNgKkFzh09GhC8/gf46yj5ZPfpqA+iYxY/mCGzcg+4Uf74vmHvxuHMBo3Gv8L4VnkafULr+h5fYdfnJqVtwXXuBYhDESd46DeiO23Pks2XgAyPEYLysEjKH1BNQzk24FyHqT0NRTfxCjVAzif+AEKOUunt3icQ4AyADUleoYoiSnhC8AOQm++tgEjFfJ0HoCqtf6gMxMIkzDqsXHH8QCpJ6a/+qzW1Nmn1JXW17h74enYl9518fitL5Knj5DQGJQrEsMwpFJfqw+b28BUvlf6WwBkLzaxCjRNMJmwGCmvNvHELB9VS9Afdcwk2M+rfOg1a6+WZVf1wCyzjYB4g7/4oxWk8HMJmAfnwogj9YGpIovnv5hVW9MYoNCIiknMetsAZC8Wj/lavcOT75IYGFikH2sQlNXXUxAph5kMmnuYp6hkJfCrjmrc11WzaR5BQKKJyCw3LIAyWMry9dX9UtiN6yEVThoBgSDYm3yIHwQWZO2i/nBzI5B5WJensVNJnuCLS9CCyy2eeiDz0WOL73jMVxWCrhnDOKZkKweuOrwDaWYZX3XDOhSqmi8IAsOhz8C5BEB8ogAeUSAPCJAHhEgj+oBDSMC1MtO2jZwxU24gPot8hYRoH5zNSIC1G/9oIgAdVjkDZp+vLEdGaAO6wctUjbPprEBapIjieUEyFIV0PMUd4ZRAeow7JO9wSSgcszqBSizywf3WmrctXLC5eYORR6pgijHp1yKMTKet4j+4OUvKqOaXQYOZT3y9Rlk0lNZipUBrX7tA5Qkjmh7xoAEidWu7OfFWU6gvPLsXYaeoZuXM/j75reia23ExCwEOG9i1hUPIIRccc3eU+8SBlP5w+Ae3Dr0j0IPPDg82Pmt5Y3Zbsufv7txIN0XYki24FmG+fq7+IzifSxS3IUOZsmv2tfcBRBEsDn06N5UNOZJcZoG3EJPQJtP4R6zEbxLucex8DcK3fz4YiG2bIn5fsJbwQ1HBIU7PCWEYjwvdOii/9riwGMRRCeITIZcDaDGJDafz7/48bdg5vv8R19qk98kefMn+MRXbuLx8zdevNUP0BTfJb9NfvP8YeARxL1/fTs9/nTGdHKCk8JbwQ1jiHCXD6s8y5DEnkhiJq+BYUeMnxJQJY11yKT/A2YCc/bPrZGpSZdjEP/fPkDuPOjrGQKaiNFQ8bycy+rGNzce7HxTgiG9VQFJd55Zwtac4CGp3AqjiJ51irHBjkEeQPXiT/1C5NGv/d5ualTyID8gpzIYLpvKqG/eO8t+98vjT2+nrAhIp5ASIJNyZKLyxCBMl8zKg9YC9Bze2T82HhbaYsNyKdYXkJ3zmD2eZ4yU0ZWBYaYeFwFJd7xOA2rIgwQfMbqfyvuozYPEt+abmhpPwH2eySaYAvSmiFdXZmsCeg/TDL8FmThkgYRl2IYsoXmJpYsjMXRn3BCQdM8SVW6ZkDCl7RdLMWEfMTX1IE8p1iT+1F+kwABeQSEGsVOJQWcsR40QtU49aPUOXDyHXrXLD8hBAuSpSTcDWv4Uzs7BDD8AQK217vI4FUAhN1b7dLlGBahJBAjUfdhnGBWgHsvjlAxccRMuoB7L48QHqOPyOHEB6rE8TmSAui+PExmgBhEg1mvoOS5A3WvSw7gAuSa7KdUBqjiFDKj7t30iA9QkJyDRIR0PoK5zVuWQBgECEaAeHWaRAepRzMeWBzXJCahiIo2bgAF17TCr2E7hJlxAne2kYwPUOQ+KDVDnxmpsgDp/DDs2QJ3bYhrQalcZnfEwwgXUpGZA2UACWk3SgGNQk1yAlIk0GJ1IQIvBLGhAvLkx1Z1CL8fjtx+rM2BABSfmWcq0QbDEskgtu86Q8yCWbX43mara0NEfHrPDX6lTAAjyJtzoGKTiUySA0MhsapdlAEkIbBRhgYc/opEQYwQIJWKQXA343+/8Zv5EWAKX1wB+8dZN7fokXEAshyRm7KSP3r92X50SZsAqphTzoEIM8lraXzI51nK1jNNe/UkRuor5c5ZapZhl2GEA5RtfhgyookdqMVecisCrkB5Ay+2wi3lbL68/K8WggoFrFdDx/s+wWzJgQDA2b/o8DsfjQh40z20T6QKgSNpiYN1R1yski3kLBPS3RgaocVRjNbkizIDlRk9jIUCoSlssRkCd1nLFtd1iA1SpBxlVW/OlxQHNJmRA9XIBcnURESC9iRFQp3GxCAF1GxeLEFC3cbEIAXUbF4sQUKdxMaf9Jm7CBdRpXMxp2IGbcAE1iQB5RIBQHdYPihJQl8ZqjIA6dXcQoKIIEKhDEnPbb+ImYEAdMmkx7BwdoHoRII8IkMyjxfdTqiJAuGgcKHO2WIuAErkyV1SAcNSwsGOrAChRYz4xATK9Zf56EAGq+isBGsYHSHxMEdRi5YXK6okRANJfRjSkbJVKMWnZERUgluGaFKuJs5wnQEwOi9V8HpgAeUSAPLo6L2o4r1WkgCgGNasISI2JESAtAmTpb94u15gBqfX0SyJAQnlS8zF2AsTk2Hzq9keAIPLwWrSzM4hVi/n4AMnujnaAooxBC8x+WgHSY2JRARKf4W6VB8UKiKlo5BABUuLRyFsPihoQ13/9gJL4ACkDxRY2imbpspgAaWVO64UCIPy+U6SAVhN3nysBElokNcu5lgcO48uDUFldNajpc+rxAOIlfO1cBAIEoz71MxEIELTm0wZ/BKhDPShOQB4RII8IEJOTwmsKMgIE31MUXwduY93BIgSk7KZq7IM66Ixu/WxkWZiptFX38Yir1T2HEwuLT2sbRUaACJBTHQDFKQLkUdumRrRqW5OOVgTIo7aAXt0aX38m9+USuEfvj8diHUp7YeWTu584Lr+8agkIHlutnfxS0IBFKI8+gIUWCwsrH46jBPTqz49VxHl07a+48xKQ6KVM1cLKRx9+HCWgo4+sdUvNKtNmKVMZg07ufRVnEiss7KoBndzdky5qYeXDvUjzIGcMenVrT3sQJ7m3SAFZeZBVitkoMDc6HIP2qtdfXrUuxfZMQSUAGT6F9BdpDHLUg0RsQRr2wsqxAopWBMgjAuQRAfKIAHlEgDzqAOh4f7rcrtjnLbearELs7ltXV+73D5kKM0sGs9Wue871OaoDoDxlLkBVJ0seQHCxDgB2FvUfdD8ntQcEb/f/Dqhpvd3zUQITNPidL3c+4z+4tMAIe/AHd3YOmDyJggFX8QxiQof0g8+IFyvPdiC35dVqX/237dtJsvkUfI5wH0wmkNSFi0IJLLKdw72mcugn3ziAhfBWE3jkVI9E45A9fwZwgz/lBwHBxcpzORC4uBwgWPvBvriY/4Mcqc18EfIclIiVAafLLf5YPwAKfouYGPg9L+RJ8Ig/6txCnssVoKlMQdxXORC4uBwgXsD96It1Wmta9PtcJIfCRvLNLXD1Dow0y50D+YV1tPbQgPDctvajH814LgYCF5cDVCz0xXrHbVtyjlKG4/KWBzP74W07Dz8g6bkcCLiVA7xMgOTMA5NDLgYmGVnTEjQgcFNJbLFhACnP5UDQrRSgSFYW3QucxGBZQH7f+hmXW7y+pvNUcRJ9Zg2ZtHj3wrMJJLUy6bQQoMqkAUcR0MXLpKHgHajaSMb3P+c3bZfKKhYtHMX8bQuQ9lwOhDFWDhCL9hT/YVoEVGe9dW6qryiWaySuZgD30+uVOy+6oBVFlyOkkupXSPLU5ed0AV24emJNDMpd059Lb1f6aW6s1sgF6PI3VuMUAfKIAHlEgDwiQB4RII8IkEf/A4dwQ0gzTxIqAAAAAElFTkSuQmCC" /><!-- --></p>
</div>
</div>
</div>
<div id="preparation-of-scaled-data-for-qsip-analysis" class="section level2">
<h2>Preparation of scaled data for qSIP analysis</h2>
<p>The scaled abundance data, taxonomy data, and metadata is then
converted to phyloseq objects and a master phyloseq object is created
for qSIP analysis.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>mag.table <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(mag_tab, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>) <span class="co">#Phyloseq OTU table</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>taxonomy.object <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">tax.table</span>(taxonomy_tibble) <span class="co"># Create a taxonomy phyloseq object</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>samples.object <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">sample.table</span>(fractions) <span class="co"># Create a samples phyloseq object</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>samples.object<span class="fl">.15</span>N <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">sample.table</span>(fractions<span class="fl">.15</span>N)</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>phylo.qSIP <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">phylo.table</span>(mag.table, taxonomy.object, samples.object) </span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>phylo.qSIP<span class="fl">.15</span>N <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">phylo.table</span>(mag.table, taxonomy.object, samples.object<span class="fl">.15</span>N) <span class="co"># Make a phyloseq table for downstream qSIP analysis</span></span></code></pre></div>
</div>
<div id="estimation-of-atom-fraction-excess" class="section level2">
<h2>Estimation of atom fraction excess</h2>
<p>The following steps estimate the atom fraction excess (AFE) in the
MAGs. This calculation is based on the mathematical model suggested by
<a href="https://journals.asm.org/doi/full/10.1128/AEM.02280-15">Hungate
et al. (2015)</a>. In the article, the authors suggest that GC contents
of the individual biological features of interest should be accounted
for to have a better estimate of AFE. Through the power of metagenomic
analysis and the recovery of MAGs, one can estimate the GC content of
features of interest and thereby obtain a better estimate of AFE. Thus,
here the user would use the GC contents of MAGs in the AFE estimations
and obtain isotopic enrichment of individual MAGs.</p>
<p>An important consideration is that the AFE calculations are based on
a mathematical model and are not an absolute estimate of isotopic
enrichment. One of the goals of AFE estimation is to determine isotope
incorporators.</p>
<p>In this example workflow, the approach reported in <a href="https://journals.asm.org/doi/full/10.1128/AEM.02280-15">Hungate et
al. (2015)</a> is used to determine incorporators.</p>
<p>For this step, the following are used:</p>
<ol style="list-style-type: decimal">
<li>Master phyloseq object from the previous step</li>
<li>GC content</li>
</ol>
<p>Thanks to <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0189616">HTSSIP
R package</a>, bootstrapping can be performed faster with
multithreading.</p>
<p>Here, the package also gives a message if the number of bootstraps
resulted in a low or high coefficient of variation of bootstrapped
values. If the coefficient of variation is &gt;30%, it pops a message
suggesting a re-run with a higher number of bootstraps.</p>
<p>Additionally, the AFE values based on the delta buoyant density
method as discussed in this paper <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2018.00570/full">Youngblut
et al. </a> can be outputted (<code>show_delbd_AFE = TRUE</code>).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>atomX <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">qSIP_atom_excess_MAGs</span>(phylo.qSIP,</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>                               <span class="at">control_expr=</span><span class="st">&#39;Isotope==&quot;12C&quot;&#39;</span>,<span class="at">isotope =</span> <span class="st">&quot;13C&quot;</span>,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>                               <span class="at">treatment_rep=</span><span class="st">&#39;Replicate&#39;</span>,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>                               <span class="at">Gi =</span> GC_content)</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; Warning: `mutate_()` was deprecated in dplyr 0.7.0.</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; ℹ Please use `mutate()` instead.</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; ℹ See vignette(&#39;programming&#39;) for more help</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; ℹ The deprecated feature was likely used in the HTSSIP package.</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co">#&gt; generated.</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co">#&gt; Warning: `select_()` was deprecated in dplyr 0.7.0.</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co">#&gt; ℹ Please use `select()` instead.</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co">#&gt; ℹ The deprecated feature was likely used in the HTSSIP package.</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co">#&gt; generated.</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="co">#&gt; Warning: `filter_()` was deprecated in dplyr 0.7.0.</span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="co">#&gt; ℹ Please use `filter()` instead.</span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="co">#&gt; ℹ See vignette(&#39;programming&#39;) for more help</span></span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a><span class="co">#&gt; ℹ The deprecated feature was likely used in the SIPmg package.</span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a><span class="co">#&gt;   Please report the issue at &lt;https://github.com/ZielsLab/SIPmg&gt;.</span></span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a><span class="co">#&gt; generated.</span></span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a><span class="co">#&gt; Warning: `summarise_()` was deprecated in dplyr 0.7.0.</span></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a><span class="co">#&gt; ℹ Please use `summarise()` instead.</span></span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a><span class="co">#&gt; ℹ The deprecated feature was likely used in the SIPmg package.</span></span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a><span class="co">#&gt;   Please report the issue at &lt;https://github.com/ZielsLab/SIPmg&gt;.</span></span>
<span id="cb6-32"><a href="#cb6-32" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb6-33"><a href="#cb6-33" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span>
<span id="cb6-34"><a href="#cb6-34" tabindex="-1"></a><span class="co">#&gt; generated.</span></span>
<span id="cb6-35"><a href="#cb6-35" tabindex="-1"></a><span class="co">#&gt; Warning: `group_by_()` was deprecated in dplyr 0.7.0.</span></span>
<span id="cb6-36"><a href="#cb6-36" tabindex="-1"></a><span class="co">#&gt; ℹ Please use `group_by()` instead.</span></span>
<span id="cb6-37"><a href="#cb6-37" tabindex="-1"></a><span class="co">#&gt; ℹ See vignette(&#39;programming&#39;) for more help</span></span>
<span id="cb6-38"><a href="#cb6-38" tabindex="-1"></a><span class="co">#&gt; ℹ The deprecated feature was likely used in the SIPmg package.</span></span>
<span id="cb6-39"><a href="#cb6-39" tabindex="-1"></a><span class="co">#&gt;   Please report the issue at &lt;https://github.com/ZielsLab/SIPmg&gt;.</span></span>
<span id="cb6-40"><a href="#cb6-40" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb6-41"><a href="#cb6-41" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span>
<span id="cb6-42"><a href="#cb6-42" tabindex="-1"></a><span class="co">#&gt; generated.</span></span>
<span id="cb6-43"><a href="#cb6-43" tabindex="-1"></a>atomX<span class="fl">.15</span>N <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">qSIP_atom_excess_MAGs</span>(phylo.qSIP<span class="fl">.15</span>N,</span>
<span id="cb6-44"><a href="#cb6-44" tabindex="-1"></a>                               <span class="at">control_expr=</span><span class="st">&#39;Isotope==&quot;14N&quot;&#39;</span>,<span class="at">isotope =</span> <span class="st">&quot;15N&quot;</span>,</span>
<span id="cb6-45"><a href="#cb6-45" tabindex="-1"></a>                               <span class="at">treatment_rep=</span><span class="st">&#39;Replicate&#39;</span>,</span>
<span id="cb6-46"><a href="#cb6-46" tabindex="-1"></a>                               <span class="at">Gi =</span> GC_content)</span>
<span id="cb6-47"><a href="#cb6-47" tabindex="-1"></a><span class="co">#Bootstrap confidence intervals</span></span>
<span id="cb6-48"><a href="#cb6-48" tabindex="-1"></a>df_atomX_boot <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">qSIP_bootstrap_fcr</span>(atomX, <span class="at">n_boot=</span><span class="dv">10</span>, <span class="at">Gi =</span> GC_content, <span class="at">isotope =</span> <span class="st">&quot;13C&quot;</span>, <span class="at">show_delbd_AFE =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-49"><a href="#cb6-49" tabindex="-1"></a><span class="co">#&gt; The coefficient of variation of bootstrapped values is &gt;30%.</span></span>
<span id="cb6-50"><a href="#cb6-50" tabindex="-1"></a><span class="co">#&gt;           To get more stricter inferences and narrower confidence intervals,</span></span>
<span id="cb6-51"><a href="#cb6-51" tabindex="-1"></a><span class="co">#&gt;           please consider increasing the number of bootstraps.</span></span>
<span id="cb6-52"><a href="#cb6-52" tabindex="-1"></a><span class="co">#&gt;           With narrower confidence intervals, you may have a lower false positive rate.</span></span>
<span id="cb6-53"><a href="#cb6-53" tabindex="-1"></a>df_atomX_boot<span class="fl">.15</span>N <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">qSIP_bootstrap_fcr</span>(atomX<span class="fl">.15</span>N, <span class="at">n_boot=</span><span class="dv">10</span>, <span class="at">Gi =</span> GC_content, <span class="at">isotope =</span> <span class="st">&quot;15N&quot;</span>)<span class="co">#Change &quot;parallel = FALSE&quot; to compute using a single-core</span></span>
<span id="cb6-54"><a href="#cb6-54" tabindex="-1"></a><span class="co">#&gt; The coefficient of variation of bootstrapped values is &gt;30%.</span></span>
<span id="cb6-55"><a href="#cb6-55" tabindex="-1"></a><span class="co">#&gt;           To get more stricter inferences and narrower confidence intervals,</span></span>
<span id="cb6-56"><a href="#cb6-56" tabindex="-1"></a><span class="co">#&gt;           please consider increasing the number of bootstraps.</span></span>
<span id="cb6-57"><a href="#cb6-57" tabindex="-1"></a><span class="co">#&gt;           With narrower confidence intervals, you may have a lower false positive rate.</span></span></code></pre></div>
</div>
<div id="plot-the-atom-fraction-excess-plot" class="section level2">
<h2>Plot the atom fraction excess plot</h2>
<p>Plot the atom fraction excess data with 5 % and 95 % quantiles as the
confidence limits. In the same way as reported in <a href="https://journals.asm.org/doi/full/10.1128/AEM.02280-15">Hungate et
al. (2015)</a>, a MAG is considered an incorporator if the lower
confidence interval of its AFE is above zero.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>CI_threshold <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>df_atomX_boot <span class="ot">=</span> df_atomX_boot <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Incorporator =</span> A_CI_fcr_low <span class="sc">&gt;</span> CI_threshold,</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>         <span class="at">OTU =</span> <span class="fu">reorder</span>(OTU, <span class="sc">-</span>A))</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>df_atomX_boot<span class="fl">.15</span>N <span class="ot">=</span> df_atomX_boot<span class="fl">.15</span>N <span class="sc">%&gt;%</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Incorporator =</span> A_CI_fcr_low <span class="sc">&gt;</span> CI_threshold,</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>         <span class="at">OTU =</span> <span class="fu">reorder</span>(OTU, <span class="sc">-</span>A))</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>(<span class="at">atom_f_excess_plot =</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(df_atomX_boot, <span class="fu">aes</span>(OTU, A, <span class="at">ymin=</span>A_CI_fcr_low, <span class="at">ymax=</span>A_CI_fcr_high, <span class="at">color=</span>Incorporator)) <span class="sc">+</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>  <span class="fu">geom_pointrange</span>(<span class="at">size=</span><span class="fl">0.25</span>) <span class="sc">+</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>  <span class="fu">geom_linerange</span>() <span class="sc">+</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">&#39;dashed&#39;</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&#39;MAGs&#39;</span>, <span class="at">y=</span><span class="st">&#39;Atom fraction excess&#39;</span>) <span class="sc">+</span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Isotope incorporating MAGs&quot;</span>))</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAIAAACb4TnXAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAgAElEQVR4nO29f2wb55ku+vBirSKRmkSIZV10SCzHKyU9DoEUamrC1Oq0LLCblQJNhV5aKASEvg26lNnFMUXgWsUuja3OnrBAFSASC9xDW3tu7w0LqAuZWKiTY2qNC4S7V2s6TPYK7YHi7praDgtyFldWCiVZ23tqX4Dnj29mOMOZoSiZQ3Ko74H/IL/5fg09r77ve+Z539dRqVRAQUFhDf6ndk+AgqKbQQ2MgsJCUAOjoLAQ1MAoKCwENTAKCgtBDYyCwkJQA6OgsBDUwCgoLAQ1sGZhJzHqcIwmdp64m9nR2Y2mzOhQozZl8p0MzQ9r1f1uzDpGR2cTG9WOqYF1FnZurK3k2j2JbkTLfthcbmVu4oJiu9TAKAAMRW5VKrciQ+2eR6tg6f36fL7c2g3JwqiBWYSdxOyoQ8Lo7IZ2M7KzYXRxJzHqGJ7LASsT6v2LcWUAG7OkmnJ9dLZ206NqO6qbhWa21S2T9HljIzFq0u9OYyNqZ7uTGHU4Zjd2EuTy7MbO4eZ/UG9ydaV+tTv9D3vU+3WMziY2yNBmPyXgmZ5WWViFojkoLPsA33JB+axFKCNXzIR0F0kzTSupJ9PKlUolEwJ8oZoKylXDWaiuHmbyNfM37bPebMkQPl+1+KD5H64343mHMoY/7CHu96CfQ4NMCEAoU1j2yb1TA2sWVP9n5GMoo34Y5P8S6f+r5qLy5BSWffr/XJPK8pMuXS5k5NqquqpZZNTTMJ+88kTJLQtG3UpVM6HauZvOtuZGG5x/473Jv7KuQPVN+8M2dL9kkr5l7RwPMLBK1cKogTULtQYGny+0vJwpFPS1av571E+p5jk4oLLm6VZdrf7VrlmwtA+ZyeQNRq32athtY7dm0m3d+R+mN3kWhUIms7ysLIyNGpjJ/ernaDqypmUmJDWjZzALMBS5EiJ00tzE8LC0c1ft6kNT4+rq41Oh2h5UOKCy5wX1SX34jA/YvrsDFO7kgNzcsEOF4bmcdPVA+M4M1/RqOuZhZqvptu78j9DbTmLU4RgeHp6YmJtbOSRlaHK/O3e3Ad/0a6o5Dr02bbRnrMH4VAi5tRs7lOSwBuPXKoXMsnKGyK3MTQy3/u3WscLG7PBcDvCFQsvLmUymUNAdFlsK2cKogVmFofHItVtk05JZDgFYWZctrPoJALCxvlKnnwMqaxaknRtrOWlRGD5jsh1sAjldZxE81K3p+lLN/7C97dzdBkKZyq1r1yKR8fHxoaGdu9sHjH0whl7wAFVGsDrHgzE+FUJu7i1qYBZgY9bhcIzKu8KhoeEXzvikXQjZYKxMqLn5iZWabYj80DVSOTd3ISFz1IkLczl5XzX02rQPKxOj1b3pzkZi1FGfYD4YQ69N+5Cbu6AQ4ORWdxqcrR5153/Y3lbWldobsxfmcqgx4Ma2x2qMT4WM5thw021KcjQLBzG/Kna9DnuutGyQpq9S1RJUi5bRFsmEp9dNXn+uVzrWddvAGwjDbg+e/2F6M6bTaylOM5re/H6PQtOrvtEVzAoMRW4Vqicw+HyhTEHZmo1f015UX8NQ5Aq5lLtT2DmoMgB4rtySH0RfaDlTuFalBcavVcxncXSMXytUH31fSD3kgbPVo+78D9PbUOSdjLrycqaQCQG5OwXpsvaHbRhDkVva+11u+Gg3PhWiL5ptDO2fS/vBpvNXGPiGQFcwCgpzEC3WrFq3NrFS90VFDaiBUVCYQ+JalPeJwxMrgG/58vjBTQmogVFQ1EHd43QDcFRoZF8KCstAVzAKCgtBDYyCwkJQA6OgsBDUwCgoLAQ1MAoKC/E7zexMTEfCKXeMj3rVJaUZpSC/xMWz0iU2mEwEmHrdybU1NY0KxXQknBIAAP4YH/WqCySQctUEpALTmtqbEgwvqfDKK6/UuxU74B/+4R/aPYVuRDNFJOXrlyYnJyfffl9TIn0tX7+kvvT+25OTl66X6/ZFatf0oSt8/22lo/L1S/pOleuGHRnWNCwy7FzGl7/8ZeXz3t4e+fDf5/8D+VdTXoPLly8bd2repOnl6vlTNBFN3yL6YzHEl/K68vxaCsFkdQHwRpNBpNb0FSWUS4Lf5wUA5twYm83lzQrzuax/RlrLmMCMXyiVNeMuxYvB+QADgAkk5BVI1adRzWpZ/c7r4bffu2T4meJYoalbRAJvNJbjlvLa3VQ+l2XHkpqHlwkkeNNOxFKRdU1LFV1ubJZEeGFU6I3y1YHyuaxSAwDE9GrWP8PrdqLi7U3BP6PdCRrVrNt5zbbw448/Vn/+vLYnclVdR8H8/Lxhub7bVpZTNAUWGBgxsUha9GpXA7er7olLi3JJgKuhwirEdCReDCaj1WHIsumtrRVOCWC15UY1D+hcc2h55ZVXTp48ST5//PHHJ0+e/K22/cmTJ0m5vmuz8jqXrC6naBYsYhG902Obi2mxWuB0scWSaN6gFk4X22ChhPwSF94c0/Im+VyWHTtXY9ZMIMHzPD9TCkeqEzSsWb/zg/C5H/7I8DPFsUJTV7DdbRFbP8tHvV4wgfmxyGL66X4R4l3Ay7jcwurtTCl1VWERp6aw/umMCS/HuNzCz/53LrUNwDnihNvHADAsBLYWAgtbjwCk1vIBpb/8n8WzAMIR1FqGmI6ES6+QAxUDiOnvkJpcyoAnJNToxSCuqjtvCNSuKJq/gm1LFAcTmB/bvLr1SCr2fnVETF29c56XcP7h+ro48lXTB/bEpz3bRX+S5/mFU9vlwZNes0JxaWZhq/88z/N8MliMS8uSmI7EHw6ywSSfHNsMkxnll7hIWswvceGUgL1fZlmXE0B+Kfye08kGk5r2Eoh1nfdnr2q5/Ibx2+9dIv/q1FlcXDxS3xQ2QNMNbPC8wiIygfmpfrk8/3dbTPDimeuchOtPT00xW39nyiI+fvaRx50Ncxy3cM/j3P04b1Io3vz5fWD3OsdxXDglQEgtZgCUS4Ln6V1ATRh6oxcdqbD0Hqw4QNa1fC7rP/cs4QYVnlBMR7ilPOFCgOx10iQb5zgDftQclEikaOoWcdDD4NkXVCwi8+oku156ATKLOBEITERVDd4w7UosFdmR+cQPGADIL3GrMouoK3x1kv0F5qVNYH6JW3W9rDQnhTLfyGBimZ+AtEWUtpeEJ4y+DqDKEyoEZyDBB8h8tG/MARzEIgKoIRJxJCqPsoi2RrtZRL2SAgDY4EV3oywi43ILcXI8EtOrWSBoVvMAGPCE9VGfRQTwW10TM8qOsojdCotIDu/02Opi+lzi9LZEchAWMZ8L10ilGGWN0EJMb94s/Z8c9wEA1n+WFCosopiOhLd6QWzIG10YCSxwWQDw+NkHck1dc7llSgDQlxPhZXQGLoSXXEa8y658I4f5PT73wx8pO8PP/fBH/2qyVszPzx+mVwo7wSqSgwnMj20upv8/uZRxDQh/FY4jJpMcMXcqHODeNDvSME9DyD6M8TzPJ1/++QfodTFksSqVJZbi4Z60JuaXFva/lSQkx72sgKcZk+ZAfilcmuH5ZJBFb3YxLZLNYDLIAv4Yz8f8YIPTehvKL3ELCltzOChEIj2DHU9YR3LUsIiHhgPYq31z5kRfNh4vBGNBJ+7tEtWUmPsQn3xWbdT7vGnzfC4rKa2AMzFyQqsKlI2kUkB+iePixeDFkZ4j3YXarj7/1l8cqQ8KG8NSqVRgfurd8DoAIJ/7gP1WcqYU5jipGhtMps1f3IoPK6z/d1NhLkX2eMWSCC/DRFdj4OKpeAo4fZHs5BjfV/DhephbB9DfD3bEadYcpSLrckl6enmLeHtTAIQ4J+1cw1wpxkedVUrDG+X5KLC18Fe6ORqSHJ9/6y8+b3T6IlDb279e/vOatoagJIe90UzlcFWj/v7bNQL29982UK/XgUbYrlO514rh3397cnJyUiXPN2yu1vOXr1/SdGkgpDe5NWMYqukrKkG9WlZfA6qm72JYwiICMslxTi6QpFLe2iXLjEVcGDNXRel72BxL8jxD9nNLMT7qNRNVKUcsoiwRA9IaaqBFbg7UPMe/Xv7zzzV/BIqORvulUsYsIvKlBSNVVHUgidMT1/5awH0hzKXkiz/NRL0TJkqrxxtxLlUGAI9HIiGrNh7mUqZOlUdiEQkUnsOMRaToYthLKgWgltNjpr/Jgg0mSaeDAPuHE6bN98pbj4moakrc7uk/zQBAuSScHpR7MLSuJ2AR1fj8W39xoGyKostgL6mUEafHBBIxdypMOt0fWUhMmDUXSw9Y/+8SUdU643m0/ysRgFgqnqrsageXpVLGIx4JdWRT9D1YF8NWUilG5vTE9CaefUGp7I3yfJSImRZGzJsDgOvbPH9FLgQAlEvCPQEQ1BtElS+o8YgAGpBKqWHof1mn/oGXKItoC9hKKlUPWo/JxkVVYqlYDaAjpiORtLNRt68DpVJq6P0vlc/U4bKL0W4WsZ5U6jAD1tCAxiyiN5oMRsIcB7DBWJBd1U2AcbmFHHESazZqZFPNH4CiI9F+FvEwDpfqgVScXtVjsiRxgIxrQEiomMXB83LgmgQfgJiOhH+2Bfc3GNJas4h6DDORPgGLqOBfL/+54XKxuLhIXcK6FbZnEdUekyqHSWdv3/0+f5InwkP/d16XWkbSGSJlLG4T1VR+KfzXwllJIpkMsv5vNFWLqEcjLpgUXQP7s4hiqcgO/06ZDFgNrOYaODWQDXMcF94ck6PFeX0eQV5CT0lKK+/01ImeD+JkUouYj3phCYtIQF0wjxvszyIyLrew6UomAgzE9GqWHZsGFG4QAIRqpA7vD3gekvfk89JI5U/3lcVJjtJhAYtoCKUaZRG7FV3AInqjPJY4joMqnnbj3GCrWEQYPcqk2vz8PGURuxW2crjUDCRTDmI6En73E1Iu3JR4DiaQSCIisRz+837hY4kbVILTs78hTRhNVICtXgwYsojNIDlAucTjB1s5XBLUUA7lLQH7w6RmsAL8/CMRQH4pnHJLhcXrH+IBQIICIMbzyaATe1e18WvE9KJp4KimkhwUxwr2crg0ohxO9PSg/wtOAGBcp4DHnwLIrGZ7JIqSCbx6+n5frxPq0PYnMHBKCm2/xEXSophe3HT7WTzcUzwyzUd8AlCS47jBXg6XRu6Pgx7m1L7kcInB0xhwAXh5zNn77oI0llyojneP3lOkT280WfpuOFwGIChenOLRHS71n+uAVFtcXKwjR6Qkh61hAYuoJjkUFhGA28V4Azwfrd+HhHJJgGu+qhssGg4EAIzLfU9wV9OC4fcBMK7h/f3hmkJFP8UEErxL1iICOMEGk4kAqtyiPi+FekQZzSI5QKNKdS/aLZV6codLQwGUYaGhfkpML26OzScY4BCR848MSnIcN7RfKvWEDpdqCKn4Q/i/w9QU/qxXUkW53MLq97nUPQA4PQi3j4GY3hQElbtmnCsaZt5sEosIalfHDF0glfrJH4f/qn+BuFH27+KjkkgKU46LPM/zCyN3t/vgBIDf/BrC46kkz/MLI+XdPjhJYG34Y1JrwBMzsK5ms4g1ainqD9bFsL9UKv/xrueFrQWO47jwLyYv+u+VyoD41NOn2F9d5TiOW9gfPXu/eFuUiJPh9bBciDI0aS+f6UXfdi4PC6VSoETiMYP9pVLaJJRLnEwU9o4l+YTcvAgooir+Cgnoi7Fva4Zkzn2dTZV8XlgolTpU5ssDu7K0nKIp6AKplLovJbi8kW+loahKMw22bpLLWhyBRTxU5ssDu7KunKJZsL9UChor9c/ISVP0AesNRVWa5sJNg3R9RiM+AfREIvUH62LYXyqF/FL43U9IcPmqQ1h+aWFrNKYNWF/eEoCvJvWiqs2xJC+XCpu/0E2GSqUojgr7S6UyP81i/0tysBo5g17uQ/Q/61R6evBQJKIqCYzrFNBzggF+8l+yfW4pOzPjOgU4DhzxyUBJjmMF20ulxIcVAFkluDzAvgg4XQOnipJ+ivWfRREARsZHkZBFVT1wfuvrAF7/y+RTkXBVVMX+/ssWSqUoyXHcYHuplOoQJwXRTkwA0OqnBHwRALxRPgLirsJ+SzlqyT1IzQMMAKukUoYkB/UH62J0gVSKIL/ExVUiDCOplGEUe+PmFoKqpY4VukIqJeX4So6tqvJTlkvEbqtSqXxJYF9+eZHjBACnWPRKFm/Y3ODWmsMigtrVcYL9pVJyBr3pkspjUlyaiX/oT8pSKdLc6RoU1v/bq2R4hwA8zZg0r4E1LCINL3UcYHupFMmgl41zZKeZjXNLeYCJrsa+kg1zmuaM6wwGiX6Ku/50MJmYMGtupVSKQG1XianXmto3RQfB9lIpJpDgA2I6soj5eSzK3B8U+o8wguNeqc9Xg+5UKlsdx6S5hVIp8qF+qPpGLlEW0RboBqmU4tKVNu5cJYASUqUZyWAii+lziQBTp/lBOBqLSD7XCVVv1qSV5RTNgv2lUmJ68b3+x2XJpcuj6kE23d4tIoAC4BwpxMk7L48HD26LgQBMmtcZsRlQc4m/9X21Sb1SdBxsL5USb98Uylvl6ujcd9OiWgCVDD7Y7hP+sQww//awvPUFWT+1jU8+M22uhTUkh8Il/ifHf2923xSdAttLpZhzr7JSespkkMUg2K+fY5BZzcoJLJnAK4MkqpT4VKVPafVSn3Fz/wXtxtYikgNUM3U8YHuplDEmZs5efUuOKtUD57cWGEBE74Af1S3ic89Akw39s0+we8rlhLVSKeVzgzwHJTlsDdtLpVRp9ZhzY6xMmHiv8JeXalRRjMstpBDjeTnU1Ahg7CRmWVQpdXkjPAclOeyOLpBKNa6KOqx+ylooPMebt/L/6e9NXwhS2Bq2k0rpgssDYFxuQACEzdyA4P4Doopy9j+WqUEPHkjGrU/Aly8Jgw5BX7M6EXlADDaRRSQgPMdvadCb7oW9pFKGweUzES7+axJAqv/WB5IqCn3lrZ6LPM/zSX9xm6iiDBPwnfi0Z/ceEVUl/cVtNV2pGpBPBp14cH3JqnXG8be3yD+L+qdoF+wllTIKLo/n3bIASlXTNSBHlQpne08DD0UYJ+AbWUjLoqpw9kQ/BgCVVEoVc0o1YpPx1r3qYkxtrMtgK6kU5ODycuxrufBV2c9EqVkTVUqKkm2UgK9GVFUbVUoCE5gPbkpXZTwhi1jF//pGnTqURbQ1bCWVUoLLH1jIGEaVMkf9qFImV5+QRVThn9Rf1HUoi2h32EoqpQSXl3LlwSUV6kVV3mjSP0NeuvX7p9hitV/JoPtyIrxMjYELqTD39xf55QntZKR0fltbZjGnngx7nhcHtiUbq3xttMm9U7QVtpJKMS63UCrnCUvxcI+sicaiKjG9mP0KKZz8aF2QqIv8Urg0QzQbvdnFtEg2gwpifvTh5PO6yUgjDvw7JcU6BUVjsJdUyom+bDxeCMaCTtzblXPlGdV876YgBaTHZ4/lwnwuq+TXOyMFoScJ+AAgvxT/iL3PftGp6ia/xM1wXLwYjAVZ7P0yy7rUV5uEgf/te8pnSnJ0GewllWKiqzFw8VQ8peTKM66Jp54+NZANc1moC0tF1uWSXmtJW0R4o8mSnMwZg5KUQ5FKeaPJbxbDKSEVF4A+f3JVPeOmkRx161CSw96oNBHl65cm336/UqlU3n/70vWyuuT9t6UrDeL9tydJD7ov2oFMahoWlq9fmpSbla9f0nSpG8L01ozx5S9/Wfm8t7dnWMesHBfeQPbvlX+NNGl6uXr+FE2E/aVShmn1DAsBsMFpskVUaXwB+S2C5RGljHH51PNvyZ8pydFlsJ1UShlIdn9kXANCoppBD4PnGYjpvxGEe6pCeMoIeF1ufHQzwq3LUaUUdj+fy7IvTy1ySzNmQsRmO1yqMT8/v0i58i6FvaRSAPTuj87evvt9/qTk0eX/zusAE/iPQRbEzyvp70Of/0+8ALwnH67ffoXneZ6/2CfIDmOAWLrTJ6yvmwWVakFseiqV6lbYSypl7P7oGjg1kA1zHBfeHEtKKxATSMTcqTARQE29FSUs/ce7ngEyg6u/G0svjEh9hlO7j0YuBqu7SuujSqmhvAQDZRG7DraSShkm4EO5JNyTFh8hRdQbAIgCajodCZdm3mCqfY65t7ezAKonQqXPiDK6lQn4TG/ZpA5lEW0NW0mlDCGWilVPSTEdiaSditgiv5aCRt9kFFXqyGiaVOr/+p5ajkilUt0EW0mlNAMpJIfSgySgGigD0ivkeBaA2pBY/wMSMoD1nxWKEotYNXJzjy8rSQ7KInYxbCWVIjChHMS0Ova1mI7E9zxgg8nk2GZYOhM+DTFLZpAc++gDPJS6Ix7NxOMroo8p1QqSo/K1UWJalOroMthLKmVEORCtk5he3HT7WTzcI2Kockk4O/CAHTvHMOfGWNlzrPeRZFXMaWcPep9myMFsjCTgOwG3Ryc2bAHJQaC2K2pjXQN7SaWMYjx5o8nSd8PhMgBBEVCJpSLbOyDsASSgwCaRSj1g/QPVqFKVh6LqjTPwGMVtduRP0KKoUgrm5+f15UqJbUiOncTo8Np04VZkqDn9tQ47GxsYH7dm2vaPKgUAJ9hgMhFA1Y9Sah7VGHBtn8QLs21RpXTlBl5hlOSwHNLfhfFxa7q3v1RKCS0P1X6ycf1U+6JK1aDytVFlZ0ipjq6B7aVS4u1NQRCqqqg4VwwmEwGXW1j9Ppe6BwCnB6XmLrfwX+SaUqG49NcC7qva46eZqLfG4RLWsogKusSuyJqQmV6bmMsBAEKZyrXx6jV9qXH5TmJ0+M708vbcXA4IZQpn3hxem172zM2tAIBvWbUbbaB55dq4qpbUwWs3SElu2LEm9ddYVw3/GLaXSjGBGT/gj/E8n5zqB04F5wMM8JtfQ3hImo+Ud4lrmPjRz7E7GJQLB096AWb6m7KmiufPDwLsH+qty2IWcXFxUfncLZqp3NybeKdSqVQqhWXfysTsBiA9u55MpVKpVDIhuVRbXljennDIF4CVuTtXKpVKRXqmc3NzyJBqmBuWqjXYfGNWqVWpFJZ9ubm3NoYitwrLPviWC5WqdTU0k0Zhf6mUWCrKGvlnnsPgvc3bosSRnCHN90fP3i/eFgHmjXf42JmUts+qporjru+PLCQm0GqplIJuIhJDV6TlZei1aR+27+4AOzfWcr7ly+T5HL+mPKobb83lQhnpy1DkSggr68pzHZrSPM9yvaHIO8u+lTcTOw033xm+rDKO6qzUOMxMGkQXSKUUMIFErCQHkAJc3zbgSKq6KCkrn1xGiENJnthyqZRhOSm0DYuoge/MsK6scCcHzxVDsk5dffiMD2t3dzBu0I/q69ALHuTuFBpuPjQ0BGBj1jGxovQ2ffSZNAr7S6UMA0jViSplHCJKJ6pqAM2LKqXIo2qJRMoiNg/EtnzLhYq8GVxrwaj2l0rBG10YCSxwWQDw+NkHMCk0CyBVLQ9zKfiNWUSL/cHIhy4nEpUFQb+I5dZu7ESkPaWy0O3oqpEli1TbubuN0JVxYKOx5hvrK2pepHAnBxisYI3OpFF0gVQqv7SwNSqn1csKcpRsXaFZAKlySfB4FKLDyLqsl0odCwy9Nu3Lzb2lMBMOiUIYv7ysLn9zpc5xp8qXXJjLkWqNN8+t3diRB1c2iuqt5mFm0iDsL5UScx+i/1mn0ujBQ9Gk0DiAlFgqOns+keVSOlCSo3kYitwqLG9POBwOh2N4zqPQ3WbleviWlzFxtObj1zKh3NwwqbU2XciEJLsanwphZYJYe+MzaRSOSqXSQLWNWccEMpVr48oh0ehtwNZCYEH8Bv+XLyxxOR8f9UL88YXwes95/i9fzy9xq67kTKl2i2g2oJiOLJYGhKy0RRSKX1TXlgd6ndT8/qbkEabUhFGhJszoYDD5lwG16B4A5A2iSiplMKKMGpLjb/7mbxr4JetB7XkJYM/z4hN22Dj+6I/+SH2k7ETYU4rV0BlsY3ZiJZSpjEurZqZybTgxOjy7UWti7UrAd09wx/goSasn4ItmhZJrC7F12VyN3claKpWan5+XX4VRkqPb0IiBbayvIJQZJy8yyMESL3gwt75xzUzB1e4EfMaFAHQBpDQsC+NyCznJnawN6HKS41jiUCxi4U4Ovmnz1wHtSsBXLhETFVI/6yVp9aBYjpiOhG8K/u8wgMqcw1wKVcJQ7pMNXnTrJ2JhAj49qF2ZYihyqxI5uFqHoRGSY/gMeem9sb4C3/RrQ5AWNWOCpdUJ+MSlmfiHJIPewsjdbammRGgQL8xdjxQwu1wSTg/KfCGxrvybXCD+QPLCfO/qhx5NhqJWJeBTo1vUUhRAYyvYUOSd5bXhYQeAUCYyhI1Zx8T2cuGWoX1JLCIf9YJR+YNJUil3iuOukwJ2aurp9b/LY8TYxmSpVBZwjhBZk1dJwCfK6fCiXi8TXY0tcSRKdrWmNxrLcUTC24Oz3yBjiKXiqcqudhzvH5x9dOcDyUmsvx/DgIrkUB3GqiM28JMdCvOqFLI1RGIreQ4KK9DYFlG7Oo9fM2Ee25KAj6nJoKcIoKaDbBHz81gMl6ROpfhTgnqDKJb22G/W5u9rcQK+unole0qlKCTYSiplmGtP009VAKW4iaWrdYwIwzp9mndO0GyHS4J/qrlKWURbw1ZSKcMEfFBbaa+UIw/pxff6H5clNy8PqUbEwFKZ/7xf+LgMOF3s4+04lyoDgMcDI2uzOAGfGjVEIl1e7I4DSY6NWVnRsjHrqKLqJ1OLVifgQ34pnHJLqqgH233CP5Yh3r4plLeU8DXbce67abHKWPB8Mli8focM9G8Py3eHkzzP88kpcRtEaaWFxQn41P5gUBGJlOfoAtQ1sJ3EqGNie7lQfZ8c0nvL1aDlCfgyq9keiY1kAq8M3u/5/0+AOfcqK5GFySCLQbBfP8egXBLkBHrMuZf7icel+PhRz/3CbREAnukBejUJLluSgK/2xlV2VaPtsDV++71L7Z5CG1Bvi0jczyoG0pTxy8u+4TcTl8eNZCstTsD38piz9yvnFKkAAB2uSURBVN0FqdvB02A8gyZ9anLt9eDs5dcBMC996VGuSAbq7wc74kQbEvBRkqNbUcfAFAGHARQJssbAWiGVqnpHSksN4xre3x9WVFEp/D7U/mDMuTFWYVGYQII/Rw5sTPDbXrlDLHFxAcBzk/qoUqr31SWf9u+BBf5gBAbhpRrsipIcnYZDkBym7LwBOlgqJVlMfomLpKVYb22OKqV+DwYtz0Hfg9kddQysjn8cdu5uG4im2pKATwWtVMrlBgRA2MwNCO4/UDcX05Fw6RWJsSgJ7Ng8Ie5Xs4Ngjf4MtCiqlAKF56D7N7ujDsmh8Y/TYuOtuZwS1aQGLU/A95M/Dv9V/4IslSIBpJCJcPFfX+R5nl/ov/UBaU7kUxIlKDMWThcrbN5WRFWGb+va5HDp+NtbA9v/RLlEW6MeizgUuZUJrUw4RhNql+mdxKhjYkWJvVOL1keVeurpU8zWAicHkEIZAJ53Y/BXVzlNc2/0oiMlv4grDpDzlhxVimxU5cjb7YkqpUa3O18eBjuJUYcGqidyY1b9bScxavwGSf2OSRUtzuEw7bhJOOAMNn6tUrmcGB12zFXLfMuFyi3DxastUaUYl/sexpK85E+JsW9LzV/VCaAwscxLUdnUjIVaVEWYk9ZGlVpcXKw5hjXS0F4sIuHof/u9S5/74Y+O1IFJuM+NdYQ8qjAahpDchStKENHRhOy3ebgookdAAyTH4d0EWhpVyjC4fCMCqOoUdKKqxmAZi0hQj0u0F4uofgP2BDamx07ize0z71zBhboWpoTHAUDCHc69eWMn0hrXl6NLpTZmZ3FNa/1tiSolpiPhdz8h5cLNMFeK8VGvmQBKl2tP/PGF8Pq+5CAGIM7BkEZsLckB6nzZCHZurGH6naHxFzwTb21ETFeiodemfXMTo2fkVWv8mrSWtWKORwh6I21nq/EbtWh1VKnyloD9YUkqVQF+/pFoJoCSRVXqXHvPPFczckEXaYdGleoArEzoz1A31jD92hAwPhVSReDVYyhyq1K5cmfY4KCl6taSI9hhDEw+E06sSJIpwz8ZLY8qdaKnB/1fcAIA4zoFPP4UJgIoVZDtE3B7hM3bojqW28XTAHuR/8/ajW0LSA7DA1g3kRzqPeETnMFkSM/dzo21nBQmamIFdS0MIHQCiTk/vTasWJKq20qlYkFAnYZicqjDDR98LGxxAr5BD3Nqfz3MrQPA4GkMuGAigKoLMf1ff9UH//PSl1Ym4KtTXqeCvUiOz/3wR009fZH1S4rSC2Bj1rG+cW3cMJzFxqxjfUoVlj5yJTS3XgCOFAr7sKhnYKpELlJM1I1Zhz73SBWdEFWKSKWMBVCKfiowH9wMpzBS7TG/ljoRTPKBajwP41tTwxp/MAXdQ3JYgZ0ba54rVTJ7fCo0sb5x7bJR1fGp0MTEbNXENtZXfGcMa1qAOltESet7pLXTOz22uajOJy5JpXQQ0xFOj0j6hFGyPJOxoslgMc5xHLeImSCrdLvqIrE3Zkph6aWWUU2CfC5rGnq0PVATG5Tk0GHnxppHExNmfCq08maigNpj1ewGMH6tkkG19M0z1eCK2sr13LCOiDor2Pi1TMgx4SCbw4beF3SOVCpfEpz9j2V/Sw8elEQ4b1ffCBB5vKcM4t28lMjiPsLcpmkwVCtZxMXFxRqXMALicGnTFaaZGIrcqlXBDkVuXdOWSNxgxPiV0vi1SqWmQWvCVNUlOaRzYWHZJ1m6GXOoRjukUinHRa1Uyom+8lbPRZ7n+aS/SFhEJjDjl0PQJ/196PP/iRSAKvzrXjaY5PmYO6VZds1GbBGoVKoL0AiLOBS5JblZAtKiar6StkUqxUqqKCXXHlwDcmE423saeCgC8EblXHvh7Impt0iK9Hwu63mahJryRnklYAeVSlE0A4d50SwtszuJ0eG5CceKftvYrgR8vWNJPiHXLBoVyu4qqlx7bzDVgcbcqVSYSylKkM5IwFe/gr1YRBydoLc3jqDkkLeuGxtmy1hHJ+DT59oTUqUZOe/lYvpcw1FtmhibnrKI3YoniCo1rgtM3zEJ+IhRIc4paVP+/hcIMOrY2UsuRRHlHCnE5ZADYG+LAQMLo1IpiiOhOxPwaXPteQD2DydApFK9HrDBZDJYjBOpFPNvD8vCK4TQT/ofQNze1U2mTSRH5WujxKOZRtK2L+oYmIG7TAPvDDojAV812V7+zfhdnJ2ZADluTQ09YMfOMUxgxi9s3haB/Me7g/ufSsHYXL19j37n8UEjthDqqFLUxuyIOlvEociV0NzECg7rNNNiqRRcA6eKklSK9Z9FsToJ8h6sz5+8ouzsPv0XAc9oZ8t7fZK3CzwesF90tloq1Yg/2BMOcYTypsPxt7eO4V637hlMfju3MesgEqkDLK0TpFJyrj14ozzvW+JWXRIdT5iPFGLENVPNfBAjEkn4mwADoKVSKZiEjgJgRnVQksMuaIjkkA1tJzHamKW1P6oUAF2uvTo1kV/i4sW6SW3bgj3Pi8ou8Rj++e8CHIpFVLQlxNJ0VtY5UikAYvo78SwAyQWT1DBM1Sc7XSJlTtK3nEVUQO3K1jgUi6jQHhfwjpk/WGdIpZBfCqf6Btlgkq8Shoap+sT0H4fXe87LzGY7pFL1D2A0H5+t0YiB1dhVfXF9Z0ilxFKR9Z/cJbOQCUMmuhr7SjasCTUl3r65C+ySacWzEFJr+Y6QShFQFtHuqLtFlDwtfcuFSqURd5WOkkrB9W0+Wrvd06fqO/cqu4kxdypFkjRLp7COk0pZN0THxzZVuSUq8C0X3sGF2mL5zLKTGB2+c0W9w1JKDDqTfB0tQh0D20m8ScTzuTlN2DYFpkxHR0ul9Nn02i2VqkvldQmLSJbfIzH18sm/xmx2EjVPoDYeWx1YHqpNjbrvwQ7rLtMxUimTQkIWkgkoqfrUUimPBw9aL5Uy8wdDt7CINZ4BFt3I0GvTvjldQpJ2ozulUsaFmgR8D7YHhVKZSKW2viDX3MYnn+knQ6NK2QE7N9ZyvjMtCbRxCDTdwDpDKmVYqEnA5+7b7YMTEJ+q9Ek9Mede6kMtKMnRwdA4/A+vTTd2mLI8VJsaFiRB7wiplFGhLgFflAFE9A74Ud0iPvcMOjGqlHVDtIDksMwzQD5Nbcw6JraX32mQquiUM9jh0TlSKRP9lEECPlk/xctBqUbQ6qhSJv5gBF1CchAbs+oYOX6tsDw6PDz7ArGcoRc8+sRbbdo9WrCCAR0glWo4AR9TRz/VfnQHydECDEXeWV4bloOzjU+FJlTRtHcSF+YwXWgL+dFUA+soqZSuUG/Lg+8h8LomS+xNwf8dKpWyJSQTIyHox69VMrMOh0O6pn3VtSKHSpNg6ZbRKhax3VIpg0LG5ZaDShEvzP5zX4fiOSYl4PP4DGbUbhaRqqUA8tpIZQg1X6WSqshICZWtUR7JAZzUsPRAZh2L2F6plFGhN5oc2wyT4eN3RxbeeYOBEmoqnBLQg7Pf8AKtlkqZvQQjoOGlbA0LSI4OkUoZFZLNIOEIF5TA2eoEfNV68sVOkkqpa9qIRTzOsITk6GiplC6qVKcm4JOaGNa0HYuI43qYtIjkaJ1UyiSAlDea9M+Ql279/im2KF1TpFJKVCkxvXjTIezKUbbr3VrbSA4aXsq+sL1UyiSAlJhezH6FNJ/8aF3odclSqfN+OarUUh6AePNdYfdXqslLsXI0aKs/GIWtYX+plGEAKbx3U5D4RHwmh4liAgl++qkiO3aOYc6NsdlcHmDeeEe2zmTQiT74Z1qegK8+KMlha3SBVMoogJT41NOnBrJhLgtNcyU7unh7U/DP1Oz3PnuA+4Sm71SpFCjJYTfYXyoFowBS5ZJg7IWp6BFZTfRsxUnsB0Q/1QkJ+KRLNSX2JTmOJ5p/BgPQ2gR8AHQZ9JzmzaVD20wprDptGRa2CvXfg9FMfLZGV0il9AGkGJdbWP0+l7oHAKcHq82V1wKs/yzxB1M6FdOR+J4H0BSajNhaULuyL7pAKmUUQAq/+TWEh6T5SJl4fiG/xM2E/3GG8I292Q/6n3VWKRIildrblj3G1Gi3VApULWVbdIFUyiiAlPiwwvrPkOZVqZTv7P2+D+JEKvUgmHznDaaqnyLrWu/FhJSEpTOiShFQItG+6AqplGEAKbi+XcORiKU99ptBKYBUFUxgPri5KEmlnpeKWimVmp+fr0PlmaXeoyyiLXDMpFJGAaTaLpWqzyKePHlS73ZJWUS7wPZSKbVvZX8/Tg8S+rBa2KOOKsVOPbvKcXEAp1j03hYDAaQX3+t/XO5cqRSoWsrOsL1UCpBI9mSQ3d/Hg115oWQCiZgfwCNgT3C7GIBx9e6t/4tEcpwS8MlngHj7plDeKlcnz3235VIpii5Gt0ilttKLm+6z/YrHZP7POI6LF4OxoBPFbb/sRnkfD8kH5wjLTr7KgDn3Kit5YSaDLAbhv9ByqVT992CgJIed0R1Sqf+bW0gBEBSPSXh/EPNz8WwqLgCnL5LkKmLpAasEkOrzJ1cZSG/MSLDRzz7B7imXE50llaIkh63RDVIpsbTHBpOJANIR2WMS6qx6EjUIAELWHeMTXonkEBMBBt4oDynDpRKavrVSKdT1B6Mkh61h/6hSCgmIBjRObHCarHDKwgWS15LnGbIdXKomE+sYUJLDvrC9VEq8vSkIghxOFIhzUqLKahj6QYn9Y1xufHQzwq0LICyiC0C5JAw65PYeDx4Y/RmwlkVsxB+M2pVNYXuplOJwmQw6AQye5xMBphqGPhl04sF1SZXhPflw/fYrPM/z/MU+oaf/NAOc+LRn9x7Jypf0F7fVdKXJiO0FEUypQ2pTdDLsL5UiENOLm184rZQygYS80zuBgVPZXB4A8h/vegbIDK7+biy9MAJgZCEtZ+ULZ0/0Y4D01lFSKQWUTrQdukMqRc5h81j8oFRDRDCB+eBmuETcKEtFdmTMvb2dBaA6EWqy8vm86KyoUoe9RFnEjoL9pVJ1tE6N59rT12wATYxNb/YqTNfEOFT9gUNQFrFdsJtUSmWQhE2pmpf443cFrScXE0gkEQlvHZxrT84KcZPU7DB/MAWUTrQd7CWVyi8RSj0ZZMHei0fSImERU2GO48Lr+zVhofJLXDglYODf+ZVce8IrSeNce8QfbMAjlMq6yXQSyVH52mjla6N7nhfJV+on1uGwlVQqn8tWgz65v+oXSmVFcsgGY1P9gCfGSxz9jCSVYrH3yyzrknLt7f9LGZBy7fX/z4zscCmmFzfdftbA4bKjSI4aUM6j82ErqZQ3yle3aHu/zLIj02RAcqz68buqOSS/WQynhFRcIKqoAAPA9dL9U0VZKgX2my+TmqXvhsNlAAJwWnG4bKFUan5+vq55mDHyteVyJ7XlZLmjJEdbYE+pFHNujE1tjiW1BvrMc/B/Q7ZAVU6ikvySmqii4gIA9puqo9aJqtJK53BZc2tqNNEfzOxAVb+rGrOsfG2UkhydBjtKpfJLXLwYNOQi6kI0UkUdSmnVYaCcR+fDblIpMR0Jp9wXg7hK/JXV+OwTZCVqEUZ8Y7kksGPzjBI/6kFJhNNMaWVwa+1nEfWgdtXhsJVUiljXeX/2qn7Rg3jz7/ar3wz4RjhdrLB5O0Ooxb1tuF2MrLSK+UkziSOpgcUs4oH+YHVAWcQOh52kUuLtTQHIXidrYDbOyXImAGJ68R/6+5TWRnwjmMCMR5CX0Af+pLQASjxhrDrXjpVK1YCyiJ0PW0mllJfSKpZP+r64ORZTZdBT+EYmkODzS1zRRfjGH/A8apgPtRhK5kjsIJWy2hGToinoaqmUNEC8GEzqo7o9OVrocGlWbpUjJkWz0G4WsZ5UqrGh6tGAR+UbW4sj5wejLGLno/0sorH/MONyC0/mcLm1ECDMRErJZmnQHIAq7yX7m3q3RllEikPDViwiQUMOl0szC1vMRemCErFe3xyKb2Yy6MTe1SUj1qWTtIh6UCKxk2EnFhENO1yKax/eB4SrHMeRqPPCzffMmpdLghTVreqaaRMWEZRI7HjYikVs2OGSia6qJFn5JW7V9XXAkBIUS0XWNQ2yFLrkgVrKIi4uLtY5hlkdno2yiJbi2LOIUk7Zo6ADWETopb2URewo2M3hsjqQTDmIZsHlZeqif0olClYMui9HOE2ni328HedSZQDweGBibR1McqiJRLocdRrs5XAJoJZyMAkuL6a/G/8Q8Mf45OQvwjJ1kV8Kl2aIfqo3S1LcMv/2sHx3mHhhTonbwNOdHVVKD+KCWYdOJFGo6AmtLbCVw6UR5WAcXF5c++sy3DE+6gVzboyVo0rlskqU+jMx8nZMfPyo537htggAz/QAvc8fNGLTceT3YI2AsiDtha0cLg1j0xsFlxdv/vw+sB1XBsNqJup9uVRkXS5p4yhvEZmXvvQoVyQD9feDHen82PRH7srwKt1VWgp7Olxq3B+Ngsu/9CX8wiUrOPJL3KrrZQC1UaVUsenjAoDnJtsSm76BBHxPUl4vEBUlOaxGu6VSzYhNb+BG6XQZN7dnbPonAZVTtRftl0o9YWz6qhslxPRqdhBsSZSG+z6XugcA6IPfxwBwuVECiE1u9RLC0Li54a1ZxSIuLi4+iUvYgagTSoDCatheKiW5UYpS3LVd6XXbb34N4fFUkuf55FT/fXxUEgF4fe7U2k+Iw+XDPXbsHGPWXIvOZhEbAWERKcnRethfKsUEEjF3KsyRnaaUbC//T7/yn//SepjjuPAvJi/67ynRDmV/zQf/Xjlu6ZvbRyrVCCiR2EZ0hVTKOx1ki5ifx6IcFoo4XEZfj0rNWZcmwFs6Ei7Va25Ph8sDuzKsQFlES3HspVIHNK+HJsamt5JFRB0ikZ7NrMaxl0qJ6cXNl1+G1NzUIaxTpVKNgBKJbcRxl0qJtzcFYX1dXkOFq6S5FvYnOSjaheMulWICM374Y3Jzf4z/z9qNLSU5KJ4Ex14qJaoL5S1ia6VSzfUHO9CE1BX2PC9SksNSUKmUrnCCnAyNb02NzvAHqy3Xn7L0IewbmRJFU0ClUkaFHR2E6tCgJEcbceylUozL/XhDdrjEIFh7OVw2iD3Pi3SlaguOvVQK2CtvnSDxpxZG9rVpL41HbDYs9QerAQ1B1WIce6lUPrftP+8m8acWnv2Gf//TcrdJpRRQOrH1OPZSqdqaRdd0d0qlDhvLnqIpoFKpI9SUYBOHS3V5o7HsKZqFrpBKSbHptzREhMp0/TOMaXOoImfjbCxhYl529gdTQOnE1qMLpFKbgpAKcxzHLWw9wu51LpIWgfxS+N1PAH/soNDZYjoS//UgEVUF9+I2DJ1N0cmwv1SKxKZPBln27OkeOTZ95qdZ7H+JeP8zgRmSgM+UsfjVmVpRVb0R7QtKcrQe9pdKAfI28cJ25IMHcmsA2Tgn70fBvkimVtucZM0UVDVz+ajX2YVRpSjJ0RZ0gVRKdQqTS1UnO0nokZgwbi7VlA5sitSqpVIp6/3BKMnRNnSDVKoJCfgkO8svcZF0p+frOzooydF62E4qVZss75AJ+HQOlwrEdCRcekVKl254azaWSimgdtVi2EsqZZAsjwnM+EEcwpJT/cCp4Lx5Aj6D2PT5JS6SFvNEP7X3yyzrcurm0nUsIhVMtQz2kkoZJssrFWVV4TPPYfBeqWyagM8oNr03etGRkt/OFQfI/rC1UqnWvARToLarge3a7EcUzYWtpFIwTJan+Jgw58bYFFxOswR8Gt/KaoDRiWV+AtIWUdqIHg+p1IFDUzw5bCWVMk6WZ+RwqRlAFkCJtb6V556MzuhMh8sGyutFq6doLtrNItaTSukKDd0oDR0upWs6FrHbfSsbgZpL3PO82N7JdD0sMjCZRXSNyd/1D7Qq8IVBe5dbyEl8nlgqSiyiUWFecbgEvD4/JN09WRvdMT7hVfVJHC6tQc2OsWVtj4Avyx/+qJWjHktYZWCSiYVTAvwAAO90cDWsIszzS+EUgklTFtHpYrOr6WlvgBFvbwr+Ga9ZoeRwGQgwQD6XhTumsi6t9Xp97vhaPhD1StTIdPOWr5rtovrrgThs/dYMQdEUWGdgABOYD26GU8q3BO+SzkqA4XlJ2zgRK3FEKhVMSsuQUaGqTCnVC6D8MT7qJSouMgN/jDd3TKm7uFJQNA5HpVJp9xy6DXQFo1DQbgMzYxG7V69EcazQbgOjoOhqNF8qRUFBoYAaGAWFhaAGRkFhIayk6bscsuOMISOjDrkTiyFOMpX1Y39fruGX39IZ9vOTP+au7wLoGVlIL4zU65z0kn+Ti39QreFXvwLUEkn1x6VoNioUR0H5+qXJt9/Xfqri/bcnL10vk4r/R3By8tKNSqVSuRGZnPzT9xvo5//9/v8y+Z2U9pNx59cvTV66Xi5fvzT5p386+fb7hpOpvP+2vqz+/CmaBrpFPBqqjjMGoXLyuax/RloUmJcYoPd5AOLDR/392zVBdQz7+Uh8NPjvXweAEc7Ts/v//MS0cxLPp1wSzg48YF1Ow7g9Yqmod3KrO3+K5oEa2JGgfmgZlxtFTfwrb7S6QRNzRfQPOQGUS+X9fWTjHMdxsreZYT9b/7zfw7xECkd+rx/7/7xl0jnyuSzrOlEqsr0PSPC6cEpA9qcZ9VzLJTmu3QHjUlgAamBHQrmkezluCDH9/ez9/q++SjL94RQrO1+7ViNp0aSfj8TGvKeJJ8584HFJePzpv0g9x/xwPvdfVXEgxVJRunTQuBQWgJIcjaIa/9cf46dN8o/pWhSnpthfPANUnUQBVJ0FDB1wXmJ6ftZY54SeEF3sCdc8rxAVJ579giqyiMYhqO64FBaArmCNwhuVw4lEvWBcbkFO2CKWinpnUjEd4eKI8Yk3XqpX07Cfkd/rfyR+RAq3/nkf/b9XQyMqnROb0nVysvegmzlw/hRNAjWwo4E4zoggyn0l0geBxllGrplf4v7sXbmmWCqST4b9vMT07P5sYQvAFr8t8x2GncuTGczGL/xYlDp58amiakIkqo/cuO64FM0H1SIeGbr3SLKTi1MnYB4EdqF+D9bfjy9991Dvwcw798f4qFf9HowNJhMBVD1uVO/B1GPQ92CtADUwCgoLQbeIFBQWghoYBYWFoAZGQWEhqIFRUFgIamAUFBai+w1sJzHqcDhGEzva0o2NHZMGTzyWfrQj9SXPcCcx2pQOKdqCdsv5rUZh2YdQKASEMpoy33Kh6WNltMM8AayaIUWr0e0r2MZbc7nQ1OUzPqysb1g81s7dbfjODFs8CoW90G4LtxaZEOBbLlQKyz55cSks++R7l9cIVZGyAJE1JKOtbFRRVV91rbDsQ2hZKgtlaipAvTzpOi3oB1WqNzRVk1W0ZopKkdK5/GuZDWTSSW2x+oJZ+bFBdxtY9YlRWVjF4JFVP2/qp7zmcdN81T0t6m5Jg2oVze6xdhjpgqpObVf6m6g/Vf2TbPoDKKPWDG8wK7NOaja0Si2z8uOErjYws2fD8PmqaVLzNBh81Z2RdAamql8oGNY0O2sZGtghpqrrs/Z4qDPmzLLP+A9QA50UVG1r7+K4GVQtuvkMtrG+gtDUOPky9Nq02TlMfW4aPuPD9t0d/QX91wOhqj80NARgY9ZBMDyXI+WFOzl4Xhg6Spd1p1qLnbvbwMqEQ8HECuTGQ5F3ljE3MefJXBuvOyvTToYiV0Jy+azqFzYrP07oYgPbWF9RPw/Dczm0gOkwm8ysw+GY2NYfcFoH3Zp0KyLZUOFODg3+NiadjF8j3zKyQUn2ZFZ+fNC1BraTeHOlZn9SWDZew3JrN5S3TIddUhrExvqKb7mgPNLkiUbNKtQAjjrVoRc8mraauc1OrIQylUxoZUJ6/E1mVa8TCcSgMqHa39msvPvRrQa2c2MtV90fEii7xKEXPMjdKQAAxi8v+3Jzb5H/dmKV2lbNgvJo7iRGJ1ZUU1KNPir/iVfPUMGTTJW0vSC/r96Yld+GE/O6No7xa4qJmc3KrJONWc3SpLyuMCs/Vmj6qa4jUI9K8y0XKpkQAC1pBnWJIQNm/lVfqKsgjQgAvuWCjn6pGb2immGDNH39udW0lSuoiXkDllI3K8NOam6vofJjA+pwSUFhIbp1i0hB0RGgBkZBYSGogVFQWAhqYBQUFoIaGAWFhaAGRkFhIaiBUVBYCGpgFBQWghoYBYWF+B/Bbv8pHW3NRwAAAABJRU5ErkJggg==" /><!-- --></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;atom_fration_excess.pdf&quot;</span>, <span class="at">plot =</span> atom_f_excess_plot)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co">#&gt; Saving 3 x 3 in image</span></span></code></pre></div>
</div>
<div id="check-incorporator-list" class="section level2">
<h2>Check incorporator list</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#Get incorporator info</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>n_incorp <span class="ot">=</span> df_atomX_boot <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Incorporator <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  nrow </span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>n_incorp<span class="fl">.15</span>N <span class="ot">=</span> df_atomX_boot<span class="fl">.15</span>N <span class="sc">%&gt;%</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Incorporator <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>  nrow </span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="co">#Get incorporator list</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>incorporator_list <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">incorporators_taxonomy</span>(<span class="at">taxonomy =</span> taxonomy_tibble, <span class="at">bootstrapped_AFE_table =</span> df_atomX_boot)</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>incorporator_list<span class="fl">.15</span>N <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">incorporators_taxonomy</span>(<span class="at">taxonomy =</span> taxonomy_tibble, <span class="at">bootstrapped_AFE_table =</span> df_atomX_boot<span class="fl">.15</span>N)</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a><span class="co">#Print incorporator information</span></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&#39;Number of incorporators:&#39;</span>, n_incorp, <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a><span class="co">#&gt; Number of incorporators: 28</span></span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&#39;Number of incorporators if isotope were 15N:&#39;</span>, n_incorp<span class="fl">.15</span>N, <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a><span class="co">#&gt; Number of incorporators if isotope were 15N: 28</span></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a><span class="fu">print</span>(incorporator_list, <span class="at">n =</span> <span class="fu">nrow</span>(incorporator_list))</span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a><span class="co">#&gt; # A tibble: 28 × 2</span></span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a><span class="co">#&gt;    OTU         taxonomy                       </span></span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a><span class="co">#&gt;    &lt;fct&gt;       &lt;chr&gt;                          </span></span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a><span class="co">#&gt;  1 NC_000913.3 Terriglobusroseus              </span></span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a><span class="co">#&gt;  2 NC_003450.3 Segniliparusrotundus           </span></span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a><span class="co">#&gt;  3 NC_008261.1 Nocardiopsisdassonvillei       </span></span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a><span class="co">#&gt;  4 NC_010067.1 Olsenellauli                   </span></span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a><span class="co">#&gt;  5 NC_014008.1 Meiothermus_Bsilvanus          </span></span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a><span class="co">#&gt;  6 NC_014168.1 Meiothermus_Bsilvanus          </span></span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a><span class="co">#&gt;  7 NC_014210.1 Hungateiclostridiumthermocellum</span></span>
<span id="cb9-30"><a href="#cb9-30" tabindex="-1"></a><span class="co">#&gt;  8 NC_014211.1 Clostridium_Pperfringens       </span></span>
<span id="cb9-31"><a href="#cb9-31" tabindex="-1"></a><span class="co">#&gt;  9 NC_014212.1 Desulfosporosinusacidiphilus   </span></span>
<span id="cb9-32"><a href="#cb9-32" tabindex="-1"></a><span class="co">#&gt; 10 NC_014213.1 Desulfosporosinusacidiphilus   </span></span>
<span id="cb9-33"><a href="#cb9-33" tabindex="-1"></a><span class="co">#&gt; 11 NC_014214.1 Desulfosporosinusacidiphilus   </span></span>
<span id="cb9-34"><a href="#cb9-34" tabindex="-1"></a><span class="co">#&gt; 12 NC_014363.1 Desulfosporosinusmeridiei      </span></span>
<span id="cb9-35"><a href="#cb9-35" tabindex="-1"></a><span class="co">#&gt; 13 NC_015761.1 Streptococcuspyogenes          </span></span>
<span id="cb9-36"><a href="#cb9-36" tabindex="-1"></a><span class="co">#&gt; 14 NC_017033.1 Thermobacilluscomposti         </span></span>
<span id="cb9-37"><a href="#cb9-37" tabindex="-1"></a><span class="co">#&gt; 15 NC_018014.1 Hirschiabaltica                </span></span>
<span id="cb9-38"><a href="#cb9-38" tabindex="-1"></a><span class="co">#&gt; 16 NC_018066.1 Hirschiabaltica                </span></span>
<span id="cb9-39"><a href="#cb9-39" tabindex="-1"></a><span class="co">#&gt; 17 NC_018068.1 Salmonellaarizonae             </span></span>
<span id="cb9-40"><a href="#cb9-40" tabindex="-1"></a><span class="co">#&gt; 18 NC_019792.1 Pseudomonas_Astutzeri_AE       </span></span>
<span id="cb9-41"><a href="#cb9-41" tabindex="-1"></a><span class="co">#&gt; 19 NC_019897.1 Pseudomonas_Astutzeri_AE       </span></span>
<span id="cb9-42"><a href="#cb9-42" tabindex="-1"></a><span class="co">#&gt; 20 NC_019936.1 Frateuriaaurantia              </span></span>
<span id="cb9-43"><a href="#cb9-43" tabindex="-1"></a><span class="co">#&gt; 21 NC_019937.1 Sediminispirochaetasmaragdinae </span></span>
<span id="cb9-44"><a href="#cb9-44" tabindex="-1"></a><span class="co">#&gt; 22 NC_019938.1 Fervidobacteriumpennivorans    </span></span>
<span id="cb9-45"><a href="#cb9-45" tabindex="-1"></a><span class="co">#&gt; 23 NC_019939.1 Coraliomargaritaakajimensis    </span></span>
<span id="cb9-46"><a href="#cb9-46" tabindex="-1"></a><span class="co">#&gt; 24 NC_019964.1 Halovivaxruber                 </span></span>
<span id="cb9-47"><a href="#cb9-47" tabindex="-1"></a><span class="co">#&gt; 25 NC_019974.1 Natronobacteriumgregoryi       </span></span>
<span id="cb9-48"><a href="#cb9-48" tabindex="-1"></a><span class="co">#&gt; 26 NC_019975.1 Natronococcusoccultus          </span></span>
<span id="cb9-49"><a href="#cb9-49" tabindex="-1"></a><span class="co">#&gt; 27 NC_019976.1 Natronococcusoccultus          </span></span>
<span id="cb9-50"><a href="#cb9-50" tabindex="-1"></a><span class="co">#&gt; 28 NC_021184.1 Natronococcusoccultus</span></span>
<span id="cb9-51"><a href="#cb9-51" tabindex="-1"></a><span class="fu">print</span>(incorporator_list<span class="fl">.15</span>N, <span class="at">n =</span> <span class="fu">nrow</span>(incorporator_list<span class="fl">.15</span>N))</span>
<span id="cb9-52"><a href="#cb9-52" tabindex="-1"></a><span class="co">#&gt; # A tibble: 28 × 2</span></span>
<span id="cb9-53"><a href="#cb9-53" tabindex="-1"></a><span class="co">#&gt;    OTU         taxonomy                       </span></span>
<span id="cb9-54"><a href="#cb9-54" tabindex="-1"></a><span class="co">#&gt;    &lt;fct&gt;       &lt;chr&gt;                          </span></span>
<span id="cb9-55"><a href="#cb9-55" tabindex="-1"></a><span class="co">#&gt;  1 NC_000913.3 Terriglobusroseus              </span></span>
<span id="cb9-56"><a href="#cb9-56" tabindex="-1"></a><span class="co">#&gt;  2 NC_003450.3 Segniliparusrotundus           </span></span>
<span id="cb9-57"><a href="#cb9-57" tabindex="-1"></a><span class="co">#&gt;  3 NC_008261.1 Nocardiopsisdassonvillei       </span></span>
<span id="cb9-58"><a href="#cb9-58" tabindex="-1"></a><span class="co">#&gt;  4 NC_010067.1 Olsenellauli                   </span></span>
<span id="cb9-59"><a href="#cb9-59" tabindex="-1"></a><span class="co">#&gt;  5 NC_014008.1 Meiothermus_Bsilvanus          </span></span>
<span id="cb9-60"><a href="#cb9-60" tabindex="-1"></a><span class="co">#&gt;  6 NC_014168.1 Meiothermus_Bsilvanus          </span></span>
<span id="cb9-61"><a href="#cb9-61" tabindex="-1"></a><span class="co">#&gt;  7 NC_014210.1 Hungateiclostridiumthermocellum</span></span>
<span id="cb9-62"><a href="#cb9-62" tabindex="-1"></a><span class="co">#&gt;  8 NC_014211.1 Clostridium_Pperfringens       </span></span>
<span id="cb9-63"><a href="#cb9-63" tabindex="-1"></a><span class="co">#&gt;  9 NC_014212.1 Desulfosporosinusacidiphilus   </span></span>
<span id="cb9-64"><a href="#cb9-64" tabindex="-1"></a><span class="co">#&gt; 10 NC_014213.1 Desulfosporosinusacidiphilus   </span></span>
<span id="cb9-65"><a href="#cb9-65" tabindex="-1"></a><span class="co">#&gt; 11 NC_014214.1 Desulfosporosinusacidiphilus   </span></span>
<span id="cb9-66"><a href="#cb9-66" tabindex="-1"></a><span class="co">#&gt; 12 NC_014363.1 Desulfosporosinusmeridiei      </span></span>
<span id="cb9-67"><a href="#cb9-67" tabindex="-1"></a><span class="co">#&gt; 13 NC_015761.1 Streptococcuspyogenes          </span></span>
<span id="cb9-68"><a href="#cb9-68" tabindex="-1"></a><span class="co">#&gt; 14 NC_017033.1 Thermobacilluscomposti         </span></span>
<span id="cb9-69"><a href="#cb9-69" tabindex="-1"></a><span class="co">#&gt; 15 NC_018014.1 Hirschiabaltica                </span></span>
<span id="cb9-70"><a href="#cb9-70" tabindex="-1"></a><span class="co">#&gt; 16 NC_018066.1 Hirschiabaltica                </span></span>
<span id="cb9-71"><a href="#cb9-71" tabindex="-1"></a><span class="co">#&gt; 17 NC_018068.1 Salmonellaarizonae             </span></span>
<span id="cb9-72"><a href="#cb9-72" tabindex="-1"></a><span class="co">#&gt; 18 NC_019792.1 Pseudomonas_Astutzeri_AE       </span></span>
<span id="cb9-73"><a href="#cb9-73" tabindex="-1"></a><span class="co">#&gt; 19 NC_019897.1 Pseudomonas_Astutzeri_AE       </span></span>
<span id="cb9-74"><a href="#cb9-74" tabindex="-1"></a><span class="co">#&gt; 20 NC_019936.1 Frateuriaaurantia              </span></span>
<span id="cb9-75"><a href="#cb9-75" tabindex="-1"></a><span class="co">#&gt; 21 NC_019937.1 Sediminispirochaetasmaragdinae </span></span>
<span id="cb9-76"><a href="#cb9-76" tabindex="-1"></a><span class="co">#&gt; 22 NC_019938.1 Fervidobacteriumpennivorans    </span></span>
<span id="cb9-77"><a href="#cb9-77" tabindex="-1"></a><span class="co">#&gt; 23 NC_019939.1 Coraliomargaritaakajimensis    </span></span>
<span id="cb9-78"><a href="#cb9-78" tabindex="-1"></a><span class="co">#&gt; 24 NC_019964.1 Halovivaxruber                 </span></span>
<span id="cb9-79"><a href="#cb9-79" tabindex="-1"></a><span class="co">#&gt; 25 NC_019974.1 Natronobacteriumgregoryi       </span></span>
<span id="cb9-80"><a href="#cb9-80" tabindex="-1"></a><span class="co">#&gt; 26 NC_019975.1 Natronococcusoccultus          </span></span>
<span id="cb9-81"><a href="#cb9-81" tabindex="-1"></a><span class="co">#&gt; 27 NC_019976.1 Natronococcusoccultus          </span></span>
<span id="cb9-82"><a href="#cb9-82" tabindex="-1"></a><span class="co">#&gt; 28 NC_021184.1 Natronococcusoccultus</span></span></code></pre></div>
<p>It appears that not all incorporators were identified with the
bootstrapping method. This could be due to the model unable to detect
the incorporators. Another way to detect incorporators, as mentioned
above, is to test with the ΔBD method or with the HR-SIP or MW-HR-SIP
methods. For a more detailed discussion, please refer this [vignette][<a href="https://zielslab.github.io/SIPmg.github.io/identifying-incorporators.html" class="uri">https://zielslab.github.io/SIPmg.github.io/identifying-incorporators.html</a>]</p>
</div>
</div>
<div id="scaling-coverage-data" class="section level1">
<h1>Scaling coverage data</h1>
<p>In this example we will look at the two scaling approaches that can
be utilized in the SIPmg package. The user can decide either</p>
<ol style="list-style-type: decimal">
<li>A robust linear regression model</li>
<li>An ordinary least squares linear regression model We will also
briefly discuss data filtering methods when there are heavily
influencing outliers, as a result of upstream methods.</li>
</ol>
<div id="basic-approach-for-scaling" class="section level2">
<h2>Basic approach for scaling</h2>
<p>Sequencing effort on each fraction and subsequently obtained coverage
is typically used to estimate relative abundance for a feature of
interest. However, as we discussed in the introductory vignette, this
data is compositional and a quantitative estimate of concentrations can
be highly informative. For this purpose, synthetic spike-in standards
(sequins) with known concentrations are added before recovery of
fractionated DNA, from the CsCl medium into TE buffer or water. For
details on sequins please refer to <a href="https://www.nature.com/articles/s41467-018-05555-0">Hardwick et
al. (2018)</a>. These reference standards in each fraction, can now be
used to estimate the concentration of the feature of interest in each
fraction.</p>
<p>The approach used for scaling is detailed in the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>For each fraction, sequin coverages and known concentrations are
obtained to make a standard curve.</p></li>
<li><p>Limit of detection of sequins, i.e., the lowest concentration
where at least one sequin has detectable coverage, is determined for
each group of sequin concentrations.</p></li>
<li><p>For each group of sequin concentrations, mean, standard
deviation, and the coefficient of variation of coverage are determined.
Groups of sequins with coefficient of variation greater than the set
threshold will be flagged.</p></li>
<li><p>Sequin groups with coverage values above the limit of detection
and below the coefficient of variation threshold are filtered in
preparation for regression.</p></li>
<li><p>Linear regression or robust linear regression, based on the user
choice, is performed on sequin coverage values as the independent
variable and concentration of the sequin group as the dependent
variable. The user can also decide on performing log scaling of coverage
and concentration values for regression.</p></li>
<li><p>The regression parameters are extracted and plots with regression
parameters and best fit line are saved for inspection.</p></li>
<li><p>Coverage values above the limit of detection and below the
coefficient of variation threshold are filtered for further analysis.
The rest of the values are flagged and reported.</p></li>
<li><p>The filtered values are scaled based on regression parameters to
estimate absolute concentrations in each library.</p></li>
<li><p>Absolute concentrations of biological features are saved as a
fresh dataset for the subsequent isotope incorporator identification and
AFE estimation pipeline.</p></li>
</ol>
</div>
<div id="choice-of-regression-model" class="section level2">
<h2>Choice of regression model</h2>
<p>Ordinary least squares regression can be very sensitive to outliers,
resulting a poor model performance when the data is contaminated with
outliers. Although there are methods, like Cook’s distance based
filtering to address outliers, it may not always be the best idea to
remove these data points.</p>
<p>See <a href="https://statisticsbyjim.com/basics/remove-outliers/">these</a> <a href="https://stats.stackexchange.com/questions/175/how-should-outliers-be-dealt-with-in-linear-regression-analysis">discussions</a>
and more for better insights into handling outliers.</p>
<p>Robust linear regression is an alternative to this situation. Robust
linear regression assigns appropriate weights to the data points,
minimizing the influence of outliers on the model performance, without
deleting data. In this pipeline, a Huber loss function from the
[<code>MASS</code> R package][<a href="https://CRAN.R-project.org/package=MASS" class="uri">https://CRAN.R-project.org/package=MASS</a>] was used for
robust linear regression.</p>
<p>The user is free to choose between the two methods for
regression.</p>
</div>
<div id="scaling-the-data" class="section level2">
<h2>Scaling the data</h2>
<p>For illustrating the differences in the regression models, a
different dataset, that was impacted by one or many upstream methods,
will be used than the one in the introductory vignette.</p>
<p>This dataset can be found here</p>
<pre><code>#&gt; Rows: 86 Columns: 6
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;,&quot;
#&gt; chr (1): Feature
#&gt; dbl (5): 52434.2.332926.AAAGGCGT-TCAACTCC, 52434.2.332926.GCAGTGAA-ACATTCGG,...
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#&gt; Rows: 80 Columns: 2
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;,&quot;
#&gt; chr (1): Feature
#&gt; dbl (1): Concentration
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#&gt; Rows: 5 Columns: 2
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;,&quot;
#&gt; chr (1): Sample
#&gt; dbl (1): Dilution
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<p>For both scaling functions the following data and parameters are
required</p>
<ol style="list-style-type: decimal">
<li><p>Coverage data. The output of <a href="https://github.com/Ecogenomics/CheckM/wiki/Utility-Commands#coverage">checkm
coverage</a> command can be directly used.</p></li>
<li><p>Sequin dilutions and metadata.</p></li>
<li><p>Whether or not log10 scaling to be performed (a BOOLEAN value of
TRUE or FALSE).</p></li>
<li><p>Coefficient of variation. In this dataset, a higher coefficient
of variation compared to the value used in the [introduction
vignette][<a href="https://zielslab.github.io/SIPmg.github.io/quick-example.html" class="uri">https://zielslab.github.io/SIPmg.github.io/quick-example.html</a>]
is used to allow data scaling to occur.</p></li>
</ol>
<p>If the robust linear regression is chosen, the function is
<code>scale_features_rlm</code> which is performed the function file
<code>sequin_scaling_rlm.R</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>mag_tab_scaled_rlm <span class="ot">&lt;-</span> SIPmg<span class="sc">::</span><span class="fu">scale_features_rlm</span>(f_tibble, sequins, seq_dil, log_scale, <span class="at">coe_of_variation =</span> coe_of_variation, <span class="at">save_plots =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="co">#&gt; Warning: There was 1 warning in `dplyr::mutate()`.</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="co">#&gt; ℹ In argument: `fit = ifelse(...)`.</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#&gt; Caused by warning in `rlm.default()`:</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt; ! &#39;rlm&#39; failed to converge in 20 steps</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt; Warning in dir.create(plot_dir):</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#&gt; &#39;C:\Users\psampara\AppData\Local\Temp\RtmpKKb1hw&#39; already exists</span></span></code></pre></div>
<p>If a linear regression is chosen, the function is
<code>scale_features_lm</code> which is performed by the function file
<code>sequin_scaling_lm.R</code>.</p>
<p>If the user chooses linear regression, they can choose to perform
outlier filtering based on the traditional Cook’s distance threshold of
4/n (n being the sample size). Only the outliers in the sequin coverage
data are flagged as outliers with the Cook’s distance threshold and are
filtered out. Later, the remaining data is subjected to linear
regression to obtain model parameters and to scale coverage values of
feature of interest.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>mag_tab_scaled_lm <span class="ot">&lt;-</span> SIPmg<span class="sc">::</span><span class="fu">scale_features_lm</span>(f_tibble, sequins, seq_dil, log_scale, <span class="at">coe_of_variation =</span> coe_of_variation, <span class="at">cook_filtering =</span> <span class="cn">TRUE</span>, <span class="at">save_plots =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="co">#&gt; Warning: There were 2 warnings in `dplyr::mutate()`.</span></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="co">#&gt; The first warning was:</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="co">#&gt; ℹ In argument: `slope = purrr::map_dbl(fit, ~summary(.)$coef[2])`.</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="co">#&gt; Caused by warning in `summary.lm()`:</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="co">#&gt; ! essentially perfect fit: summary may be unreliable</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="co">#&gt; ℹ Run `dplyr::last_dplyr_warnings()` to see the 1 remaining warning.</span></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="co">#&gt; Warning: There were 2 warnings in `dplyr::mutate()`.</span></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="co">#&gt; The first warning was:</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="co">#&gt; ℹ In argument: `slope_filtered = purrr::map_dbl(fit_filtered_lm,</span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="co">#&gt;   ~summary(.)$coef[2])`.</span></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="co">#&gt; Caused by warning in `summary.lm()`:</span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a><span class="co">#&gt; ! essentially perfect fit: summary may be unreliable</span></span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a><span class="co">#&gt; ℹ Run `dplyr::last_dplyr_warnings()` to see the 1 remaining warning.</span></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a><span class="co">#&gt; Warning in dir.create(plot_dir):</span></span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a><span class="co">#&gt; &#39;C:\Users\psampara\AppData\Local\Temp\RtmpKKb1hw&#39; already exists</span></span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a><span class="co">#&gt; Warning in file.remove(plot_dir): cannot remove file</span></span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a><span class="co">#&gt; &#39;C:\Users\psampara\AppData\Local\Temp\RtmpKKb1hw&#39;, reason &#39;Permission denied&#39;</span></span></code></pre></div>
</div>
<div id="comparison-of-the-fits" class="section level2">
<h2>Comparison of the fits</h2>
<p>Robust linear regression provides a better model fit compared to the
linear regression. Additionally, robust linear regression has less
variability compared to the linear regression model.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co"># Robust linear regression plot</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>EBImage<span class="sc">::</span><span class="fu">display</span>(rlm_example)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAIAAACb4TnXAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAgAElEQVR4nO2de1xU1fr/nz0Mw30AEUTQVAQR08q8HF6WevLS0aNQeUzq9DUtTY+ppXbT6kcny3OsLNOTmprXbpoeMslKUutrphje7RsiogSCysAAw8zAMJf1+2PhbjszDHNhrz3MPO+XL19rrVlrP2tv5jNr7XV5FkcIAQRBxEEmdQUQxJdBgSGIiKDAEEREUGAIIiIoMEesWrWqQ1wT8Vo4PxxF5DiODx8/fnzo0KEOcrr3fE6fPj1ixIiGhgYH13T74t6PD9+aq8ilroA08H9+Cb8KPvwV9OFbcxXsIrawa9cujuNSUlI4jtu4cSOfzt1kzpw5AEAIETaAfJjjuOjoaJoTAO6++26tVstH7SIsK5PJaObU1FSaOHfuXI7jxo4dy3Hc0aNHAUCj0XAC+LKnT5+2a4jjuLS0NI7jgoODAUAmkymVSppCM6hUKrsXXLt2LZ8yZMgQjuPuu+8+juMaGxsBYOPGjRzH3XbbbbR6ABATE8NxXOfOnTmOu3HjhvDW3nzzTY7jkpOTOY774YcfrB4px3HLli1z+k/UMSH+B3/XAwcOjI+Pt0oUhm0TLRaLbeL48eNXr14tNHHq1Knw8HDH1h1YaS3RbraVK1c6NkQIycrK+vHHH2n4yJEjEydOdGD39OnTfHpoaKhVHgeVcfIuHN+aj+Hjt2cX/sdl0aJFNMVWNiaTiTgtML1eDwAhISH8l9hDgRUUFNj9HQwKCrJKcfAFtbqsELlcbteuVeLkyZNtqxEVFQUAXbp0sbr46NGjra5WX1+PAvPTLiK9+ffee6+0tBRuHfagOOja2RISEkIIKSwsXLx4sUsFWyMiIiItLU34d6JVOnv2bMufTebyH85sNvNXMxqNzhQJCwubPn26VTVqa2sJIVu2bOE4Ljw8HG4+zKysLI7jDh06xBdXKBSuVtIHEU263gs411gJAyaTyTbx3LlzNPzJJ59Ylf39998VCoVj67aB1hL1er1VCh2bIU63YGvWrOGLEELy8vJohqtXrxJCysvL7dq1itLM1dXVNGo2m60yb9myhQrS+Vvz+W+gj9+eXYR/VJ1OR6M0QGloaOBz8vAtQGBgIE356KOPaNk//elPfLY33njDqmxr1h1/4WjTKrzIpUuX+KhMJrMt6+A2CSGdOnXii0+ePFlYySeffLI1gX333XdW1RBG+d8mq5vlA2q1mv+oqanJwf36KjhfgQDgzJVo+Ok7GAIA48aN44fLP/nkE6mr45t47+/WhAkTDAaD1LVAfIcDBw6wN+q9Kznq6up+/vnnNrNpNBqlUum2lerq6s6dO7tdXK1WC99tGFuXtrherw8NDXW7uIePzmKxuDSUmpeX57YtT8AuIoKICAoMQUQEBYYgIoICQxARQYEhiIigwBBERFBgCCIiKDAEEREUGIKIiPeu5AAA4Qr31jCZTM5k89BKa5Bbl+Ezti5tcaPR6Mk6Ow8fHbnVd0ObNDU1uW3LE7xaYGFhYW3mMZvNzmRrjcbGRk+KGwwGCa1LW9zDpVIePjpXl0pRxyTswS4igogICgxBRAQFhiAiggJDEBFBgSGIiKDAEEREUGAIIiIoMAQRERQYgogICgxBRAQFhiAiggJDEBFBgSGIiKDAEEREUGAIIiIoMAQRERQYgogICgxBRAQFhiAiggJDEBFBgSGIiKDAEEREUGAIIiIoMAQRERQYgogICgxBRAQFhiAiggJDEBFBgSGIiKDAEEREUGAIIiIoMAQRERQYgogICgxBRAQFhiAiggJDEBFBgSGIiKDA7MBxnDDa2NjIcZxVohWEkNby8Ikcx8nl8jFjxrRjVREvx1sE9vvvv0tdhRZkMutnEhoaSgghhIwcOdJBKZonLCxMmJ6cnEwDn332mclkMplMKpWq3euMeC1eIbDVq1f36NHjgw8+ENXKn//8ZwAYNmwYjR45ciQjI+Oxxx7LyMior6/ns1ksFquChJCGhobExMSDBw+2aUWv1/Ph6urq77//nob//ve/BwQEAMC5c+eE+YcPH15YWJibm+vGHSHej1zqCgAAPPPMM9euXcvLy5s3bx4A7Nq1CwDMZrPBYGizrMVicSYbAOTn5xsMhuDgYJr/7rvv/uSTTxoaGiIiIoKCgqwuYhVVKBSXL18ODw+nUjSZTDTdaDTyOW0DPXv2rKmpEaYEBwc3NTUJL378+PF77723srLSybuwwr1S7VLcZDJ5UpwQ4mFxx512K4xGo9u2PMErBAYAXbt23b17t8lkksvlGRkZAPD+++8rFIo2CzY1NTmTDQAMBkN4eHhzczONGo3Gpqampqam0NBQhUJh9dcSXnPAgAHnz5+nV6Dp/Kd8Cp/IcRwNGAyGfv36DR8+HACGDx9+/PhxmUxm2zyazebw8HAnb8EWtwt6XtxkMklo3WKx2HbmHSCXS/NVZ2314MGDv//++7hx4xISEvjE9957b9GiRYcPH6YDAMHBwTTdmZ+oNocfeAICAnJycvjMCoWiU6dOFoulU6dOdi9L/yeEPPzww6NHjz59+vSxY8dsbdGUmpqaLl26cBxHJUQL/vLLLzT8yy+/jBgxAm6+4BFCaNnY2FgHoyPO4HZBz4t7Um32xT18UO5DWAEAn332GR+VyWTZ2dkO8g8bNsyZy9bX1ztfAdtElUrlZHG71NTUeFLcQ+vSFtfpdJ4U9/DRmc1ml/Lv37/fE3Nuw6gFKysrIzd/tilms5mNaQptUlhaRBBg1kVsbGx84IEHbNO/+uorNhVAdSGSwEhgtbW1tlriR8wRxFdhJLD09HRhtLGx0Wg0Hj16lI11BJEKaSaahdNHCOLDeMVKDgTxVRgJLD8/nxMQGRnJxi6CSAu7dzAcx0P8EKZdRGEjhovKEX+A6VIpvhGrqamx2taBID6JNIMcnTp1stq1gSA+CdMWjF9wGRMTU11dzdI0gkgCI4GNGTNm1KhROM6B+BuMBHbgwAEaoFvrzWazS5t5EKSDwvpbbrFYCCEymQxHERF/gKnAQkNDm5ubOY4bNGiQ3W2OCOJjMBVYVFSUQqHIzMw8efLk/v37WZpGEElgOop47dq16Ojompqafv364TA94g9IM9H822+/sbSLIFLBtIu4c+dOjuNyc3Ml80CCIGxhKrBZs2aRm85ncLcl4g8wFZhGo6GBgoKCfv36sTSNIJLAVGCEkICAgMzMzIKCgqioKJamEUQSGA1y6HQ6OrCRn59PU6gTXzbWEUQqGH3Fw8LChgwZwsYWgngPrJdK0d2WKDbET2C9XYWOImo0mqqqqri4OJbWEYQ9TFswfgW9UqlUq9UsTSOIJDBtwQ4dOsRxXEJCQmVlJe4NQ/wBpgIbOXIk6grxK9CrFIKICNMWTKFQoMdsxK9g2oJduHCB47iIiIiIiAh0eoP4A0xbsKSkJHwHQ/wKpi0Yf/gygvgJTFuwtLS0u+++m4YPHjwYHR3N0jqCsIepwE6dOsWHtVotS9MIIgmsh+m//fbb3Nzc4ODg8PBwlqYRRBKYCqxLly7jx48/c+ZMU1NTcXExS9MIIglMu4gqlaqioiI7O1upVGZlZbWZ35khR0KIhyOTWNztsh2ouFTD10wFZjabAYAQYrFYnHGd7YxvHLouxJNaYXG3y3ag4lL5WWLURSSE7N69m+O43bt37969+/XXX6+oqGBjGkEkhFELxnHc5MmTAUD4P4L4PNIccWKxWMrKyiQxjSAsYfcOtnz5cv7/2NjYGTNmMDONIFLBTmCLFy9evHgxH21sbAwJCWFmHUEkgfVE89atWzds2KBQKFBdiD/AeqJ5+vTpNTU1zc3NpaWlLE0jiCQwFZhKpVKpVC+//PLnn38eFBTE0jSCSIJXTzQjSEeH0bec95jdYlUmA4Bhw4axsY4gUsGoBbvjjjvOnz9vlbh+/Xo21hFEKhgJLDQ0dMCAAS+++CKfMmDAgKlTp7KxjiBSwfQdbN26dRqNhuO4iRMnvv3227wnbQTxVSTwiwgAX3/9dXV19UsvvcTSOoKwh6nApk6dynHcqFGjOI7r3Lnz22+/zdI6grCHqcDWrFlDCDl06BDtGVosFpbWEYQ9TAU2Z86c9PT0xYsXS7X7DUEYw1Rgn376aX5+/vLlywkhP//8M0vTCCIJTAXW0NBAAz/++OOAAQNYmkYQSWAqMELIwIEDg4KCamtrlUolS9MIIgmM5sHq6+tPnDgBACtWrKApzc3NCoWCjXUEkQpGAouMjBw9ejQbWwjiPTDqIk6bNu3KlStWiUOGDGFjHUGkglELtm3bNovFIhydX7JkSUFBARvrCCIV7NYiymQyXHmIiEpdXV1UVJTUtbgFpot9EUQ8CgsLAQAFhiDtTElJSXNzs9S1sA/TebATJ05wN1GpVCxNIz6JwWAoLCz0WnUB4xZsyJAh+BqGtBe0T+jlYBcR6XhUVlbW1dV1iCXjrF07YRcR8QRCSGFhYX19ve1HG/MSpr6fxr5KjmHaghFCampq6uvrk5KSWNpFfIPW+oT5F5VrvkkEgMAAAuBdzRpTgUVGRh4+fDg5OTkqKqquro6laaRDU1tbe/36ddv0MlXwK5/2ouGPF1D5eVcjxlRgQUFBd955JwDU1dV9991348aNY2kd6aDYbbgaGgPmbkil4Y/mFgUFeunueKYCq6mpoYeqpKSkFBUVsTSNdEQuXLhgO+xsITBtVUsztXb2xYgQM/N6uQBr19nHjh07ffr0xYsXO8QQECIVWq22vLzcNn366r5mCwcAz04sH5ysZV4vl2EksPz8/E6dOqWmtrTpc+fOraqqio2NZWMd6VjY7RM+8Z++JjMHAH8ZqP6fkTcI8brxDLswElh6ejoA4Cwz4hi7fcL5G1PqdHIA6BLVvGJ6SWtlg4ODe/XqJW79XIeRwAghjzzyyBdffDFlyhQAUKvV27dv79q1K/1UrVZ/8MEHZ86cycnJYVMfxNtoaGi4evWqVeKG/Qk/FUYCgDLUtGZWcWtlExMTvdYDBSOBcRy3c+fOffv27dy50/ZThUKRnZ3NpiaISwhfladOnbp9+/bWco4cOfLw4cN8VC6XG41GYfGkpKTLly/bNXHvvffSMH8eyLEi5dpvEwEgQEa2PnPhpZdemj27TiaTrVu3DgBmz57dku3YMa1WGxwcDACEEGrOqzpKTAc5VCpVZGQkABBCSkpK+Hew8PBwAMjIyMjNzQWAoUOH0t2Z1dXVzlzWyWxY3MPiH3/88XvvvSdM0ev1fFioLgAwmUxWF7errmHDhgUEBPBRrVZbrw9c/NldNLp2RgHHgVYLdNbUYrFotS0DGz/99BMNRERE0MCMGTNooLS0lH6jhPAezVhDGELNXbx4cffu3VYfzZw50ypl2LBhzlyzvr7ekyqpVCpPitfU1EhonUHx559/HgBox16n0wk/sooWFxcDQJcuXej3avv27YSQ/v37A8Bzzz0HAEeOHBGeRpCSkjJ8+PAFCxYMv8nY8X8bNIvQf6fOXPhNwL59+4YPH75u3bri4mJCSFZWFtzc+kUICQgI4A8Skcvldm9k//797j4nj2B6vklYWJhOp/vb3/723//+Ny8v7/7776fpjzzyCA3s2LGDz3zPPfc445xUo9F40v+urq7u3Lmz28XVanWnTp2ksi5tcb1eHxoa6kZBi8VSVFSk0+nCwsJoisnMPfGfvjT8n6eKo8JMtqWUSmViYqLwIi4dkir8vrGEaRdxxYoVKpXqyy+/lMlkwj08Ql0hvo3tOCE/tfXcA+V39bIztdWjRw/3lOwNMBXYc889N2fOHDzzwT+5du2a1QLUp9f3aWgMAIBxA9WPjbxhW6R3794d3XkmU4EFBgbi8ed+iMFgsBrhWLg5uVoTCACJMYblU+0MfqSledeaXbdhKrDp06cvXLiQhpctW2Y71IP4HlbLMlZ/3a3gUgQAKOSWTfPsrEf1GWlRmAps9erVfP+7vLwcBebbXLx40Wz+YyXuT79FbshLgJtTWzqdDiCM/zQoKMgndwmyE1hMTAz/f0BAQFVVFTPTCGOuXr0qnHe6Uad4fmtvGr65a+sPQkJCevbsyaxujGEnsJqaGma2EKnQ6XRlZWV8lBDu8VUt4+/r/nExPPiWrSWxsbGezBN0CJh2ETmOGzRoUEhICADk5uZ6m49IxBPo7NYfUcGurWX/c+W2zk3CzHQ+2pMpxI4CU4GFhobSQ4wQH8NqJIN3PvPCQ+V39LhlaispKSkoKAgA1Go1s+pJCFOB7dix46233ho6dCgA3HvvvYGBgSytI2JgNZIxb0NKvV4OABMH12Tde8trdq9eveiqXL+CqcAyMjIyMjJYWkTEo7y8XLhmYOkXPYsrQwAgKb7x9UdKhTl9YL7YbVg7HuU4bu/evZMmTTIajYxNI+1FRUWFRqPhzyhd821iflHLclCrQcK+ffv6uW8I1oMchJDc3Fy9Xl9RUSFcu4l0CNRq9Y0bf6xp+uF89OaD8TRsJS1/brWEMBUYv0hKLpc3NTU5zox4FY2NjaWlpXy0oiZo8cctIxlW0kpOTsa3ax6mAisrK+M4LiAgwGw2s9wmg3iC2Wy+ePHiH1ELN311y9TW5vkXAgP++DumpqbiQlMrmAosMTGxsbExODg4Pz+fpV3EPZqbm0tKbnEyw4+/v/nYla6RmsCAlk5g9+7dceGbXSR4BwMA6jobJ5q9FpPJRHco8/DSemL09VEDagGAbuiLi4ujy98QuzAVGP/W27dv33PnzqHAvBCrBRkAMGttamOzDADG3FE7bdQfDuKVSqUXuknzNpgKbNeuXRzHRUdH19bW4juYt2HbIVy2u8eFq6EAkJqof/Xh3/l0ujxX6PEGaQ2mAsvMzERdeSG2HcJP/jd+/+loAAhRWDY8/UeDJpPJePfMiDMwFZhSqeR3MaDrbG+gqanpypUrwhR+1xbYjL/72FZINjAdVDWZTLw7K7fVtWfPHnx5c8yLL77Yp08fx3lWrlw5a9YsoboWL9089f00qq7tzxZ+vKDQaDTOnTtXr9c///zz8+fP96RK165dCwoKMpvN2dnZvXv35tOjoqJ27drlyZW9HNYuA7RaLZ2FVCgU7i2ieeihh0AwIInY8s477wDAgw8+uGfPHttP6WqML7/8kuZ84YUXWrymKd8BgNDCEQDAcesBYN68eWfOnBk8eDAteOnSpeTkZPeqlJCQAAByecv3TaFQXL9+nX4BpkyZ4sN/SqYCW7BgQWVlJQ337t1b6NLVVcaOHdtOlfJZFixYYJVSXl7Oe8al/PWvf+XH34MvT5cZLsNNX7nR0dH5+fnCJaOe7zumawwA4NixYwCQmZm5d+9eD6/p5TAVWJ8+fWQyGSHk1VdffeONN9y7iA//2rUXto/oypUrVmvT1q9fP21V3+XfcwDw+H03xt6pBmhxuxsQEEB7mELfle1eJbVa/dVXX7XX9b0WaSaay8vLa2tro6OjWVr3T+yetTVtVV8L4QDgzp7a5x+85Zw7HMloX1if0UwD3bt3P3ny5KBBg1ha9zfsnrX1zx09S66HAECPuKbFmeeF65v4vcZIO8JUYO+88w7HcWPHjv3++++xpycStpNalJ1H4r4+EQOCs7b417HIyEg6CIG0O0wFNn/+fA9HexEH2DrQpZwtDV+xpzsN23pNwz6hqLATGP8C9vDDD+fk5AgdOSAecuPGDbs+ZCrVipe2t0w6bXu2UHbrtEjPnj1xrl9sGAmMHghGw7t27aqqqrp+/Xp8fDwb6z5MSUmJ3bG+pmbZU2tb1jRtfLooWHHLgRsJCQmRkZEeHt6HOAMjgX3zzTeTJ0/mo3FxcUePHkWBeUJRUZFGo7G7C+vxVWn012zF9JIuUbfIDxcTMoaRwDZt2pSens7vs4yNjVWpVGxM+xi220mE8NJ6auy1EbfXWX2Kr1vsYXcI+r59+/i1UfgC5gZWXjGsmPFBarNJBgCDkxuenXjV6tOePXtSh8oIY5ge/oBD8+5B3aS19ulrO3pdvh4MAF2jm9+eVmL1Kb8sA5EE1n4REZe4dOmSAweSX/7SPe9cPACEBFk2zPH9s7Y6IpIJzGw2e7LY17chhFy4cMFBhuJrIUt39gQAmYxse8ZOTlyW4SVIILCYmBi1Wm0wGFBgthiNxkuXLjnIUNMQuGBTy56R/zx5IkoZZpVBLpenpKSIVT/ERdgJ7LPPPnvsscdmzpwZHx+PZ4XZolKpHE9M6Q0Bs9e1vE19NLcoKNCi1Vq/02Kf0NtgJLD8/Py1a9daLBaO426//XYnSzk5E+rhhKnkxUtLSx0P/xCApz8aQsOvTzkfp2wyGsBoAADg93fFx8eHhIS4WhkPK++h3xuWM93CEzdZwkhg6enpR44cWbp06Wuvvea8S3pnjj/UaDRKpdLtilVXV3tyyKJarfbkFLkTJ06EhYWFhVl384TMXJNqMMoAYOaYayP71wHIAVoml7VabXh4uNtzxx7eu16vDw0Ndbu4h4/OYrG45EWY7iJlD9N3sOzs7OzsbABISEg4e/as3y6E42e0HDdcT6/v09AYAADD+tbPGVdpNw8eX+LlSDOKyDsO8DdaW5VrxaLNKSqNHFoffweAmJiY2NhYVJeXw0hger1eeIDAwIED9Xq9X60taG1VrhWf/9Tlm5OdACAo0PLR3FaXRNHBDFyt6/0wElhoaOhdd90FAC+88MKWLVv8Z0mH46WDQk6WRLyf2w0AAmRkq72pLUpKSgrvmwnxflj75Kirq6NOxXye1rY/2mKxcNNuHgi07ZkLMpn9X5+goKCkpKR2qx/CBEYCKyws7Nev35IlS9566y2a8sorrzgePeu4VFZW1tfXO5PTbOZ4r2mrZxZHh5tay4kTXB0URgJLS0vzh26hXT8zVsyePRsAYmLiyuN205R/T73SLcbOeZ8052+//Ubn5fkrX758mTrHjY2NraqqarNW/EBIUFAQdd5GU0aNGnXw4EGnbswhL7/88r///W9Aj3r2YNpFjI+PV6vVCQkJDrZddETaXDpohb7vQT0XCAAzRpX8+Y5WRz4IIUeOHLFN/+abb2jA1T11BoNBGD106JBLxVuDqguxiwQ+OcCHHF/X19cLDwVvk+e2JuvTDgPAsL6aOeMqtFotP2tsRWpq6r59+/r379+rVy+NRvPkk0/yH82bN2/v3r3FxcUFBQXOGF2yZMnhw4fLy8v537Wvv/564cKFR48edb7mDiCE3HHHHc8++2y7XM3HYCSwyspKoaIqKio6uuPR4uJik8mk0+mcfJN8c1ePoopQaGXXlhClUkkXuyiVyrKyMrt58vLywOlh+n/9619WKRMmTJgwYUI7jvKfO3euvS7lYzASWFlZmdDzXnh4uFV3pQPhzIuWkG0/xB842/JTYus1zQoczPAxmHYRhVFn3s69Csc79u3CT21xHGx/tg1p9ejRw5OlfYh3wm6xb8d96XK8Y98utTr5MxtbNmW12Wrhrn4fBtcEOIK+aLlUxEJg2qqWbt4HTxVHhrVRHFfr+jYoMDu05t69TfhZ4xcfKhvQQ+c4c1RUFK7W9XlQYLdQW1t7/fp1Nwry0pp+3/XRd9a2mR9X6/oJKLAWbI+oc5LntvSuqlcAwPC0+ll/aXsbDrqj8Sv8XWCEELtH1DnDu191P3MlHAASYwzLp7a9rjcwMNDtM46RDor/CqypqenKlSt0172rZb/4OS63IAacG3+n4ASXf+KPArM9C9x5+KktAPh4QaFOpwNoYyVHTExMXFyce+aQjo5/Cczt3iAAVGsCF25u6eC1ObXFgw2Xn+MXAnN1tbst/CDhpnkXFHKnZsz79OmDnlURHxeYk05mHMAfCPTSpLL+t7UxtUVB37oIj88KrKioyGKxtJ2vdXhpPZRePSnd2Z1X2CdEhPiawDzvDYLAIeHAJO2izHInS0VGRgp3DCAI+JLA1Gq1S3sf7bIyt/upknAA6BZj+LcTU1s82HAhdvEFgV28eNHzIzO/O93p0//tAm05JLQFt5kgDujYArtw4YLnDkzPlYa/s6c7DTs//g4AHMclJSWhuhAHdEiBNTQ0XL1qfQyxG1Q3BC7c1HJqyZZnLshbcUhoF7rNxMMhSsTn6ZAC89y1vdnCTb/p63PzvKJAuQvjjcHBwb169fKwAoif0CEF5iH8rPHrj5bGhakC5S6sRcTBDMQl/Etg01alWQgAwORhqgeGVgOA82sS4+PjO7QbLEQS/EVgs9amNjbLAKBPQuP/m1LqanFsuBD38H2BLf/vbf9XHgYAMRHG92c4Ol/cLsnJyYGBgSLUC/ELfFlgB85Gb/shHgCCAy0bXZnaouBgBuI5vimwMlXwK5+2aGP7s4Vu+JXBPiHSLviawKobAhduatm15eCsLQfExsZ6cjQ4ggjxHYEZjLKZa1JpeP3TRaEKl5fScxzXt2/f9q4X4td4kcDy8/PT09PdK8uPv7/+aGlSl0Y3roDOnhAx8BaBZWZm0vPmXGX+5rsMRhkA/H34jfGD3Fm4hPsjEfHwFoHt3bt33759NDxp0iSLxWI0Gmtr7Xvw1Ol0hJBntw5sapYBQFKc9qUHiwBAr3fZbs+ePeVyeWuG2sRisbhdltJxi1ssFk+OyPH80bmE226OPMRbBCYkJycHAO65557WVk4sWBv8f2UhABAWbF457WxISAiAy0vaIyIiunXrVl1d7cn6DLVa7UlxD61LW1yv13uyk8DDR2exWGQymfP53XDO1y54o8Da5GqNgp/aanTnhQtH4RFGeJHAJkyY4GTOtbMvue1vIzExUalUulcWQVzFiwQmNjgKj7DHXwSGSwoRSfB9geGRC4iE+LjA8PxIRFp8VmB45ALiDfimwHAUHvESfE1g6KUQ8Sp8R2ABAQF9+vSRuhYIcgs+IjAczEC8kw4vsLCwsNTUVKlrgSD26dgCS0tL02g0UtcCQVqlowqsa9euUVFRUtcCQdqgQwoM+4RIR8GFHTUIgrgKCpAM/WEAAAWeSURBVAxBRAQFhiAiggJDEBFBgSGIiKDAEEREUGAIIiIoMAQRERQYgogICgxBRIQjxOUDftiwaNEik8nUZjZCiCcbVU6cODF48GC3i/uzdQ85e/bsnXfeydLi6tWrWZqjeK/A2DB48OATJ06gdfaMGjXq0KFDUllnhr93ERMSEtC6JMTHx0tonRn+3oIhiKj4YwtmMpmysrKWL1/Op2RkZGRmZjKrwKRJk5588kmh9YyMjBdeeIFZBd59910+/Nprrz388MNu+/p3A+FBcFlZWfT2mVlnDfE/Jk6cSAjZtGkTjWq1Whr45z//ycB6ZmYmIeTSpUt8ypw5cxjY5Xn99ddXrFjBR3NzcwkhGRkZbKw/8cQTs2bN4qP0b+HD+GMLRuGHsM6cOUMDV65cYWCXthW9e/fmU9LS0hobGxctWsTAOgBkZ2cLoxMnTgQAwupNYfPmzcIoHQVl2X1gjP8KrKSkhAZ4Z28sX/pv3LjBh+fPnx8SElJcXMzMupDz589LYpfy8ccfS2idBVI3oRJQU1Mzc+bMxx9/nNzsomRkZEyZMoWNdaPR+Pjjj9MuGbU+ceLE6dOn6/V6NhUghNAu4uTJkwkhGRkZM2fObGhoYGaddhFpx/iBBx6YPXv29u3bmVlnDI4iIoiI+G8XEUEYgAJDEBFBgbkDx3EPPvhgYGDgqVOnGJj77rvv4OaAm+Na/eUvf3HpyqNGjbJKKS8vV6lUDoo0NDTU1tYCwMqVK/nE+++/f9myZQBw4MABlyrg86DAXIbjOELInj17jEbjoEGDaOIbb7xh923WbDYLZ7T5L+Urr7zCJ+7YscNoNPJR4UeUhx56SBhdv369wWCgYULI559/zn+0f/9+ANDpdDS6bds2PicAvPzyy3xYWCse4Qh+c3Pzhx9+yEe/+uors9kMAGq1mk9ctWoVDeTl5dGA8yfZ+wvSjrF0RGwfGk359ttvf/zxx27dutHEAQMG5OTknD17ls9A/zeZTG+++SYhpHPnzoQQKlcqCUJIr169CCETJkzgL24wGOgQn/AiWq320UcfLSkp+eWXX6w+Gjp06K+//qrVamnUaDQuXLiwpKSkvLzcKufBgwfvu+8+q7tISEioqqqaP3++rdFLly7l5+eXlpaq1WpCSEhICCGEv196U/v27WuXh+wzoMBcxlZgOTk5NNC1a1dCyJo1a5YuXUoIiYiIEP6Q8QXffvttAAgICBAmAkBDQwOff8OGDTT9yy+/FNodMmQIH+XLRkVFEYHAhBn4bPwCJUJIYmIizcMLTKfT0cUltbW1VVVVVmVpVQkhcXFxvMDeffddQsirr75KP6ICI4QcP37cncfqo2AX0WWEsqHvRTk5OTR6zz33AMDcuXNpXyslJYU+5aeeeoovkp+fP3PmTEKIrW/98PDwnTt30iJjx46liSkpKcI8BQUFfHjp0qU0UFdXZ7eq9FKvvvrqwIEDP/zwQ3JTdRUVFVY5Q0NDadf0+vXrACCTyWjZMWPGAADtHALAjBkzrO7dCrVaLeEeMy+kQ/qmlxaNRhMUFNTc3Aw326XQ0FD6baPRrVu3UhGePHmSpguXt6anp3McJ5PJCgsLm5qajEYjx3H8IpJly5ZlZWWBYO3S7bff3r9//19//ZVGDxw4ILQVHBwsfMsSkp+fT3NaLJYlS5bQMP0JOHv2LMdxnTt3HjBgAJ9fLpdzHLd06dJ//OMfZrOZ5l+zZg39lDdKCKHys2t0/Pjxx48fd/WR+jKMWkqkFeifwGw2jxgxorU8fHtiRV5ensFgILe+KYmBM1emXcTo6GiR6tBBwZUc0rN58+YRI0YkJyc7yFNeXt69e3fb9NLS0h9++OGJJ54AgKtXrwJAt27d2r2GrVnnuXr1qlKpvHHjhlWHFkGBIYiI4CAHgogICgxBROT/A/UtaiVCX8rKAAAAAElFTkSuQmCC" /><!-- --></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="co">#Linear regression plot without filtering sequin data</span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>EBImage<span class="sc">::</span><span class="fu">display</span>(lm_example)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAIAAACb4TnXAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAgAElEQVR4nO2deXwURfr/q2dyTkJOSCAJR0LIBQmGQ1n1xym6vEgCCuEURREQWFBwUa4fuCD75VpBkeXYLwjoohEEJaCAxEW5wgLBIJqQg0AuSCYzOefKHPX9o4be2UkymU66a67n/Ude1TXV9VR35jNdXfXUUwzGGAEAIAwiWzcAAJwZEBgACAgIDAAEBAQGAALiugLLzc01y9m9e7dNWkKB3bt3NzQ02Mp6amrquXPn2vqUYRiajaGM6wrsiSeeMMs5cuSITVpCgSNHjjQ2Ntq6Fa6I6wqsJRcvXjQ9LCwsTEpKMiszdOjQrVu3socY41deeWXu3Lkds5ifnx8TE2MwGEwzR48evWHDBgtnYYyTkpKOHTtmmjl9+vQtW7aY5vzxj3/88ccfSfrixYvh4eHsR0OGDNm/f79ZtSdOnBgxYoQFuytWrFi4cCFJT5kyZd26daaffvjhh0899ZTZKWPHjl27dq1Z5qRJk6ZNm2bBkFOBXZWW187mkDvzySeffPHFF2zmmjVr/P39McZEhyQzJiYmJyfnxo0bCKHy8nKM8bVr13r06IEQ6tu3b11dnVn9Xbp0EYvFEomEHB48eBBjHB0dPWrUKLbMlStXrl69ihA6e/YsxnjatGlDhgxBCPXq1QtjPHLkyP79+2OMyfMWY/zLL78wDCOVSn/++WeSk5ycHBsb29DQ8Omnn5IchFB1dTXG+KeffkIIyWSyzz77zOx6jx49unHjxla/EqTADz/8MHz4cDc3N4TQTz/9FBIS4u3tzRY4ceKEVCpFCN2+fZvNPHPmzJUrV9hrUavVCCGlUvnw4UNT6xz/dY6EM1+bZSwLTKfTkTTDMC3Ltzw3Kytr4MCBGONr166NGTOmXYs1NTU+Pj5mHzEMw2py3LhxrMCOHj3aaiUkPW7cuJdfftnMkF6vN8shAjM9vXfv3llZWe1emoUCJP3tt98mJCSYZbq7u8tkMpKTkpJCrsX03Hnz5m3btq0ti06DG19PQmdFIpGw6Zav4yRn+PDhzz//vF6vt77azz//XKFQmFWIMfb392/3XLOzvvvuO4ZhPv/8c5FIRNrQ2NgoFosRQh4eHhqNpq16Jk2atHnz5tGjR1vf7FaZNWvWyZMnzTK1Wm1QUJDlxk+dOrWTpu0feAfjgOkvE0Lo9ddff+GFFzDGP/3004QJEzhVNXDgQNJRNK0QPe7OcWoGm9PU1ES+vr6+viQnKyvLwhhddnb2yJEjOTW7VWbNmpWRkWFlYdPGf/nll523bue4tMAaTWi3cHBwcFRUFEmTQedBgwZdunQJIWQwGBITEzmZHjlypFKp/O2338hhr169EEJXrlwRiUS3b99eunTp999/3+qJbm5ukydPJul//OMfCKGEhATymPL29ib5fn5+JJGQkGB2elxc3IABAxBCGOMrV66sXLmSU7Nb5ZNPPmFnOMLDw8lz6a233iIP/6amplOnTpFPt2/fzgq+pKQkJyen89btHX57nA5Ey/uAWnsHM31T2rlzJynMvvOEhYUhhPz9/X/99dcBAwZgq9/BCKwk8vPzSU5tbW1aWtrDhw/HjBlTWFiIW7yDYYxffvllctb69esxxqYTXKTAL7/8Qg7ZxqPH72AY4+nTp5NPtVpty4a1+pVoqwCbVqlUpM5Vq1axnxIlh4aGsu9gGOOCggJSslevXgaDoS2LTgODwZveLiGDK7ZuBdBZYJDDvmB7UGQ8HXB07Pdncvz48RZGwACAK+fPn6dv1H6fYHV1dZcvX66pqenatav1ZykUCh8fH05WAgICuLeOA1ybpNFoPD09rS/P9RbJ5fJWB9D5qt9gMIhEHAbP9Hq9SCTi5JGIMeZUvqamxlYDKi49iggAQgMCAwABAYEBgICAwABAQEBgACAgIDAAEBAQGAAICAgMAAQEBAYAAmK/nhwIIYVCwf61Eq1Wy6m8wWDgVL4DcG2STqfT6XScTHCqH2PM9ZK51s/JzcJgMDAMI6gnB0KIRCugj10LzMfHR6VScfIz4uqXpNVqOZXvAEK7SnG9RRqNhlN5rvXboauUSqXy8vKyvjyPQBcRAAQEBAYAAgICAwABAYEBgICAwABAQEBgACAgIDAAEBAQGAAICAgMAAQEBAYAAgICAwABAYEBgICAwABAQEBgACAgIDAAEBAQGAAICAgMAAQEBAYAAgICAwABAYEBgICAwABAQEBgACAgIDAAEBAQGAAICAgMAAQEBAYAAgICAwABAYEBgICAwABAQEBgACAgIDAAEBAQGAAICAgMAAQEBAYAAgICAwABAYEBgICAwABAQEBgbeLu7m6W07VrV4SQ5f3txWKxWRmFQjFnzpzS0tLExESEkEgkSktLS0tLMxgM/DcasDPcbN0AI6mpqUFBQYcOHRLaUN++fYuLi00Ty5cvZz/dunUrSTAMk5OTY3piQUFBTU0NQmjEiBGm+enp6UePHiXp06dP6/V6kskW8Pb23r9/P0Lozp07CCGM8cmTJ81a1a9fv8LCwpSUlK+++opIFHAO7EJgt27dyszMzMzMpGDr3r17CKEzZ87cvn2b5LCiMgVjfOvWLdOcmJgYkv/zzz+b5hPZEH744Yfx48cjhFJTU9lMkUiEEGIYpqqqiuQUFRWNHj06Pz9fIpGQHLFYXFlZOXjwYIlEotFoOnuRgN1gFwJLTk7etWvX2bNnyffyySefNBgMDMOQJwb5az0qlcrCp5MnT66pqRk3bpxUKiUlTbtzUqmUTdfV1ZmZ1ul0ERER1dXVJL9Hjx4IIb1e7+7uvm3btpkzZ/r7+5OP7t27Z3quSqWSSqWhoaFVVVXERE5OTlBQUHl5OSlw6dKlyMjIkpISclZjYyOnS+Z6i4Qub4dwvaW8ge2AkpISkigrK2Mzn376aYyxVCrlVFVTU5PlAoWFhZMnT2YPa2tr2yqZk5NjltO7d++WxeLi4ti0TqfLyMjAGJveWIlEQhIkkxyq1eolS5awZRBCUVFRJK1Wqy1fghlcb5FMJhO0fr1ez6m8TqczGAycTuFaXiqVnj17ltMpfEF7kCMrK+vAgQOVlZWmmcHBwX/+859TU1MjIiKEbkB0dLTZy1VbeHt7k8S5c+cQQqNGjfLz80tKSkpKSjItFhsby6bFYvGtW7e8vLwUCgVCaNq0aQih6urqhIQEf39/MqpRVFTk6ek5c+bMjz76iJyVlJSEMS4uLh44cCAPVwjYFdSkjBA6cuTIf5QtEq1du9ZCeSGeYLW1tWaXbOEJxhftPlTNgCdYuzjQE4zSO1hpaSnG2DSHjLZRJiAgwKwZACAolASmUqkmTJjQMv/bb7+l0wAAsAmUBFZbW9tSS08//TQd6wBgKygJbNiwYaaHKpVKq9VeuXKFjnUAsBW2cZXSarUwnQq4AuCLCAACQklg2dnZjAn+/v507AKAbaH3Dgbj44ALQrWLaPoQM/X6AwBnhaqzL/sQk8lkPj4+NE0DgE2wzSBHUFAQu1oEAJwYqk8wdmFIcHCwE6yAAIB2oSSw5557bvTo0TDOAbgalAR2/vx5khCJRBhjvV5P1vkCgHND+1tOFhqIRCIYRQRcAaoCk0gkzc3NDMMMHjw4KCiIpmkAsAlUBRYQEODh4ZGWlnbz5s2zZ8/SNA0ANoHqKOLDhw8DAwNlMllCQgIM0wOugG0mmn///XeadgHAVlDtImZkZDAMk5mZaTk4LgA4DVQFNm/ePPw4ehmstgRcAaoCa2hoIInr168nJCTQNA0ANoGqwDDGYrE4LS3t+vXrAQEBNE0DgE2gNMihUCjIwEZ2djbJ0el0bm52EbgbAISD0lfcx8dn6NChdGwBgP1A21WKrLYEsQEuAu3lKmQUsaGhobq6OiQkhKZ1AKAP1ScY60Hv5+cnl8tpmgYAm0D1Cfbjjz8yDBMWFlZZWQlrwwBXgKrARowYAboCXAqIKgUAAkL1Cebh4QERswGXguoTLD8/n2GYLl26dOnSBYLeAK4A1ScY2YaYpkUAsC1Un2BeXl40zQGAzaH6BIuPjx80aBBJZ2VlBQYG0rQOAPShKrCcnBw23dTURNM0ANgE2sP033//fWZmppeXl6+vL03TAGATqD7BQkNDx40bt2HDBrVaXVhY2K9fP8vl6+rq2L9WYjAYtFqt9eX1ej2n+jsA1yYZDAaVSsXJBNdbxPWSOZXHGHMKCUHGvYSOIqFQKAStvy2oCkwqlVZUVKxdu9bPz2/q1Kntlg8ICKipqeG0NFOhUHDat6Wurk7opZ9cm6TRaDw9Pa0vz/UWyeVyTuW51m8wGDiFbSZhnrlqklP5mpoaW+3mQ1Vger0eIYQx5vo/AAAHhdK3HGN87NgxhmGOHTt27Nixv/zlLxUVFXRMA4ANofQEYxhm8uTJCCHTvwDg9Nimn2YwGEpLS21iGgBoQu8dbNOmTezfbt26zZkzh5ppALAV9AS2YsWKFStWsIcqlcrb25uadQCwCbQnmg8ePLhv3z4PDw9QF+AKUBVYaGjo7NmzZTJZc3Pz/fv3aZoGAJtAVWBSqVQqla5ateqLL77gNJcKAA4KTDQDgIBQ+pazEbONVkUihNDTTz9NxzoA2ApKT7CkpKRff/3VLHPv3r10rAOAraAkMIlEkpiY+O6777I5iYmJs2bNomMdAGwF1Xew3bt3NzQ0MAyTkpKyZcsWNpI2ADgrNoiLiBA6depUTU3Ne++9R9M6ANCHqsBmzZrFMMzo0aMZhunateuWLVtoWgcA+lAV2K5duzDGP/74I+kZGgwGmtYBgD5UBbZgwYJhw4atWLFC6PXhAGAnUBXYP//5z+zs7E2bNmGML1++TNM0ANgEqgJrbGwkiQsXLiQmJtI0DQA2garAMMbJycmenp61tbV+fn40TQOATaA0D1ZfX3/jxg2E0LZt20hOc3Ozh4cHHesAYCsoCczf33/MmDF0bAGA/UCpi/jqq6+WlJSYZQ4dOpSOdQCwFZSeYIcOHTIYDKaj8ytXrrx+/Tod6wBgK+j5IopEIvA8BFwNWPUIAAICAgMAAaEqsBs3bjCPkUqlNE0DTo9er8/Ly7N1K8yhuh5s6NCh8BoGCIEdSotAVWAAwDv37t3TaDS2bkWb0H4Hgy4iwBdKpTIvL8+e1YUoP8EwxjKZrL6+PioqiqZdwPmw2z6hGVSfYP7+/uXl5aGhoUJvKgk4MYWFhfn5+bZuhbVQFZinp+fAgQN9fHzq6urOnDlD0zTgBMhksry8PJ1OZ+uGcIBqF1Emk5FNVfr163f37l2apgGHxmAwOOgXhnbo7KtXr966daugoACiBgBWkp+fb83sTuFD7/UZfW7YWTBbSgLLzs4OCgqKjY0lh4sWLaquru7WrRsd64CDUlxc3Nzc3G6xo1e6nfx3Vwrt6QCUBDZs2DCEUFu/Q1evXh04cOA777yze/duOu0B7Jz6+vrKysp2i839e6y62TiO8NnbZFwxXsh2cYaSwDDG06ZN++qrr6ZMmYIQksvlhw8f7tGjB/n0D3/4A0Jo+/btdBoDdB62h09+NOvr68nIsOlv6M2bN4cMGWJlhWq12svLCyHk5+fHbhVSVlb2wQcfmBaLjY1dtmwZQui1nXE6PYMQEovxwcXGQcX58+dfvHjRbGGUbaEkMIZhMjIyTp8+nZGR0WoBnU63cuVKorH09HSMsVarraurQwiRv1ZiMBi0Wq315fV6Paf6OwDXJhkMBpVKxckE11vE9ZItlCcfvf/+++SwqKgoODiYfL/Xr19vvYm7d+8yDPPss88ihJRKJcm8ePGiWbGC+/WzdhifUfHhirfH5yOESHF2RW9lZaWPj4/ZiQqFwvrG8AjV6PAqlap79+4IIYxxcXEx+w6m1Wq/+eab9PR008LPPPPM5cuXa2pqunbl0L1WKBQtb64F6urqhJ6U49okjUbDaXdCrrdILpcHBQV1sn61Wu3t7a3T6cRiMckJDQ0dMmTI6dOnTTd/8/LyYj0tZDJZcHAwQujNN9/cs2cPQig3N3fgwIEIof79+x87duzSpUuHDx/es2eP6fNn0aJFgwcPvnbtWurMDzJyhpPMheMq/hDb0LKpq1evrqqqahkRsKamJicn5/nnn7f+qvmCqsDIbg+FhYW3b9+eNGkSm//Xv/6VJFatWsVmgsCsxCYCswDX3RVLSkpUKpWFTt2+s2EX8/xJeuPLJb26qi3UFh/fyjuYDQVGdZheIpEghFasWPH111+fO3eOvWBTXQGuQ01NjVQqtfDKNO/vsapmEULITYw/Xeww3humUBXYtm3bpFLpiRMnRCKRNcOvgLOiVCofPHhgocCrH8UbMEIIMQw6/JZjuB22ClWBvfPOOwsWLIA9H1wZjUZz7969tj6tlHu+d9joCP7/EurnPd/+SL2dQ1Vg7u7usP25y6LT6QoLC9v69MKdgP3njdM2y18sTerNbdBPIpH07t27U+0TBqoCmz179tKlS0l648aNvr6+NK0DtgJjbMH//f2MqJIqb5LeNa/QT8LBlzckJISMTNotVAX28ccfs4OWZWVlIDCnR6/XFxQUtPXpG7tiNVoRQshdjA9wGcPo3r17YGAgD+0THnoCI7805K9YLK6urqZmGqCPBWmptaK5u4xeqYG+uh2v32UYBqH2fS/69u1L9jNwoMgu9AQmk8mo2QJsiFqtbmuE8H611/8/EknS056tHj9EhhBqd8wrKiqK08SgXUG1i8gwzODBg729vRFCmZmZsK7ZySAeuhjjlvNaJ7K7Hs82Ou6sTn8QF65st7bIyEjinejQ0J5oJpsYAU6GhdBOC/bENKnFyOoZrZiYGNb9ygmgKrAvv/xy8+bNTz75JELo2WefdXd3p2kdEIK7d++2NbH5+idxWh2DrBvD6Nmzp1MOelEVWGpqampqKk2LgECo1eqW+1ER5E3ub+/vR9Jx4crV6ZY8Nvz8/Lp37y4SiexngQm/0A48yjDMyZMnX3rpJU4rOAD7oaioqK3/XUGlZMNXxtnepWnlg6IaLdTDeuXq9Xp+W2hX0B7kwBhnZmYqlcqKiorw8HCa1oHOoNVqi4qK2vp05+nwfxcaN93eNKsoPLjNX0+HHhLsAFQFxjpJubm5qdWWFh0A9kN5ebmFMMzsGIZYhA8uyUfGSSrz/p6vr2/Pnj0Fbad9QlVgpaWlDMOIxWK9Xu9Ac4WuSV1d3cOHDxFCWq221WfOKx/Fk/+h5bUkcXFxzvp+ZQ1UBRYeHq5Sqby8vNigC4C9YbkriBCqkHuueOzwPiaxdvaYR60W8/Hx6dWrF//tczRs8A6GEAoICKCwlBiwHoPBUFBQYLlbkV3gt+s742vz8ollSX2aWi0WERHh5+fHfxMdE6oCI45kCKG4uLjbt2+DwOyBwsLCdoNRrzgcVSE39hL3Lbzr7dH6xBfpDcJ6P1OoCuzo0aMMwwQGBtbW1sI7mG158OABG7zJAvN2x6o07Ti8i8XimJgYntvnLFAVWFpaGujKthQVFdXX11dVVVkuZurwHuDTvHNucavF3N3do6OjeW6ic0F1cbGfn1/nN+Ajp+/cuZPftjk3RUVFeXl5eXl5pnPECxYsmD9//vz58xctWsRm3nvkNWtHPFHXjBFVkrzhzTeea1mhn59ffHz8N998wzCM6RJ1MkpsWvLjjz8m/zLTTLI5AcHNzZm3WaV6bTqdjq8n2JIlSxYvXsxLVc6KwWAoLi628H7FviyRMqYR3r3uLxSp7kT+0bj8/Lfffuvfvz9JBwQEsCGZly9fjkxWZ61YsYIkTCPVvfXWWy1Ns7sUOD20QwY0NTURH18PD4+OTY/4+/vX19eDp1VbWL/TT1hYGIn/rorJnLXDH5k4vM+ff0ckEpFBC4wxUVfLkXcSYYUEk0UIbdq0afPmzSKRyDQOJBGb2UYfZEkLGUwmy5ecFaoCe/vtt9mI/n379u3YqgShI107KE1NTWVlZZxOWbdu3ZxP4pqJw7sbPvCn/4xh7N1r3AVoz549TU1Nfn5+rfq1sfMuLBhjs1FEiUTSareFzQRfRN6IiYkRiUQY4zVr1mzYsIGmaWdFJpOVlJRwfaGtaXBfesA4OJHYW/Hui6VtlfT29u7WrRunyL6AKbaZaC4rK6utrXWUuCV2iDWTV61SUuW19gujG/ubLzx8Jr7N7gDDMHFxcQihmpqajjUSQJQFxrq09ezZ8+bNm4MHD6Zp3dExGAxFRUUd7lBt+7ZnbolxReO214pD/S1FVm41wjvQAagKbOvWrQzDjB079ocffoAJMSvpwMuVGW/uiVGoxeixV65SqSSbBLQKSItfqAps8eLFMLZuJaWlpZ3f0oqN8G7N5glOFgzDTqAnMPYFLD09/fjx4849dtQxyMyVWq3u5Nxrmcxz1WdGh/cXkuUvj2jHbyMsLMzf378zFoG2oCSwY8eOsX3Co0ePVldXP3r0iJ0/cXEUCkVpaZvjeJy4+HvAvnPGWeBlaeXJFhftI4S8vb379OnDi2mgVSgJ7Lvvvps8eTJ7GBIScuXKFRcXmIVQZx3gvcN9K+XGxQp7FxRIPNvvIMDrFgUoCWz//v3Dhg1j11l269atw76IDo1arb5//z6/AzzLDg9SNYsRlwjvLr7KmCb0NkE/ffo0+091tRcwspUjv3UqNaL5u40efd0Dm7e+2rrDuxl9+vSxvF8rwC9UN39wtaF5EuGsubmZXWnKCw+qvdY8jvA+5ZnqUfEPrAnZyXoSqlQqHhsDWMaZVwrYBDZWjBAc+Tn0+xzj/uVr0h/EhisRQk2tr9z/D6xPBkAfmwlMr9c7zawLxvjevXuC7jq9cG9Mo4pDhHcWGMmwLTYQWHBwsFwu12g0ji4wIUYsWtLhXepgdsseoCewI0eOzJw584033ujevbtD7xVWWVlZX18vtBVTh/fkqKZlaRy8pTw8PPr27StMuwBuUBJYdnb23//+d4PBwDAMuzbWgbB+FWPnuVsh2XTCKI+5Yx8O789t/Rv0Ce0KSgIbNmzYpUuX1q9fv27dOgcKSf/w4UOa6zt3ZEbcLO5C0ptfuRcWxG0auk+fPs69OtgRofoOtnbt2rVr1yKEwsLCcnNzzZaR2wnNzc337t2jPKPAOryLRfh/F97h6osokUh69+4tTNOATmGbUUQ2cID9IMRcsDXM2mHs0bFjGFwXUkZGRsKKY7uFksCUSqXpnvPJyclKpdLm/RmMcVlZmXDTVhZ4VOex/KDxRWtUYt3rYzrSBrIVEKw4tmcoCUwikTzxxBMIoeXLl3/66ae2dekge3WTtFarpRyXL+t24MEfjV7Oy9LKkqPamyduDdhawVGgHZOjrq5u69atNI2yWNhNmA7LDkRLG4zbUu//U76HWwd/ZawcJxw0aNCtW7cQQhKJRKlUSqXSsLAwEu4OY+zp6dnc3EzCp3l6ei5cuHD79u3BwcEymay2ttb6bQPYejp2LU4PJYHl5eUlJCSsXLly8+bNJGf16tWm0fMEora29tGj1vfXocn83TFKzX/tUtcxoqOjrd85nqgLIURi0JsOKU2cOJH4nRCvX41Gs337doQQmZ8MDAy0XjCknvDw8IqKCitPcSkoCSw+Pp7aj1y7EW2p0aQWL9hj3BWhdzf1BzNb3zXcGjqwQ6S7u7tpeNbFixez8ca//vpr0jG+du3aU089hR5vKEU+3bVrF9fm5ebmcj3FRaDaRezevbtcLg8LC7t//z7vlTc0NFRUVGg0GnvYArisxnPV58ZF+5OffjThydrO1NaxueOWvpEff/wxQkgul4vFYvb3rpM/fNA5tIwNYnKg1iLCdgyMcWlpqTXb8FDjy4shp28Gk/TqyQ/iIpTNzc0IdXC5Cqc+IWCHUBJYZWWlqaIqKio6E3iUxyAWPNJhh/dWgblj54CSwEpLS8PCwthDX1/fzoSjKC8v56NRvPH6J3FaHYMQchPhTzsxhsEC/oROA9UuoulhdXU1NdMCIWt0f3u/0eF9YGTTnyd0KjwoITIyEtbzOxP0nH2d6W34TqnP5uPGed45zz0cOYAHh2BPT8+oqCiEEI+hpgCbAyEDuLHpeK/fSo3Tdx++VtTNn59tyqBP6KyAwKxl0b6YBqUYISRi0KFOj2Gw9O7d20KkeMDRAYG1g97AzP7YGDFG4mnYu4C3ZZdisTgmJoav2gD7BATWJg/lHu8eNjq8jx1Y+8ooPl2uoE/oIoDAWuGn3wL+9wdjhPc/ja94ql8Dj5VDLBqXAgT2X8zfHavUiEj6s7d5e9FigQeXqwECM8I6vItE6NAS/qUF4eBdE1cXmFIjnrfX+FSJCNb8z6x7vJsIDAyExZEui+sKrFTqtfqfjyO8P1udOkSQUI3x8fGd36gScFxcUWCfX+h+9hejn/HytLtJUYIsc46NjRWJRELUDDgQriWwhftiGpXGeN1kDEOlUiHEc+ydDiyOBJwVVxHYazvjdHoG8e2H0RIYJwRMcXKBmTq8D4pqWsolwjtX+vbty+8+YIAT4LQC+73M53++No7dzR5dNSZJLpwt2GwBaAsnFNgHR3vfrTC6z340pyioCz8O720BfULAAk4lsAV7YprUxkX7h97KE3paNyIiokuXLgIbARwbZxCYqcO7l7v+H4sKLJfvPCKRKDY2VmgrgBPg2AJ7WOvx7iHjy8+4QfIZw6s0Gg1CwoZtA6cnwHocVWDnfgn67EIoSS9JqRgazafDe1sEBweHhIRQMAQ4DQ4psJkfGrtnh9/Ko/YsgcEMoAM4pMDee6l8QK9Gaub69etHeQcWwGlwyO9NUh8FnW1SYJcgoJM4pMDoAH1CoPPYkbt3SUnHNx/hlz59+oC6AF6wF4Glpqb+/vvvtm4Fcnd3j4+Pt/netoDTYC9dxMzMzNOnT5M02WVHp9ORbVNabp6i0WjaihOs1+tbbttjAYPBwJaPjo5mGIb3vVq0Wi2nOnU6nV6v52SCU/0YY67XyLV+TvOEBoOBYRhOp3A1gWwXL3G//PUAAAgESURBVNleBGbKkiVLEEIZGRlk79OWcTk9PT3b2gwWY8zJpV2v13t4eHTv3r3DW720C8aYU2hRrluctXqLLKBWqzmV51q/wWDgtNJUr9eLRCJBBaZUKm21a5w9CowmDMPA6xYgHHYksPHjx1O2GBMT09hIbz4NcEHsSGA08ff3N92vDAAEwhUFBn1CgBquJbDIyEgvLy9btwJwIVxFYLCqH7AJLiEw6BMCtsLJBRYaGhoUFGTrVgCui9MKDFb1A/aAcwoMVvUDdoKzCczHx6dfv362bgUAGHEqgcFWJoC94SQCi4qKspU3JwBYwOEF5u7uHh0dbetWAEDrOLbAYIILsHMcVWAwwQU4BA4pMJjgAhwFe4nJAQBOCQgMAAQEBAYAAgICAwABAYEBgICAwABAQEBgACAgIDAAEBAQGAAICAgMAASEaWsXBZuzbNkynU5348aNIUOGCGclLy9PaI/hsrKynj17Clc/11vENbC70P8CpVLp5eXFKZw9V8glkE1FaIPtm8GDBwta//Tp0wWtH2O8ceNGQesX+hYJXf/t27elUqmgJoS+BAvYexdR6ADXISEhgtaPEAoICBC0fqFvkdD1u7u7C/r4QsJfggXst4sIAE6A/T7B0tLSJk6cyB6mpqampqa++uqrvFQ+YcKEtLQ09rC0tHTGjBm5ubm8VI4QOnDgwLRp05qamtgc0n6+6kctNsWbOHGi6RXxwmuvvcam58yZw+8lTJs2bcqUKaYxVNLT0999912+6r969eqMGTM2bNjA5vD+L7AKW/VNLdPyvWXGjBl8VS6Xyw0GA8Z4x44dJGfixIkY45deeokvE6tXr8YYp6SkkMNff/2Vr5oJKpUqPz+fPWxsbCRXtG3bNr5MvP7667Nnz2YP2Wvhi9LSUtNqU1NTMcaXL1/mq/5169bhx/8IG2KnT7Ds7GySkMlkJEHWL8+cObPzlVdVVZFhtGvXrpEcd3d3hFBCQkLnKycMGjQIIcTutXno0CG9Xv/ee+/xVb+Xl5fpqtP6+npyRVevXuXLxP79+00PfXx8EEI8PgF69uyZlZWVkZFBDjHGCCEeQ+69//77CCG2E5GTk1NUVHThwoXKykq+TFiD3Qns1KlTp06dSk5OJofBwcEk8dFHHyGEGhoaOm+CrTMmJoYkdDodQujBgwedr5yQl5eHEGJ3f16+fLlYLN68efOjR4/4MmGKr68vSbBXxDt79+5FCPn7+/NV4erVq0eNGmW2Oa1cLuerfrVavWbNmh07dpDDmJiY6OjokSNH7tq1iy8TVmHbB6gFpk+fTroNL774IsY4JSXljTfeuHv3Li+Vp6SkpKenY4z37duHMT569OjChQt37tzJS+UY4yVLlsydO7egoABjfOPGjZqamtmzZ5PL4R3Sy0pJSZkyZQq/NZMuIqk/LS1t7ty5J0+e5KvyCRMmvPTSS2y3XKFQzJkzZ9KkSXzVn5KSwtZPLmHq1Knku0QTGEUEAAGxuy4iADgTIDAAEBAQWEdgGGbixInu7u45OTkUzJ05c4YYbbdVL7zwAqeaR48ebZZTVlYmlUotnNLY2FhbW4sQ2r59O5v5/PPPb9y4ESF0/vx5Tg1wekBgnGEYBmP8zTffaLXawYMHk8wNGza0+jar1+s3bdrEHrJfytWrV7OZX375pVarZQ9NPyK8+OKLpod79+7VaDQkjTH+4osv2I/Onj2LEGJnbw8dOsSWRAitWrWKTZu2imXt2rVsurm5ec+ePezht99+q9fr0X8P9JGhXYTQuXPnSGL8+PEtq3VpKA+qOAEtbxrJ+f777y9cuBAREUEyExMTjx8/npubyxYgf3U63QcffIAx7tq1K8aYyJVIAmMcGRmJMR4/fjxbuUajaWxsNKukqalp+vTpxcXF//73v80+evLJJ+/cuUPmfzDGWq126dKlxcXFZWVlZiWzsrJGjRpldhVhYWHV1dWLFy9uabSoqCg7O/v+/ftyuRxj7O3tjTFmr5dc1OnTp3m5yU4DCIwzLQV2/PhxkujRowfGeNeuXevXr8cYd+nSxfSHjD1xy5YtCCGxWGyaiRBqbGxky5P5A4zxiRMnTO0OHTqUPWTPDQgIwCYCMy3AFps/fz6bDg8PJ2VYgSkUiqKiIoxxbW1tdXW12bmkqRjjkJAQVmB/+9vfMMZr1qwhHxGBYYyvXbvWkdvqpEAXkTOmsiHvRcePHyeHzzzzDEJo0aJFpK/Vr18/cpfnzp3LnpKdnf3GG29gjFt62fv6+mZkZJBTxo4dSzLNnBuuX7/OptevX08SdXV1rTaVVLVmzZrk5OQ9e/bgx6qrqKgwKymRSEjXlEyFi0Qicu5zzz2HECKdQ4TQnDlzzK7dDLlcLujiMYfDIWPT25aGhgZPT0/ipUG+shKJhHzbyOHBgweJCG/evEny2acHQmjYsGEMw4hEory8PLVardVqGYZh11Ns3Lhx6tSpbFUIof79+w8YMODOnTvk8Pz586a2vLy8TN+yTMnOziYlDQbDypUrSZr8BOTm5jIM07Vr18TERLa8m5sbwzDr169/88039Xo9Kc/6PbBGMcZEfq0aHTduHOuABiAE72C2hvwL9Hr98OHD2yrDPk/MOHfunEajwf/9piQE1tRMuoiBgYECtcFBAU8O23PgwIHhw4db3kawrbgD9+/f/9e//kXWlZSXlyOEIiIieG9hu1EPysvL/fz8qqqqYINsM0BgACAgMMgBAAICAgMAAfk/0TuYD0/ufxkAAAAASUVORK5CYII=" /><!-- --></p>
<p>As discussed previously, removing the outliers may not always be the
best idea. However, the outliers can be visualized to assess how “far
out” are the outliers and how influential the data points are,
negatively influencing the linear regression method.</p>
<p>The <code>mag_tab_scaled_lm</code> function provides the plots for
the visualization of Cook’s distance for the sequin data in each
fraction. In the plot below, an outlier is visualized. For this data,
the Cook’s distance threshold was 0.09, and the outlier had a Cook’s
distance of 10.8. The highly influential data point could well be the
reason the linear regression model had a poor fit.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="co">#Cook&#39;s distance threshold of the data set</span></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a><span class="dv">4</span><span class="sc">/</span>(<span class="fu">length</span>(mag_tab_scaled_lm<span class="sc">$</span>scale_fac<span class="sc">$</span>cooksd[[<span class="dv">3</span>]]))</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="co">#&gt; [1] 0.1142857</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a><span class="co">#Cook&#39;s distance of the outlier before filtering the data</span></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a><span class="fu">max</span>(mag_tab_scaled_lm<span class="sc">$</span>scale_fac<span class="sc">$</span>cooksd[[<span class="dv">3</span>]])</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a><span class="co">#&gt; [1] 0.3588489</span></span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a><span class="co">#Before filtration</span></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a>EBImage<span class="sc">::</span><span class="fu">display</span>(cooksd_example)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAACIlBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4QEBASEhITExMUFBQWFhYXFxcZGRkdHR0eHh4gICAhISEjIyMlJSUnJycpKSkqKiorKyssLCwuLi4wMDAxMTEzMzM0NDQ1NTU4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj5BQUFDQ0NGRkZISEhJSUlKSkpMTExNTU1PT09QUFBRUVFSUlJTU1NVVVVWVlZXV1dYWFhZWVlaWlpcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRmZmZoaGhpaWlsbGxtbW1ubm5vb29wcHBxcXF1dXV5eXl7e3t8fHx/f3+AgICBgYGEhISFhYWHh4eIiIiJiYmLi4uMjIyNjY2QkJCRkZGTk5OVlZWWlpaXl5eYmJiampqbm5ucnJygoKChoaGioqKkpKSlpaWmpqaoqKipqamqqqqrq6usrKytra2urq6vr6+zs7O0tLS1tbW2tra4uLi5ubm6urq7u7u8vLy9vb3AwMDDw8PGxsbHx8fIyMjKysrLy8vNzc3Ozs7Q0NDS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dnb29vc3Nzd3d3e3t7f39/g4ODh4eHj4+Pl5eXo6Ojp6enr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///88yfMuAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAHH0lEQVR4nO3diVsUdRzH8SErECw7tcyyg0wLswPwSDuttJRuSosC7bTDqzvLLDvs8CorK1FELNCM6/v/tTM7M79Zdmc+u/s8uTz1fvm4/PrN/NjhzezAsj3iGTJ5tT6AyY5AAoEEAgkEEggkEEggkEAggUACgQQCCQQSCCQQSCCQQCCBQAKBBAIJBBIIJBBIIJBAIIFAAoEEAgkEEggkEEggkEAggUACgQQCCQQSCCQQSCCQUCLQ5pYauGlSLksJlH9zMr3qqYzig+mb/s5Ydix903h1y7L0lbfbaSNQJgIJBBIIJBBIIJDwfwnU6U38KAlUoIlAwrwtEyYIJBBIIJBAIIFAAoEEAgkEEggkEEggkEAggUACgQQCCQQSCCQQSCCQQCCBQAKBBAIJBBIIJBBIqDbQPuswAjlFgbbZ82bLl5wI9J9IlbEpa9vxjGV96ZsGqluWpcxlRycGMvvGOIOc4mvQ6k4jkMNFWiCQQCCBQAKBBAIJBBIIJBBIIJBAIIFAAoEEAgkEEggkEEggkEAggUACgQQCCSmBtgwE+gdSZWwaOJa+6XjGsr6MbTVbVuKFQx9nUISHmEAggUACgQQCCQQSCCQQSCCQQCDBBRqtS5QiUMQF8pJrCBRxgb58mzOoBK5Bggt09YjNjKcJFCm4Bn0cTxMokniIed6meJpAkUSggwcOxNMEihQ8xBwCRVyge5PTBIq4QA/lxNMEiiSuQT3PPuC/WdNmBHIKrkFzc29e2uNXIVDEBaqzRf25N8/YLrPdPSOBwZFUGZtG/kjfdCpj2bH0TcPVLcvSV95uQ3Eg2/1ec/C5etA4g5z4DBr1RkcfjqcJFOH7IMEFGjvi8eOOYskz6Paj8TSBIi7Qo7P5gVkJ/MBMiAM97YunCRSJAy3wxdMEiriHWLvZ8niaQJGC74MIVCxxkb5/lpsmUISvYgKBhMRr84d5qlFC8iK9gtfmi7lA3R5PNUpwgX5KThMo4gL90nfR2niaQJHEV7GvEx8XgSJ8mRcK/ueFx+NpAkUKnovtjKcJFHGBZptdFU9vyr8olPHi11DGa0n/0dfFdtziunEGRQquQdfG0wSKFFyD9sTTBIq4QMvMpsfTBIokvw9yX8QIFHOBOr/jyWoJBdegg/E0gSIu0LjnXRNPEyjCczGBQAKBBAIJBBIIJBBIIJBAIIFAAoEEAgkEEggkFAfq+N0I5BQFOm2rjUBO8Rn0eatZc0sf8nonBnrFdhhnkFN8Br05bARy+ComEEggkEAggUACgQQCCQQSCCQQSCCQQCCBQAKBBAIJBBIIJBBIIJBAIIFAAoEEAgkEEggkEEggkEAggUACgQQCCQQSCCQQSCCQQCCBQAKBBAIJBBIIJBBIIJBAIOHfC1T6uIsCJe6ivEDPnZ6wbb8bjmQcVCBx1H1lLSgOlP/9Ym81BRpzfxs8L3cb3Ex1w3PCYWPBbJMbeu62MRye64ZTC3aY6mYb3fuNhg0Fs9GyuoI7anDDOndvTe495HnJ46tvappSvO+U+GPJmzYx0Dr70Wz5ooFAf+5v7p0GtzPdcLobem54vhueV3KHxPAstUPq8GI3nFJyh0vdsMEN84LhLDd7qxvWe96RcHiVWzAwcHRioG223QoeYpcF/9S95/1gdmE47DVbFg5/M5sbDjeaXRAOX8vdzvcfFp/NyN3cGM6OmT0SDv0Td4b/kPrL/20wH4Szr5t9FA5fNdsQDt832xcOb/NX+Ad2tjce7WA3+w+EYDhyuf/fs3YF+64w2xsuuzJ+ePg7fBvOXpK7XXMoGF6X+1DD2Waz4cQ/7l/iGrS0w7hIO3wVEwgkEEggkEAggUBCSqB3FgZaFlZlfnXL5p7RZdeXuV/pQKGXq/rc2JPVLWupbtkCvUspzeXvmh4IAQIJqYE6s67EqdpsyJZUvOpF/7dQLqt42YfWOj62uuJl9muXDa0vd+fUQKf8Z/ZVeMLerWJV6wtl/HimyNrN++3Pypf93LWv/J1TAx23w5XfdU6Xbah8Uatts71V3NmGXuuv/M7a247aWLl7pwZqu6Pie867r6PiJW1Ll9o97RUvG111ly2vfJn/SVzSWu6+XKQFAgkEEggkEEiYbIG2RoM55nne7FoeSl5tA3Uuzn0v0zxmh24YXP+Yrevu8AMt6vY3zbEr/G/Ma66mgQ6ZtW/0j8H/84l/LG1b/duVFgY6+X0tDy9Q2zNonvfVuDfNs7tzxzHoH4uXC1RfX29hoN6MH7+dITUN9Kn/ap1/DHGgVcEZ5P++xSDQJLhC1vYQuu40W/xFj223bhs2b0OP/wx55VP+pjesu3unWH4mTILPUWwyHUtsUh7UZEIggUACgQQCCQQSCCQQSCCQQCCBQAKBBAIJBBIIJBBIIJBAIIFAAoEEAgkEEggkEEggkEAggUACgQQCCQQSCCQQSPgH7Hht8CQyA1QAAAAASUVORK5CYII=" /><!-- --></p>
<p>Upon filtering the data to remove the outlier, the fit becomes much
better for the linear regression model</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co">#Linear regression plot with filtered outliers in sequin data</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>EBImage<span class="sc">::</span><span class="fu">display</span>(filtered_lm_example)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAC/VBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhIUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxsdHR0eHh4fHx8hISEiIiIjIyMkJCQlJSUmJiYpKSkqKiosLCwtLS0uLi4vLy8wMDAxMTEzMzMzZv80NDQ0Zv81NTU1Z/41aP82NjY2aP44ODg4av45OTk5av45a/06Ojo7Ozs7a/07bP08PDw8bPw9PT1AQEBBQUFCQkJDQ0NEREREcvtFRUVHR0dHdPpISEhIdPhJSUlKSkpMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NTfPdUVFRVVVVXV1dYWFhYgPZZWVlaWlpbW1tcgvVdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVnZ2dnifJoaGhpaWlqampra2tsjfFtbW1tjvBubm5vb29vj/BwcHBxcXFycnJzc3N0dHR1dXV3d3d4eHh5eXl7e3t8fHx8mO1+me2AgICCgoKCnOuDg4OEhISFhYWIiIiKioqNjY2Ojo6Pj4+QkJCRkZGUqeaVlZWWlpaXl5eYmJiYrOWZmZmampqareWbm5udnZ2er+Sfn5+goKChoaGioqKjo6OkpKSmpqanp6entuGoqKiotuGpqamqqqqrueGurq6wsLCxsbGysrKzs7O0tLS1tbW3t7e4uLi5wt67u7u9vb2/v7/AwMDAx9zCwsLCyNvDw8PExMTFytrGxsbGy9rHx8fHzNrIyMjIzNrJycnJzdnKysrLy8vMzMzNzc3Pz8/Q0NDR0dHR0tXR0tfR09fS0tLS09fT1NfU1NTU1dbV1dXV1dbW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///9Qd+ImAAAACXBIWXMAAA7DAAAOwwHHb6hkAAANxUlEQVR4nO2dCZzcVB3HX+nSSqtUpa2AB2opAjKeVVQ8pq3Vek3tgYUqlaOKWkQERcECXiCIVayCaFXECy9KFVA8a5V1kaqrtaYHFDqtSDvbZQu0u732//FdmUkymfxfZl9mk8n/t7tJJvnln/989+XlTSZ5YUCKFBvuBNIuAoSIACEiQIgIECIChMgCoA1Nr7kWdej0VtSvVUucscZxXNfumJnVRxjC2nuaXh1POwzQbjmbeSy7PdbQ9atbiv92LQC6gf8PmUi5NhiUC9geBgOMPeZZ5Dd6po+S8eZLi1iVHQvVsE/QgI6Ai8QmVRR2pDTCJSoOzGHsbBVtnLudiUxZgltlcd60HUBi6oD7mrlBGej/tUhWTl7C54yFXTXnGD58Va1AzK+tMA76btGTYzQgCY9pN/99prsV5hmvEcND+cSp3iW7wZfZk2O+xablBTSSqX/PdD8gXVIUIKY9J4mS5SbMIAwQ7NLLxBouoEsZPLsGaIm7lSogJkKP09u5EXyAGJumt8fOiPGu7QGaLed8ur4EgQeQXu8CNTcK0Md4yarGUoBWrATYB1GA5Ixx7kTFB2iJjnZtvHc9REDi33uDm7LewX2ADmMj5KKJ1QJ1OLC3Ct8hCswCBvWA9rPnugEfZke4JaiGRm0jAEjUQaPYOgFIrPouXYfJV6N0tPvdFFsEKH3bsa6WJL6Yse5WbCcJNQ/opIwrcUABVRKy7A3O6LUT94P4SlIECBEBQkSAEBEgRAQIEQFCRIAQxQM061Y+WLRokVEKdiyZAnQqPMqHJcMU7FgyBQiuPJ0PSj+/UnzQqGRb5yUCSCusDFEJ4urZVwLxs9ooBTuWTAGKl4IdCwFCLAQIsRAgxEKAEAsBQiwECLEQIMRCgBALAUIsBAixECDEQoAQCwFCLAQIsRAgxEKAEAsBQiwECLEQIMRCgBALAUIsBAixZBfQgF+VAVRNWR4PzthpJ26i36yGiUpQ7BTsWAgQYiFAiIUAIRYChFgIEGIhQIiFACEWAoRYCBBiIUCIhQAhFgKEWAgQYiFAiIUAIZZsANp8UxMp2LFkARC7RQyujpuCHUsGAB00CpRjQJ1MKW4KdixZAGTkzDEgM+kUxh6t+qo5VnViU9sOE5adctbFj2BRPMoWoFPlcF5ECjreQRjwdqikpiti1A/w27YFNEd2P/bDwB2HTILTKXwX4EyA77+R+UuQBjRBdcwlAbFJIVvIJKDOWhW99DQ5KvnuWb2+wvjwAi4+WlapnFOpXF2prBZz+e9PuSp6Wo5GViqbxWpPvHV4blbVauk9qwPg2cX2i1EvdDfYxdj27ds3yVlhm85kCYJqr1+/OSjvWR3cERJDp3CWcN4BI9kh9ZV0da97BABY6KYzCiha1zZIoU7tfZjf2mjBmv4GKdSpfQGZdKyXY0Av+oxJoBwDggZHHCwFO5ZsAAJ1KI+Xgh1LJgDxQ/KY+CnYsWQE0JPgrtgp2LFkAxCDEfFTsGPJBCAT5RrQbWwDtYOiTRvQZx/kHdC++CnYsWQCEK+ll8VPwY4lG4AMRIBip2DHkg1AjFE7aKim3AN6PHYKdiyZANRDJ8yGrlwDolOuFky5BnQwrAQN+lUZRNWUZSA4o9dOXCpBiIUAIRaqpBELHeYRi90S1NdHJSjadBlgZ8zyDWgPuyd+CnYsmQBkojwDupndfDMaL8+AqB1kw5RnQMvZ8uU/iZ+CHUsWABkp14BYzzZqKEabrpIPA4+Xgh1LNgDtZzvjp2DHkglAJsoxILpfzIpyDehErhtjp2DHkglA3DSRdjHMdFrcFOxYMgHobezp4XcvRaZgx5IJQFUtFxvaujXszp9cAzplzEXSp+7mpeeshprWyymBqDRrC8DUqT1+VXpQJWVpaiWr96yqu7mhWnjoOatBjWTifnA4p6cHvgiz4T1GKdixZABQ3wqh+CnYsWQAkJlyDOhlgybOHANSFy9cHjsFO5ZMADIRAYqdgh0LAUIsmQD0OF1ANXQTAYqdgh1LNgDRLoYJ70sx14CYgS/vgKjnBcQ0On4KdiyZAAQD32kiBTuWDADqpKOYDeUZ0CQ2adLT4qdgx2IZkONOUEs61OIkA2iwXeogJyFA3PSgYQrWLTYBbUkO0GHwe6MU7FssAtrrJAVo6YG22MWcxAC1RyXtEKDIuE6CgE7gMkghEYstQE6SgNqgBLl8Cu4cm4AMnqqRdkCdLp9EABkp1YCcTpdPQrvYqPHjMU+aAd2nAP2D86E6KES7HAnoD4JPMoDWrcQ9KQbkSEC/kHySAWSi9AJyJKBPKj4JAcpyX64SS+e7NZ+k6iCsJ9f0AlJYXufySQ4Q9tVhSgG5h/cpibak92e2oejyeYuTKCATpRKQy+fDbku6dhOF7XYQdk9vKgG5fL7htqQ9y6w3FLcjnhQCWq/5/MrRgLwLrQJak8k66CHN55+OBuRbSnVQv+bzb9VQdAI9RNEFVO7Hd9VQdIKLc/9h1cfH6axbbhXQHANPygD5+WygixcC8vNJ+uIFUQldIMcb5fCvYZZUAQrySRoQ01dxniVfXRa6y7Ue0HPEYPB4tdIJR544Fl665Jj3Tl659qi51zhOUfGZ78yFCZ64f5mlJ6xX0tvklLrXsAQwc2avX5VeVE1Zdjay8COrHvI5I5hHxWKxi/Pp4ioWj2fa4q71AjVh9Z7VZe5hfgG4gOrU8hKkntTA2Mlipcl+PsV/FQpFUYBm6gedV+My9oCaSKSSvoL/3Qmz4PaQZampgwSXPwVPjyVbB3FPR2YaigLLD4J8kgX0LaNAKQEksHxK89kUFdcioLfL4SbElQ5Agss7A6fnw+Pa3MVGm/hSAWgD5/LqED4JAzK5RDEVgDZxLi/Wp5+xuHn8qPGAbD6/QeBZj8bNIaAtks9HBJ89eNz8Adom+Xy1fvcKj2sZUPpPmD0k+dwdzidhQHexuamvpP8r+fy9AZ9kAXU+38Q2vIDKtdPPhnGt7mLnp/0wv7l6+qdBn7PJV9KproM2VvmYx83TUawT5ZPvEiS+9EL4JA7ozSmug8SXXhifFpyTRjVcgMSXXiifZAGtYe/nwlzDBMjB65/wuDmppKt87o8b1/Z386hnWABxMn8sFHgl/VjsuJbPSafzKMb5fI+Xn06nicu7LALqMwo0DIA4H3n1c/2VCQZxrfa8YKKWAxLX/7xD1j9NxbUJKJW3Q4nTq1PU3RfDDchMLQbkSD6vaHDphkHcNgekDu8vaXTphkHc9gak+HxANw8JkE+9Lp+vuc3n3AOqHiM6+JixfsXnd868omyhqYVna0/4ESUngMSYvfZcR1/9XGQeQCzHgPibvk29YP2wZGrt4ynbLGeqlZjuho4zrL8zqc0BeeQ4XXUf33NfB3nkCEDB0xsEyNV2Aaar7vQPAdJyPFf3DjluGwJqxIcAKbl8ugwu/jGI226AHq2WHw7oILaSQdw2A6Tw/EfsX13BO78IkMvnbln/dNV9/U6AFJ8fqfp5rZ24bQSoR/H5rOKzvvle8HxqH0AKj/Mm9+peAuRXt+Lz8sILFR8C5JejAE0pvFLzIUBe3ecoQIXCa1w+BMgjwaRb8vlQlU/KAQ3ESGHIFkcDKhS+VOOTbkClLmjZc1YPOBpQofBrD590AwIJSBBynP1+VfajimV5cK3SmkLhb3LCXdA7tLiuEgTUmntWy4739IbnA3z6S9DFrQTk8umvLkg3oDgpDM1S1nxUO2hLzUKAlMq6/EhA3tu/CJBSWe9f3cEzrARIqazrn+7gGVYCpFTW9XN38Aw0AVIq6+NXd/ACaAKk5B7fu4MWAiT1rONEHyVOsSgueFG/+qINDyA9J5eAmARULEo0IxjrCAfEmt50ewBa91QWXYIONL3prANSDcWN2XgAm5ESAQQEqKGlrBuIBKiBpawbiASogaWsG4gEqIGlrMcECLEQIMRCgBALAUIsBAixECDEQoAQCwFCLAQIsRAgxEKAEEt2Ac3wa9oMVElZpuMr1VtenziggE5qleU03GKiZYY+a4BmtsryPtxiom8b+qwBalfZAVQSD2MohV+b51pKchDt4X+rIh3ScntkFLWwdHaEZU5pN2D5urICqBQYh+hyzMA1WyxeEGlSmPF0SnBHhGG1sKB9Dim1CpBadHr0mxMLv4xbStGWe6Slrnng08OiJ/qzIi1arQJ0Hm5RC8/DLQDXRTjOVLZtEZZ7+yMWBmSpDhL/VLQOEpbgzXB+DyyEM+Y9Em3ZBgsXzo3ckCxlsxDL3lbWQe0sAoSIACFqHaD/secZbIx5Mzo0sHCp66lqjHxxLwPThz3FVusA6d4YL9Ivfyyer30N7LgU9CHuqppvMR9dz/8mA3xCvVr5eT74uAL0M+74wj61muB5IUAHwKgE026N1KMcuuV7+rrYrpgYoUaHwUerBBlMgNFqYrIYHQ7j1KsO6XjwgCpl50oT/9knFyX2RloH6Jt6czvgGaCvh1qlLhITDwbU10cJEjfx8d4qoMfkAj4oA/xZOzxdLrns2gCQ2NJTeON1Bv848RXx6kLoVG/vaOgXE6eAfvsAg54SBO7sPpjGJ4/TVD7Hpw9pK0Cgi8LJemdioo/PX4qZN7Bt4rWoYe7U5UO5JSAx0SE752Kr3DCPCsdiNg+OYW0EqKEYTK9Oj6rOg/GikjZTh+2UakoDIDi/Nln9jH3dYKwQaIfgzSoVgNIsAoSIACH6P/1Id9IfSa0gAAAAAElFTkSuQmCC" /><!-- --></p>
</div>
<div id="what-if-there-were-no-sequins-added-in-the-study" class="section level2">
<h2>What if there were no sequins added in the study</h2>
<p>In our study, we realized that the sequin-scaled data provided the
best correlation (Spearman correlation coefficient = 0.85), compared to
methods without the use of sequins. However, not all studies would have
access to sequins or could choose not to add sequins. We tested qSIP
methods with absolute concentration of MAGs obtained as a product of
fraction DNA concentration and either with MAG relative abundance (as
performed by <a href="https://journals.asm.org/doi/10.1128/mSystems.00151-20">Greenlon
et al.</a>) or relative coverage (as performed by <a href="https://journals.asm.org/doi/10.1128/mSphere.00085-21">Starr et
al.</a>), or simply with relative coverage, i.e., without converting the
MAG coverages to absolute concentration. We found that without sequins,
simply using relative coverage provided better specificity (0.99) and
good correlation (Spearman correlation coefficient = 0.76), compared to
the other two methods. The user could choose any of the three approaches
to obtain normalized coverages using the equation
<code>coverage_normalization()</code> where the parameter approach would
be chosen accordingly (either “greenlon”, “starr”, or
“relative_coverage”). In this example, simple relative coverage will be
estimated without converting to absolute concentration based on fraction
DNA concentrations. The relative coverage table can then be converted to
a phyloseq object to run incorporator identification and/or AFE
estimation. For more details, please refer to the [incorporator
identification section][<a href="https://zielslab.github.io/SIPmg.github.io/identifying-incorporators.html" class="uri">https://zielslab.github.io/SIPmg.github.io/identifying-incorporators.html</a>]</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>f_tibble <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">&quot;mock_input_data/coverage_metadata.csv&quot;</span>)</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="co">#&gt; Rows: 124 Columns: 61</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a><span class="co">#&gt; Delimiter: &quot;,&quot;</span></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a><span class="co">#&gt; chr  (1): Feature</span></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a><span class="co">#&gt; dbl (60): 12C_rep_1_fraction_10, 12C_rep_1_fraction_1, 12C_rep_1_fraction_2,...</span></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a>rel.cov <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">coverage_normalization</span>(<span class="at">f_tibble =</span> f_tibble, <span class="at">approach =</span> <span class="st">&quot;relative_coverage&quot;</span>)</span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a>mag.table <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(<span class="fu">as.matrix</span>(rel.cov <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;Feature&quot;</span>)), <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>) <span class="co">#Phyloseq OTU table</span></span></code></pre></div>
</div>
<div id="decision-on-the-method-to-scale-coverages" class="section level2">
<h2>Decision on the method to scale coverages</h2>
<p>It can be seen from the above plots that the regression parameters
differ based on the method of regression and whether or not data
filtering is used. These regression parameters directly influence the
estimation of concentrations of features of interest in the microbiome.
Thus, the method for regression must be a careful choice.</p>
</div>
</div>
<div id="sip-metagenomics-and-identification-of-isotope-incorporators" class="section level1">
<h1>SIP metagenomics and identification of isotope incorporators</h1>
<p>The isotope incorporators could be identified using the below
methods:</p>
<ul>
<li>AFE mathematical model by <a href="https://journals.asm.org/doi/10.1128/AEM.02280-15">Hungate et
al. (2015)</a> or
<ul>
<li>Delta buoyant density <span class="math inline">\(AFE_{ΔBD}\)</span>
= <span class="math inline">\(\frac{W_{Lab}-
W_{Light}}{I_{max}}\)</span>. Where <span class="math inline">\(I_{max}\)</span> is the maximum linear shift in
DNA BD (upon 100% labeling), as discussed by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2595786/">Birnie and
Rickwood (1978)</a></li>
<li>MW-HR-SIP <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2018.00570/full">Youngblut
et al. </a></li>
</ul></li>
</ul>
<p>While the first two methods gives a quantitative estimate of isotopic
enrichment in a taxon to determine if it is an incorporator, the third
method relies on differential abundance analysis over multiple
overlapping windows. Please refer to Youngblut et al. and <a href="https://CRAN.R-project.org/package=HTSSIP">HTS-SIP R package</a>
for more details.</p>
<div id="comparison-of-the-methods" class="section level2">
<h2>Comparison of the methods</h2>
<div id="afe-method" class="section level3">
<h3>AFE method</h3>
<p>Isotope incorporators from the introductory vignette were identified
using the AFE method Reproducing the code here, we have</p>
<pre><code>#&gt; Rows: 124 Columns: 61
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;,&quot;
#&gt; chr  (1): Feature
#&gt; dbl (60): 12C_rep_1_fraction_10, 12C_rep_1_fraction_1, 12C_rep_1_fraction_2,...
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#&gt; Rows: 115 Columns: 5
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;,&quot;
#&gt; chr (2): Feature, Sequence
#&gt; dbl (3): Length, GC_content, Concentration
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#&gt; Rows: 60 Columns: 2
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;,&quot;
#&gt; chr (1): Sample
#&gt; dbl (1): Dilution
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#&gt; Rows: 33 Columns: 19
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;\t&quot;
#&gt; chr  (2): user_genome, classification
#&gt; lgl (17): fastani_reference, fastani_reference_radius, fastani_taxonomy, fas...
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#&gt; Rows: 5 Columns: 19
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;\t&quot;
#&gt; chr  (2): user_genome, classification
#&gt; lgl (17): fastani_reference, fastani_reference_radius, fastani_taxonomy, fas...
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#&gt; Rows: 38 Columns: 2
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;,&quot;
#&gt; chr (1): OTU
#&gt; dbl (1): Gi
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#&gt; Rows: 60 Columns: 6
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; Delimiter: &quot;,&quot;
#&gt; chr (2): Isotope, Sample
#&gt; dbl (4): Replicate, Fraction, Buoyant_density, DNA_concentration(ng/uL)
#&gt; 
#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.
#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre><code>#&gt; Warning in dir.create(plot_dir):
#&gt; &#39;C:\Users\psampara\AppData\Local\Temp\RtmpKKb1hw&#39; already exists
#&gt; Warning in file.remove(plot_dir): cannot remove file
#&gt; &#39;C:\Users\psampara\AppData\Local\Temp\RtmpKKb1hw&#39;, reason &#39;Permission denied&#39;</code></pre>
<p>Here, a stricter inference of incorporator based on Delta buoyant
density is made, as a differnece between AFE estimated based on the
delta buoyant density approach and the standard deviation of these
bootstrapped values. However, you could check if such an AFE is positive
or negative to infer isotopic incorporation.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>atomX <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">qSIP_atom_excess_MAGs</span>(phylo.qSIP,</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>                               <span class="at">control_expr=</span><span class="st">&#39;Isotope==&quot;12C&quot;&#39;</span>,</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>                               <span class="at">treatment_rep=</span><span class="st">&#39;Replicate&#39;</span>,<span class="at">isotope =</span> <span class="st">&quot;13C&quot;</span>,</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>                               <span class="at">Gi =</span> GC_content)</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a><span class="co">#Bootstrap confidence intervals</span></span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>df_atomX_boot <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">qSIP_bootstrap_fcr</span>(atomX, <span class="at">n_boot=</span><span class="dv">10</span>, <span class="at">Gi =</span> GC_content, <span class="at">isotope =</span> <span class="st">&quot;13C&quot;</span>, <span class="at">show_delbd_AFE =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a>CI_threshold <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a>df_atomX_boot <span class="ot">=</span> df_atomX_boot <span class="sc">%&gt;%</span></span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Incorporator_qSIP =</span> A_CI_fcr_low <span class="sc">&gt;</span> CI_threshold,</span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a>                <span class="at">Incorporator_delbd =</span> A_delbd <span class="sc">-</span> A_delbd_sd <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a>         <span class="at">OTU =</span> stats<span class="sc">::</span><span class="fu">reorder</span>(OTU, <span class="sc">-</span>A))</span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a></span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a>df_atomX_boot <span class="ot">=</span> df_atomX_boot <span class="sc">%&gt;%</span></span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">inner_join</span>(taxonomy_tibble <span class="sc">%&gt;%</span> </span>
<span id="cb20-15"><a href="#cb20-15" tabindex="-1"></a>               dplyr<span class="sc">::</span><span class="fu">select</span>(user_genome, classification) <span class="sc">%&gt;%</span></span>
<span id="cb20-16"><a href="#cb20-16" tabindex="-1"></a>               dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">OTU =</span> user_genome)) </span>
<span id="cb20-17"><a href="#cb20-17" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(OTU)`</span></span></code></pre></div>
</div>
<div id="mw-hr-sip" class="section level3">
<h3>MW-HR-SIP</h3>
<p>Check incorporators in the overlapping windows of 1.71 - 1.74; 1.72 -
1.75; 1.73 - 1.76. The windows must be chosen in a more judicious manner
that fits the hypothesis of the study.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>windows <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">density_min=</span><span class="fu">c</span>(<span class="fl">1.71</span>,<span class="fl">1.72</span>, <span class="fl">1.73</span>), </span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>                     <span class="at">density_max=</span><span class="fu">c</span>(<span class="fl">1.74</span>,<span class="fl">1.75</span>,<span class="fl">1.76</span>))</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>padj_cutoff <span class="ot">=</span> <span class="fl">0.05</span></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a><span class="co">#ncores = 6</span></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a><span class="co">#doParallel::registerDoParallel(ncores)</span></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a></span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>mw.hr.sip <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">HRSIP</span>(<span class="at">physeq =</span> phylo.qSIP, <span class="at">design =</span> <span class="sc">~</span>Isotope,</span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>                                    <span class="at">density_windows =</span> windows,</span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a>                                    <span class="at">sparsity_threshold =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="fl">0.3</span>, <span class="fl">0.05</span>), </span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a>                                    <span class="at">padj_cutoff =</span> padj_cutoff)</span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0</span></span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a><span class="co">#&gt; Density window:1.71-1.74</span></span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-16"><a href="#cb21-16" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-17"><a href="#cb21-17" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.05</span></span>
<span id="cb21-18"><a href="#cb21-18" tabindex="-1"></a><span class="co">#&gt; Density window:1.71-1.74</span></span>
<span id="cb21-19"><a href="#cb21-19" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-20"><a href="#cb21-20" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-21"><a href="#cb21-21" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-22"><a href="#cb21-22" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.1</span></span>
<span id="cb21-23"><a href="#cb21-23" tabindex="-1"></a><span class="co">#&gt; Density window:1.71-1.74</span></span>
<span id="cb21-24"><a href="#cb21-24" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-25"><a href="#cb21-25" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-26"><a href="#cb21-26" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-27"><a href="#cb21-27" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.15</span></span>
<span id="cb21-28"><a href="#cb21-28" tabindex="-1"></a><span class="co">#&gt; Density window:1.71-1.74</span></span>
<span id="cb21-29"><a href="#cb21-29" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-30"><a href="#cb21-30" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-31"><a href="#cb21-31" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-32"><a href="#cb21-32" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.2</span></span>
<span id="cb21-33"><a href="#cb21-33" tabindex="-1"></a><span class="co">#&gt; Density window:1.71-1.74</span></span>
<span id="cb21-34"><a href="#cb21-34" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-35"><a href="#cb21-35" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-36"><a href="#cb21-36" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-37"><a href="#cb21-37" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.25</span></span>
<span id="cb21-38"><a href="#cb21-38" tabindex="-1"></a><span class="co">#&gt; Density window:1.71-1.74</span></span>
<span id="cb21-39"><a href="#cb21-39" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-40"><a href="#cb21-40" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-41"><a href="#cb21-41" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-42"><a href="#cb21-42" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.3</span></span>
<span id="cb21-43"><a href="#cb21-43" tabindex="-1"></a><span class="co">#&gt; Density window:1.71-1.74</span></span>
<span id="cb21-44"><a href="#cb21-44" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-45"><a href="#cb21-45" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-46"><a href="#cb21-46" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-47"><a href="#cb21-47" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0</span></span>
<span id="cb21-48"><a href="#cb21-48" tabindex="-1"></a><span class="co">#&gt; Density window:1.72-1.75</span></span>
<span id="cb21-49"><a href="#cb21-49" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-50"><a href="#cb21-50" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-51"><a href="#cb21-51" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-52"><a href="#cb21-52" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.05</span></span>
<span id="cb21-53"><a href="#cb21-53" tabindex="-1"></a><span class="co">#&gt; Density window:1.72-1.75</span></span>
<span id="cb21-54"><a href="#cb21-54" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-55"><a href="#cb21-55" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-56"><a href="#cb21-56" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-57"><a href="#cb21-57" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.1</span></span>
<span id="cb21-58"><a href="#cb21-58" tabindex="-1"></a><span class="co">#&gt; Density window:1.72-1.75</span></span>
<span id="cb21-59"><a href="#cb21-59" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-60"><a href="#cb21-60" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-61"><a href="#cb21-61" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-62"><a href="#cb21-62" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.15</span></span>
<span id="cb21-63"><a href="#cb21-63" tabindex="-1"></a><span class="co">#&gt; Density window:1.72-1.75</span></span>
<span id="cb21-64"><a href="#cb21-64" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-65"><a href="#cb21-65" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-66"><a href="#cb21-66" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-67"><a href="#cb21-67" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.2</span></span>
<span id="cb21-68"><a href="#cb21-68" tabindex="-1"></a><span class="co">#&gt; Density window:1.72-1.75</span></span>
<span id="cb21-69"><a href="#cb21-69" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-70"><a href="#cb21-70" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-71"><a href="#cb21-71" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-72"><a href="#cb21-72" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.25</span></span>
<span id="cb21-73"><a href="#cb21-73" tabindex="-1"></a><span class="co">#&gt; Density window:1.72-1.75</span></span>
<span id="cb21-74"><a href="#cb21-74" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-75"><a href="#cb21-75" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-76"><a href="#cb21-76" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-77"><a href="#cb21-77" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.3</span></span>
<span id="cb21-78"><a href="#cb21-78" tabindex="-1"></a><span class="co">#&gt; Density window:1.72-1.75</span></span>
<span id="cb21-79"><a href="#cb21-79" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-80"><a href="#cb21-80" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-81"><a href="#cb21-81" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-82"><a href="#cb21-82" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0</span></span>
<span id="cb21-83"><a href="#cb21-83" tabindex="-1"></a><span class="co">#&gt; Density window:1.73-1.76</span></span>
<span id="cb21-84"><a href="#cb21-84" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-85"><a href="#cb21-85" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-86"><a href="#cb21-86" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-87"><a href="#cb21-87" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.05</span></span>
<span id="cb21-88"><a href="#cb21-88" tabindex="-1"></a><span class="co">#&gt; Density window:1.73-1.76</span></span>
<span id="cb21-89"><a href="#cb21-89" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-90"><a href="#cb21-90" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-91"><a href="#cb21-91" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-92"><a href="#cb21-92" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.1</span></span>
<span id="cb21-93"><a href="#cb21-93" tabindex="-1"></a><span class="co">#&gt; Density window:1.73-1.76</span></span>
<span id="cb21-94"><a href="#cb21-94" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-95"><a href="#cb21-95" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-96"><a href="#cb21-96" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-97"><a href="#cb21-97" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.15</span></span>
<span id="cb21-98"><a href="#cb21-98" tabindex="-1"></a><span class="co">#&gt; Density window:1.73-1.76</span></span>
<span id="cb21-99"><a href="#cb21-99" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-100"><a href="#cb21-100" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-101"><a href="#cb21-101" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-102"><a href="#cb21-102" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.2</span></span>
<span id="cb21-103"><a href="#cb21-103" tabindex="-1"></a><span class="co">#&gt; Density window:1.73-1.76</span></span>
<span id="cb21-104"><a href="#cb21-104" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-105"><a href="#cb21-105" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-106"><a href="#cb21-106" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-107"><a href="#cb21-107" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.25</span></span>
<span id="cb21-108"><a href="#cb21-108" tabindex="-1"></a><span class="co">#&gt; Density window:1.73-1.76</span></span>
<span id="cb21-109"><a href="#cb21-109" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-110"><a href="#cb21-110" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-111"><a href="#cb21-111" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-112"><a href="#cb21-112" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold:0.3</span></span>
<span id="cb21-113"><a href="#cb21-113" tabindex="-1"></a><span class="co">#&gt; Density window:1.73-1.76</span></span>
<span id="cb21-114"><a href="#cb21-114" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb21-115"><a href="#cb21-115" tabindex="-1"></a><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span id="cb21-116"><a href="#cb21-116" tabindex="-1"></a><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span id="cb21-117"><a href="#cb21-117" tabindex="-1"></a><span class="co">#&gt; Sparsity threshold with the most rejected hypotheses:0</span></span>
<span id="cb21-118"><a href="#cb21-118" tabindex="-1"></a></span>
<span id="cb21-119"><a href="#cb21-119" tabindex="-1"></a>mw.hr.sip <span class="ot">=</span> mw.hr.sip <span class="sc">%&gt;%</span></span>
<span id="cb21-120"><a href="#cb21-120" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">incorp =</span> padj <span class="sc">&lt;</span> padj_cutoff)</span></code></pre></div>
<p>Compare number of incorporators</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co">#Get incorporator info</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>qSIP_incorp <span class="ot">=</span> df_atomX_boot <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(OTU, classification, A, Incorporator_qSIP) <span class="sc">%&gt;%</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Incorporator_qSIP <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>classification)</span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>n_qSIP_incorp <span class="ot">=</span> <span class="fu">nrow</span>(qSIP_incorp)</span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a></span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a>delbd_incorp <span class="ot">=</span> df_atomX_boot <span class="sc">%&gt;%</span></span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(OTU, classification, A_delbd, Incorporator_delbd) <span class="sc">%&gt;%</span></span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Incorporator_delbd <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-11"><a href="#cb22-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>classification)</span>
<span id="cb22-12"><a href="#cb22-12" tabindex="-1"></a>n_delbd_incorp <span class="ot">=</span> <span class="fu">nrow</span>(delbd_incorp)</span>
<span id="cb22-13"><a href="#cb22-13" tabindex="-1"></a></span>
<span id="cb22-14"><a href="#cb22-14" tabindex="-1"></a>mw.hr.sip_incorp <span class="ot">=</span> mw.hr.sip <span class="sc">%&gt;%</span></span>
<span id="cb22-15"><a href="#cb22-15" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(OTU, taxa, incorp) <span class="sc">%&gt;%</span></span>
<span id="cb22-16"><a href="#cb22-16" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(incorp <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-17"><a href="#cb22-17" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;Incorporator_mw_hr.sip&quot;</span> <span class="ot">=</span> <span class="st">&quot;incorp&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-18"><a href="#cb22-18" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>taxa)</span>
<span id="cb22-19"><a href="#cb22-19" tabindex="-1"></a>n_mw.hr.sip_incorp <span class="ot">=</span> <span class="fu">nrow</span>(mw.hr.sip_incorp)</span>
<span id="cb22-20"><a href="#cb22-20" tabindex="-1"></a></span>
<span id="cb22-21"><a href="#cb22-21" tabindex="-1"></a>all_incorp_tibble <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(qSIP_incorp, dplyr<span class="sc">::</span><span class="fu">full_join</span>(delbd_incorp, mw.hr.sip_incorp, <span class="at">by =</span> <span class="st">&quot;OTU&quot;</span>), <span class="at">by =</span> <span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb22-22"><a href="#cb22-22" tabindex="-1"></a></span>
<span id="cb22-23"><a href="#cb22-23" tabindex="-1"></a><span class="co">#Print incorporator information</span></span>
<span id="cb22-24"><a href="#cb22-24" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&#39;Number of incorporators detected by qSIP:&#39;</span>, n_qSIP_incorp, <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb22-25"><a href="#cb22-25" tabindex="-1"></a><span class="co">#&gt; Number of incorporators detected by qSIP: 25</span></span>
<span id="cb22-26"><a href="#cb22-26" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&#39;Number of incorporators detected by ΔBD:&#39;</span>, n_delbd_incorp, <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb22-27"><a href="#cb22-27" tabindex="-1"></a><span class="co">#&gt; Number of incorporators detected by ΔBD: 25</span></span>
<span id="cb22-28"><a href="#cb22-28" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&#39;Number of incorporators detected by MW-HR-SIP:&#39;</span>, n_mw.hr.sip_incorp, <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb22-29"><a href="#cb22-29" tabindex="-1"></a><span class="co">#&gt; Number of incorporators detected by MW-HR-SIP: 6</span></span>
<span id="cb22-30"><a href="#cb22-30" tabindex="-1"></a></span>
<span id="cb22-31"><a href="#cb22-31" tabindex="-1"></a>rmarkdown<span class="sc">::</span><span class="fu">paged_table</span>(all_incorp_tibble)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["OTU"],"name":[1],"type":["chr"],"align":["left"]},{"label":["A"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Incorporator_qSIP"],"name":[3],"type":["lgl"],"align":["right"]},{"label":["A_delbd"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Incorporator_delbd"],"name":[5],"type":["lgl"],"align":["right"]},{"label":["Incorporator_mw_hr.sip"],"name":[6],"type":["lgl"],"align":["right"]}],"data":[{"1":"NC_000913.3","2":"0.01559359","3":"TRUE","4":"0.02406311","5":"TRUE","6":"NA"},{"1":"NC_003450.3","2":"0.04847647","3":"TRUE","4":"0.07472437","5":"TRUE","6":"NA"},{"1":"NC_008261.1","2":"0.40987487","3":"TRUE","4":"0.62217710","5":"TRUE","6":"TRUE"},{"1":"NC_010067.1","2":"0.02115236","3":"TRUE","4":"0.03263313","5":"TRUE","6":"NA"},{"1":"NC_014008.1","2":"0.05571908","3":"TRUE","4":"0.08588456","5":"TRUE","6":"NA"},{"1":"NC_014168.1","2":"0.19241466","3":"TRUE","4":"0.29510881","5":"TRUE","6":"NA"},{"1":"NC_014210.1","2":"0.26572526","3":"TRUE","4":"0.40677094","5":"TRUE","6":"NA"},{"1":"NC_014211.1","2":"0.25457996","3":"TRUE","4":"0.38969078","5":"TRUE","6":"NA"},{"1":"NC_014212.1","2":"0.17145305","3":"TRUE","4":"0.26328878","5":"TRUE","6":"NA"},{"1":"NC_014213.1","2":"0.18323983","3":"TRUE","4":"0.28122785","5":"TRUE","6":"NA"},{"1":"NC_014214.1","2":"0.17572222","3":"TRUE","4":"0.26974532","5":"TRUE","6":"NA"},{"1":"NC_014363.1","2":"0.18240538","3":"TRUE","4":"0.27994855","5":"TRUE","6":"NA"},{"1":"NC_015761.1","2":"0.02075433","3":"TRUE","4":"0.03202098","5":"TRUE","6":"NA"},{"1":"NC_017033.1","2":"0.17788410","3":"TRUE","4":"0.27309952","5":"TRUE","6":"NA"},{"1":"NC_018014.1","2":"0.11933315","3":"TRUE","4":"0.18348339","5":"TRUE","6":"NA"},{"1":"NC_019792.1","2":"0.15582689","3":"TRUE","4":"0.23937276","5":"TRUE","6":"NA"},{"1":"NC_019897.1","2":"0.12756736","3":"TRUE","4":"0.19611237","5":"TRUE","6":"NA"},{"1":"NC_019936.1","2":"0.15923443","3":"TRUE","4":"0.24457921","5":"TRUE","6":"NA"},{"1":"NC_019937.1","2":"0.05922982","3":"TRUE","4":"0.09129203","5":"TRUE","6":"NA"},{"1":"NC_019938.1","2":"0.10114853","3":"TRUE","4":"0.15558613","5":"TRUE","6":"NA"},{"1":"NC_019939.1","2":"0.16398224","3":"TRUE","4":"0.25182466","5":"TRUE","6":"NA"},{"1":"NC_019964.1","2":"0.18205895","3":"TRUE","4":"0.27942488","5":"TRUE","6":"NA"},{"1":"NC_019974.1","2":"0.17720421","3":"TRUE","4":"0.27195722","5":"TRUE","6":"NA"},{"1":"NC_019975.1","2":"0.06581470","3":"TRUE","4":"0.10139967","5":"TRUE","6":"NA"},{"1":"NC_019976.1","2":"0.12916152","3":"TRUE","4":"0.19853018","5":"TRUE","6":"NA"},{"1":"NC_002737.2","2":"NA","3":"NA","4":"NA","5":"NA","6":"TRUE"},{"1":"NC_009012.1","2":"NA","3":"NA","4":"NA","5":"NA","6":"TRUE"},{"1":"NC_017095.1","2":"NA","3":"NA","4":"NA","5":"NA","6":"TRUE"},{"1":"NC_018066.1","2":"NA","3":"NA","4":"NA","5":"NA","6":"TRUE"},{"1":"NC_018515.1","2":"NA","3":"NA","4":"NA","5":"NA","6":"TRUE"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>It is evident here that the AFE estimates using qSIP model or the ΔBD
method are not the same, however, both provide the same incorporator
information. Although the incorporators may not be the same everytime,
we found an overlap between qSIP and ΔBD methods in our metagenome
datasets. We also found that qSIP model provided more accurate estimates
of AFE compared to ΔBD method. It also appears that MW-HR-SIP outputs
lower number of incorporators. In our study and by <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2018.00570/full">Youngblut
et al.</a>, MW-HR-SIP was found to have better sensitivity compared to
qSIP or ΔBD methods. It is interesting that the AFE based methods and
differential abundance based methods do not provide the same
incorporator identity, which is perhaps due to comparistive analysis in
only the multiple overlapping windows in the MW-HR-SIP method, while the
AFE methods examine the complete BD gradient. Due to the higher
sensitivity of the MW-HR-SIP method, we attempted a sequential analysis
with first MW-HR-SIP to eliminate likely false positives, and then
perform qSIP to estimate AFE. Such a method reduces multiple comparisons
and increases confidence in detecting true incorporators due to higher
statistical power. We are merely providing the different methods for the
user to decide which method to apply for detecting incorporators in
their study. The choice of the method is to the user’s discretion that
fits best their study. ## Session information</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="co">#&gt; R version 4.3.0 (2023-04-21 ucrt)</span></span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a><span class="co">#&gt; Running under: Windows 10 x64 (build 19045)</span></span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a><span class="co">#&gt; [1] LC_COLLATE=C                    LC_CTYPE=English_Canada.utf8   </span></span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a><span class="co">#&gt; [3] LC_MONETARY=English_Canada.utf8 LC_NUMERIC=C                   </span></span>
<span id="cb23-12"><a href="#cb23-12" tabindex="-1"></a><span class="co">#&gt; [5] LC_TIME=English_Canada.utf8    </span></span>
<span id="cb23-13"><a href="#cb23-13" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-14"><a href="#cb23-14" tabindex="-1"></a><span class="co">#&gt; time zone: America/Vancouver</span></span>
<span id="cb23-15"><a href="#cb23-15" tabindex="-1"></a><span class="co">#&gt; tzcode source: internal</span></span>
<span id="cb23-16"><a href="#cb23-16" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-17"><a href="#cb23-17" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb23-18"><a href="#cb23-18" tabindex="-1"></a><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb23-19"><a href="#cb23-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-20"><a href="#cb23-20" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb23-21"><a href="#cb23-21" tabindex="-1"></a><span class="co">#&gt; [1] ggpubr_0.6.0    ggplot2_3.4.4   HTSSIP_1.4.1    phyloseq_1.44.0</span></span>
<span id="cb23-22"><a href="#cb23-22" tabindex="-1"></a><span class="co">#&gt; [5] SIPmg_2.1      </span></span>
<span id="cb23-23"><a href="#cb23-23" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-24"><a href="#cb23-24" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb23-25"><a href="#cb23-25" tabindex="-1"></a><span class="co">#&gt;   [1] rstudioapi_0.15.0           jsonlite_1.8.5             </span></span>
<span id="cb23-26"><a href="#cb23-26" tabindex="-1"></a><span class="co">#&gt;   [3] magrittr_2.0.3              farver_2.1.1               </span></span>
<span id="cb23-27"><a href="#cb23-27" tabindex="-1"></a><span class="co">#&gt;   [5] rmarkdown_2.25              zlibbioc_1.46.0            </span></span>
<span id="cb23-28"><a href="#cb23-28" tabindex="-1"></a><span class="co">#&gt;   [7] ragg_1.2.6                  vctrs_0.6.2                </span></span>
<span id="cb23-29"><a href="#cb23-29" tabindex="-1"></a><span class="co">#&gt;   [9] multtest_2.56.0             RCurl_1.98-1.12            </span></span>
<span id="cb23-30"><a href="#cb23-30" tabindex="-1"></a><span class="co">#&gt;  [11] rstatix_0.7.2               htmltools_0.5.7            </span></span>
<span id="cb23-31"><a href="#cb23-31" tabindex="-1"></a><span class="co">#&gt;  [13] S4Arrays_1.0.4              polynom_1.4-1              </span></span>
<span id="cb23-32"><a href="#cb23-32" tabindex="-1"></a><span class="co">#&gt;  [15] broom_1.0.5                 Rhdf5lib_1.22.0            </span></span>
<span id="cb23-33"><a href="#cb23-33" tabindex="-1"></a><span class="co">#&gt;  [17] rhdf5_2.44.0                sass_0.4.7                 </span></span>
<span id="cb23-34"><a href="#cb23-34" tabindex="-1"></a><span class="co">#&gt;  [19] bslib_0.5.1                 htmlwidgets_1.6.2          </span></span>
<span id="cb23-35"><a href="#cb23-35" tabindex="-1"></a><span class="co">#&gt;  [21] desc_1.4.2                  plyr_1.8.8                 </span></span>
<span id="cb23-36"><a href="#cb23-36" tabindex="-1"></a><span class="co">#&gt;  [23] cachem_1.0.8                commonmark_1.9.0           </span></span>
<span id="cb23-37"><a href="#cb23-37" tabindex="-1"></a><span class="co">#&gt;  [25] igraph_1.4.3                lifecycle_1.0.4            </span></span>
<span id="cb23-38"><a href="#cb23-38" tabindex="-1"></a><span class="co">#&gt;  [27] iterators_1.0.14            pkgconfig_2.0.3            </span></span>
<span id="cb23-39"><a href="#cb23-39" tabindex="-1"></a><span class="co">#&gt;  [29] Matrix_1.6-0                R6_2.5.1                   </span></span>
<span id="cb23-40"><a href="#cb23-40" tabindex="-1"></a><span class="co">#&gt;  [31] fastmap_1.1.1               GenomeInfoDbData_1.2.10    </span></span>
<span id="cb23-41"><a href="#cb23-41" tabindex="-1"></a><span class="co">#&gt;  [33] MatrixGenerics_1.12.0       digest_0.6.31              </span></span>
<span id="cb23-42"><a href="#cb23-42" tabindex="-1"></a><span class="co">#&gt;  [35] colorspace_2.1-0            S4Vectors_0.38.1           </span></span>
<span id="cb23-43"><a href="#cb23-43" tabindex="-1"></a><span class="co">#&gt;  [37] ps_1.7.5                    DESeq2_1.40.1              </span></span>
<span id="cb23-44"><a href="#cb23-44" tabindex="-1"></a><span class="co">#&gt;  [39] rprojroot_2.0.4             pkgload_1.3.3              </span></span>
<span id="cb23-45"><a href="#cb23-45" tabindex="-1"></a><span class="co">#&gt;  [41] textshaping_0.3.7           GenomicRanges_1.52.0       </span></span>
<span id="cb23-46"><a href="#cb23-46" tabindex="-1"></a><span class="co">#&gt;  [43] vegan_2.6-4                 labeling_0.4.3             </span></span>
<span id="cb23-47"><a href="#cb23-47" tabindex="-1"></a><span class="co">#&gt;  [45] fansi_1.0.4                 abind_1.4-5                </span></span>
<span id="cb23-48"><a href="#cb23-48" tabindex="-1"></a><span class="co">#&gt;  [47] mgcv_1.8-42                 compiler_4.3.0             </span></span>
<span id="cb23-49"><a href="#cb23-49" tabindex="-1"></a><span class="co">#&gt;  [49] bit64_4.0.5                 withr_2.5.2                </span></span>
<span id="cb23-50"><a href="#cb23-50" tabindex="-1"></a><span class="co">#&gt;  [51] backports_1.4.1             tiff_0.1-11                </span></span>
<span id="cb23-51"><a href="#cb23-51" tabindex="-1"></a><span class="co">#&gt;  [53] BiocParallel_1.34.2         carData_3.0-5              </span></span>
<span id="cb23-52"><a href="#cb23-52" tabindex="-1"></a><span class="co">#&gt;  [55] pkgbuild_1.4.2              highr_0.10                 </span></span>
<span id="cb23-53"><a href="#cb23-53" tabindex="-1"></a><span class="co">#&gt;  [57] ggsignif_0.6.4              MASS_7.3-58.4              </span></span>
<span id="cb23-54"><a href="#cb23-54" tabindex="-1"></a><span class="co">#&gt;  [59] DelayedArray_0.26.3         biomformat_1.28.0          </span></span>
<span id="cb23-55"><a href="#cb23-55" tabindex="-1"></a><span class="co">#&gt;  [61] permute_0.9-7               tools_4.3.0                </span></span>
<span id="cb23-56"><a href="#cb23-56" tabindex="-1"></a><span class="co">#&gt;  [63] ape_5.7-1                   glue_1.6.2                 </span></span>
<span id="cb23-57"><a href="#cb23-57" tabindex="-1"></a><span class="co">#&gt;  [65] callr_3.7.3                 EBImage_4.42.0             </span></span>
<span id="cb23-58"><a href="#cb23-58" tabindex="-1"></a><span class="co">#&gt;  [67] nlme_3.1-162                rhdf5filters_1.12.1        </span></span>
<span id="cb23-59"><a href="#cb23-59" tabindex="-1"></a><span class="co">#&gt;  [69] grid_4.3.0                  cluster_2.1.4              </span></span>
<span id="cb23-60"><a href="#cb23-60" tabindex="-1"></a><span class="co">#&gt;  [71] reshape2_1.4.4              ade4_1.7-22                </span></span>
<span id="cb23-61"><a href="#cb23-61" tabindex="-1"></a><span class="co">#&gt;  [73] generics_0.1.3              gtable_0.3.4               </span></span>
<span id="cb23-62"><a href="#cb23-62" tabindex="-1"></a><span class="co">#&gt;  [75] tzdb_0.4.0                  tidyr_1.3.0                </span></span>
<span id="cb23-63"><a href="#cb23-63" tabindex="-1"></a><span class="co">#&gt;  [77] data.table_1.14.8           hms_1.1.3                  </span></span>
<span id="cb23-64"><a href="#cb23-64" tabindex="-1"></a><span class="co">#&gt;  [79] xml2_1.3.5                  car_3.1-2                  </span></span>
<span id="cb23-65"><a href="#cb23-65" tabindex="-1"></a><span class="co">#&gt;  [81] utf8_1.2.3                  XVector_0.40.0             </span></span>
<span id="cb23-66"><a href="#cb23-66" tabindex="-1"></a><span class="co">#&gt;  [83] BiocGenerics_0.46.0         foreach_1.5.2              </span></span>
<span id="cb23-67"><a href="#cb23-67" tabindex="-1"></a><span class="co">#&gt;  [85] pillar_1.9.0                stringr_1.5.0              </span></span>
<span id="cb23-68"><a href="#cb23-68" tabindex="-1"></a><span class="co">#&gt;  [87] vroom_1.6.4                 splines_4.3.0              </span></span>
<span id="cb23-69"><a href="#cb23-69" tabindex="-1"></a><span class="co">#&gt;  [89] dplyr_1.1.2                 lattice_0.21-8             </span></span>
<span id="cb23-70"><a href="#cb23-70" tabindex="-1"></a><span class="co">#&gt;  [91] survival_3.5-5              bit_4.0.5                  </span></span>
<span id="cb23-71"><a href="#cb23-71" tabindex="-1"></a><span class="co">#&gt;  [93] tidyselect_1.2.0            locfit_1.5-9.8             </span></span>
<span id="cb23-72"><a href="#cb23-72" tabindex="-1"></a><span class="co">#&gt;  [95] Biostrings_2.68.1           knitr_1.45                 </span></span>
<span id="cb23-73"><a href="#cb23-73" tabindex="-1"></a><span class="co">#&gt;  [97] IRanges_2.34.0              SummarizedExperiment_1.30.2</span></span>
<span id="cb23-74"><a href="#cb23-74" tabindex="-1"></a><span class="co">#&gt;  [99] stats4_4.3.0                xfun_0.41                  </span></span>
<span id="cb23-75"><a href="#cb23-75" tabindex="-1"></a><span class="co">#&gt; [101] Biobase_2.60.0              matrixStats_1.1.0          </span></span>
<span id="cb23-76"><a href="#cb23-76" tabindex="-1"></a><span class="co">#&gt; [103] fftwtools_0.9-11            stringi_1.7.12             </span></span>
<span id="cb23-77"><a href="#cb23-77" tabindex="-1"></a><span class="co">#&gt; [105] lazyeval_0.2.2              yaml_2.3.7                 </span></span>
<span id="cb23-78"><a href="#cb23-78" tabindex="-1"></a><span class="co">#&gt; [107] evaluate_0.23               codetools_0.2-19           </span></span>
<span id="cb23-79"><a href="#cb23-79" tabindex="-1"></a><span class="co">#&gt; [109] tibble_3.2.1                cli_3.6.1                  </span></span>
<span id="cb23-80"><a href="#cb23-80" tabindex="-1"></a><span class="co">#&gt; [111] systemfonts_1.0.5           munsell_0.5.0              </span></span>
<span id="cb23-81"><a href="#cb23-81" tabindex="-1"></a><span class="co">#&gt; [113] processx_3.8.2              jquerylib_0.1.4            </span></span>
<span id="cb23-82"><a href="#cb23-82" tabindex="-1"></a><span class="co">#&gt; [115] roxygen2_7.2.3              Rcpp_1.0.10                </span></span>
<span id="cb23-83"><a href="#cb23-83" tabindex="-1"></a><span class="co">#&gt; [117] GenomeInfoDb_1.36.0         png_0.1-8                  </span></span>
<span id="cb23-84"><a href="#cb23-84" tabindex="-1"></a><span class="co">#&gt; [119] parallel_4.3.0              readr_2.1.4                </span></span>
<span id="cb23-85"><a href="#cb23-85" tabindex="-1"></a><span class="co">#&gt; [121] prettyunits_1.2.0           jpeg_0.1-10                </span></span>
<span id="cb23-86"><a href="#cb23-86" tabindex="-1"></a><span class="co">#&gt; [123] bitops_1.0-7                scales_1.2.1               </span></span>
<span id="cb23-87"><a href="#cb23-87" tabindex="-1"></a><span class="co">#&gt; [125] purrr_1.0.1                 crayon_1.5.2               </span></span>
<span id="cb23-88"><a href="#cb23-88" tabindex="-1"></a><span class="co">#&gt; [127] rlang_1.1.1</span></span></code></pre></div>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
