\nonstopmode{}
\documentclass[a4paper]{book}
\usepackage[times,inconsolata,hyper]{Rd}
\usepackage{makeidx}
\usepackage[utf8]{inputenc} % @SET ENCODING@
% \usepackage{graphicx} % @USE GRAPHICX@
\makeindex{}
\begin{document}
\chapter*{}
\begin{center}
{\textbf{\huge Package `SIPmg'}}
\par\bigskip{\large \today}
\end{center}
\inputencoding{utf8}
\ifthenelse{\boolean{Rd@use@hyper}}{\hypersetup{pdftitle = {SIPmg: Statistical Analysis to Identify Isotope Incorporating MAGs}}}{}
\ifthenelse{\boolean{Rd@use@hyper}}{\hypersetup{pdfauthor = {Pranav Sampara; Ryan Ziels}}}{}
\begin{description}
\raggedright{}
\item[Title]\AsIs{Statistical Analysis to Identify Isotope Incorporating MAGs}
\item[Version]\AsIs{1.4.1}
\item[Description]\AsIs{Statistical analysis as part of a stable isotope probing (SIP) metagenomics study to identify isotope incorporating taxa recovered as metagenome-assembled genomes (MAGs). 
Helpful reading and a vignette in bookdown format is provided on the package site <}\url{https://zielslab.github.io/SIPmg.github.io/}\AsIs{>.}
\item[URL]\AsIs{}\url{https://zielslab.github.io/SIPmg.github.io/}\AsIs{}
\item[BugReports]\AsIs{}\url{https://github.com/ZielsLab/SIPmg}\AsIs{}
\item[License]\AsIs{GPL-2}
\item[Encoding]\AsIs{UTF-8}
\item[Roxygen]\AsIs{list(markdown = TRUE)}
\item[RoxygenNote]\AsIs{7.2.3}
\item[Imports]\AsIs{HTSSIP, dplyr, lazyeval, phyloseq, plyr, stringr, tibble,
tidyr, magrittr, ggplot2, ggpubr, purrr, rlang, MASS, DESeq2,
data.table, utils}
\item[VignetteBuilder]\AsIs{knitr}
\item[NeedsCompilation]\AsIs{yes}
\item[Depends]\AsIs{R (>= 3.5.0)}
\item[Suggests]\AsIs{rmarkdown, knitr, EBImage, readr, BiocManager}
\item[LazyData]\AsIs{true}
\item[Author]\AsIs{Pranav Sampara [aut, cre],
Kate Waring [ctb],
Ryan Ziels [aut]}
\item[Maintainer]\AsIs{Pranav Sampara }\email{pranav.sai.4@gmail.com}\AsIs{}
\item[ExperimentalWindowsRuntime]\AsIs{ucrt}
\end{description}
\Rdcontents{\R{} topics documented:}
\inputencoding{utf8}
\HeaderA{atomX}{Atom fraction excess table}{atomX}
\keyword{datasets}{atomX}
%
\begin{Description}\relax
Data table generated from the "qSIP\_atom\_excess\_MAGs" function
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(atomX)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "list"
\end{Format}
\inputencoding{utf8}
\HeaderA{calc\_atom\_excess\_MAGs}{Calculate atom fraction excess}{calc.Rul.atom.Rul.excess.Rul.MAGs}
%
\begin{Description}\relax
See Hungate et al., 2015 for more details
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_atom_excess_MAGs(Mlab, Mlight, Mheavymax, isotope = "13C")
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{Mlab}] The molecular wight of labeled DNA

\item[\code{Mlight}] The molecular wight of unlabeled DNA

\item[\code{Mheavymax}] The theoretical maximum molecular weight of fully-labeled DNA

\item[\code{isotope}] The isotope for which the DNA is labeled with ('13C' or '18O')
\end{ldescription}
\end{Arguments}
%
\begin{Value}
numeric value: atom fraction excess (A)
\end{Value}
\inputencoding{utf8}
\HeaderA{calc\_Mheavymax\_MAGs}{Calculate Mheavymax}{calc.Rul.Mheavymax.Rul.MAGs}
%
\begin{Description}\relax
This script was adapted from https://github.com/buckleylab/HTSSIP/blob/master/R/qSIP\_atom\_excess.R
for use with genome-centric metagenomics. See Hungate et al., 2015 for more details
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_Mheavymax_MAGs(Mlight, isotope = "13C", Gi = Gi)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{Mlight}] The molecular wight of unlabeled DNA

\item[\code{isotope}] The isotope for which the DNA is labeled with ('13C' or '18O')

\item[\code{Gi}] The G+C content of unlabeled DNA
\end{ldescription}
\end{Arguments}
%
\begin{Value}
numeric value: maximum molecular weight of fully-labeled DNA
\end{Value}
\inputencoding{utf8}
\HeaderA{coverage\_normalization}{Normalize feature coverages to estimate absolute abundance or relative coverage using MAG/contig coverage values with or without multiplying total DNA concentration of the fraction}{coverage.Rul.normalization}
%
\begin{Description}\relax
Normalize feature coverages to estimate absolute abundance or relative coverage using MAG/contig coverage values with or without multiplying total DNA concentration of the fraction
\end{Description}
%
\begin{Usage}
\begin{verbatim}
coverage_normalization(
  f_tibble,
  contig_coverage,
  sequencing_yield,
  fractions_df,
  approach = "relative_coverage"
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{f\_tibble}] Can be either of
(1) a tibble with first column "Feature" that contains bin IDs, and the rest of the columns represent samples with bins' coverage values.
(2) a tibble as outputted by the program "checkm coverage" from the tool CheckM. Please check CheckM documentation - https://github.com/Ecogenomics/CheckM on the usage for "checkm coverage" program

\item[\code{contig\_coverage}] tibble with contig ID names ("Feature" column), sample columns with \strong{same sample names as in f\_tibble} containing coverage values of each contig, contig length in bp ("contig\_length" column), and the MAG the contig is associated ("MAG" column) \strong{with same MAGs as in Feature column of f\_tibble dataset}.

\item[\code{sequencing\_yield}] tibble containing sample ID ("sample" column) with \strong{same sample names as in f\_tibble} and number of reads in bp recovered in that sample ("yield" column).

\item[\code{fractions\_df}] fractions data frame
A fractions file with the following columns
\begin{itemize}

\item{} Replicate: Depends on how many replicates the study has
\item{} Fractions: Typically in the range of 2-24
\item{} Buoyant\_density: As calculated from the refractometer for each fraction and replicate
\item{} Isotope: "12C", "13C", "14N", "15N" etc.
\item{} DNA\_concentration
\item{} Sample: In the format "'isotope'\emph{rep}\#\emph{fraction}\#".
For instance, "12C\_rep\_1\_fraction\_1"

\end{itemize}


\item[\code{approach}] Please choose the method for coverage normalization as "relative\_coverage", "greenlon", "starr" to estimate only relative coverage without multiplying DNA concentration of fraction, or as per methods in Greenlon et al. - https://journals.asm.org/doi/full/10.1128/msystems.00417-22 or Starr et al. - https://journals.asm.org/doi/10.1128/mSphere.00085-21
\end{ldescription}
\end{Arguments}
%
\begin{Value}
tibble containing normalized coverage in required format with MAG name as first column and the normalized coverage values in each sample as the rest of the columns.
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

data(f_tibble)

rel.cov = coverage_normalization(f_tibble)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{DESeq2\_l2fc}{Calculating log2 fold change for HTS-SIP data.}{DESeq2.Rul.l2fc}
%
\begin{Description}\relax
The 'use\_geo\_mean' parameter uses geometric means on all non-zero abundances
for estimateSizeFactors instead of using the default log-tranformed geometric means.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
DESeq2_l2fc(
  physeq,
  density_min,
  density_max,
  design,
  l2fc_threshold = 0.25,
  sparsity_threshold = 0.25,
  sparsity_apply = "all",
  size_factors = "geoMean"
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{physeq}] Phyloseq object

\item[\code{density\_min}] Minimum buoyant density of the 'heavy' gradient fractions

\item[\code{density\_max}] Maximum buoyant density of the 'heavy' gradient fractions

\item[\code{design}] \code{design} parameter used for DESeq2 analysis.
See \code{DESeq2::DESeq} for more details.

\item[\code{l2fc\_threshold}] log2 fold change (l2fc) values must be significantly above this
threshold in order to reject the hypothesis of equal counts.

\item[\code{sparsity\_threshold}] All OTUs observed in less than this portion (fraction: 0-1)
of gradient fraction samples are pruned. A a form of indepedent filtering,
The sparsity cutoff with the most rejected hypotheses is used.

\item[\code{sparsity\_apply}] Apply sparsity threshold to all gradient fraction samples ('all')
or just heavy fraction samples ('heavy')

\item[\code{size\_factors}] Method of estimating size factors.
'geoMean' is from (Pepe-Ranney et. al., 2016) and removes all zero-abundances from the calculation.
'default' is the default for estimateSizeFactors.
'iterate' is an alternative when every OTU has a zero in >=1 sample.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
dataframe of HRSIP results
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(phylo.qSIP)


df_l2fc = DESeq2_l2fc(phylo.qSIP, density_min=1.71, density_max=1.75, design=~Isotope)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{df\_atomX\_boot}{Bootstrapped atom fraction excess table}{df.Rul.atomX.Rul.boot}
\keyword{datasets}{df\_atomX\_boot}
%
\begin{Description}\relax
Data table generated from bostrapping the AFE table using the "qSIP\_bootstrap\_fcr" function
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(df_atomX_boot)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "data.frame"
\end{Format}
\inputencoding{utf8}
\HeaderA{filter\_l2fc}{Filter l2fc table}{filter.Rul.l2fc}
%
\begin{Description}\relax
\code{filter\_l2fc} filters a l2fc table to 'best' sparsity cutoffs \&
bouyant density windows.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
filter_l2fc(df_l2fc, padj_cutoff = 0.1)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df\_l2fc}] data.frame of log2 fold change values

\item[\code{padj\_cutoff}] Adjusted p-value cutoff for rejecting the null hypothesis
that l2fc values were not greater than the l2fc\_threshold.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
filtered df\_l2fc object
\end{Value}
\inputencoding{utf8}
\HeaderA{filter\_na}{Remove MAGs with NAs from atomX table}{filter.Rul.na}
%
\begin{Description}\relax
This function enables removing NAs from the atomX table.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
filter_na(atomX)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{atomX}] A list object created by \code{qSIP\_atom\_excess\_MAGs()}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A list of 2 data.frame objects without MAGs which have NAs. 'W' contains the weighted mean buoyant density (W) values for each OTU in each treatment/control. 'A' contains the atom fraction excess values for each OTU. For the 'A' table, the 'Z' column is buoyant density shift, and the 'A' column is atom fraction excess.
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(atomX)



### Remove NAs in atomX table
atomx_no_na = filter_na(atomX)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{fractions}{Fractions table}{fractions}
\keyword{datasets}{fractions}
%
\begin{Description}\relax
Fractions data used for many functions in the package
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(fractions)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "data.frame"
\end{Format}
\inputencoding{utf8}
\HeaderA{f\_tibble}{Coverage table}{f.Rul.tibble}
\keyword{datasets}{f\_tibble}
%
\begin{Description}\relax
Coverage data used for many functions in the package
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(f_tibble)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "data.frame"
\end{Format}
\inputencoding{utf8}
\HeaderA{GC\_content}{GC\_content table}{GC.Rul.content}
\keyword{datasets}{GC\_content}
%
\begin{Description}\relax
GC\_content data
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(GC_content)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "data.frame"
\end{Format}
\inputencoding{utf8}
\HeaderA{HRSIP}{(MW-)HR-SIP analysis}{HRSIP}
%
\begin{Description}\relax
Conduct (multi-window) high resolution stable isotope probing (HR-SIP) analysis.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
HRSIP(
  physeq,
  design,
  density_windows = data.frame(density_min = c(1.7), density_max = c(1.75)),
  sparsity_threshold = seq(0, 0.3, 0.1),
  sparsity_apply = "all",
  l2fc_threshold = 0.25,
  padj_method = "BH",
  padj_cutoff = NULL,
  parallel = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{physeq}] Phyloseq object

\item[\code{design}] \code{design} parameter used for DESeq2 analysis.
This is usually used to differentiate labeled-treatment and unlabeld-control samples.
See \code{DESeq2::DESeq} for more details on the option.

\item[\code{density\_windows}] The buoyant density window(s) used for for calculating log2
fold change values. Input can be a vector (length 2) or a data.frame with a 'density\_min'
and a 'density\_max' column (each row designates a density window).

\item[\code{sparsity\_threshold}] All OTUs observed in less than this portion (fraction: 0-1)
of gradient fraction samples are pruned. This is a form of indepedent filtering.
The sparsity cutoff with the most rejected hypotheses is used.

\item[\code{sparsity\_apply}] Apply sparsity threshold to all gradient fraction samples ('all')
or just 'heavy' fraction samples ('heavy'), where 'heavy' samples are designated
by the \code{density\_windows}.

\item[\code{l2fc\_threshold}] log2 fold change (l2fc) values must be significantly above this
threshold in order to reject the hypothesis of equal counts.
See \code{DESeq2} for more information.

\item[\code{padj\_method}] Method for global p-value adjustment (See \code{p.adjust()}).

\item[\code{padj\_cutoff}] Adjusted p-value cutoff for rejecting the null hypothesis
that l2fc values were not greater than the l2fc\_threshold.
Set to \code{NULL} to skip filtering of results to the sparsity cutoff with most
rejected hypotheses and filtering each OTU to the buoyant density window with the
greatest log2 fold change.

\item[\code{parallel}] Process each parameter combination in parallel.
See \code{plyr::mdply()} for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
The (MW-)HR-SIP workflow is as follows:

\begin{enumerate}

\item{} For each sparsity threshold \& BD window: calculate log2 fold change values (with DESeq2) for each OTU
\item{} Globally adjust p-values with a user-defined method (see p.adjust())
\item{} Select the sparsity cutoff with the most rejected hypotheses
\item{} For each OTU, select the BD window with the greatest log2 fold change value

\end{enumerate}

\end{Details}
%
\begin{Value}
dataframe of HRSIP results
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(phylo.qSIP)


## HR-SIP
### Note: treatment-control samples differentiated with 'design=~Isotope'
df_l2fc = HRSIP(phylo.qSIP, design=~Isotope)
## Same, but multiple BD windows (MW-HR-SIP). For parallel processing change to parallel = TRUE
### Windows = 1.7-1.74, 1.72-1.75, and 1.73 - 1.76
windows = data.frame(density_min=c(1.71,1.72, 1.73), density_max=c(1.74,1.75,1.76))
df_l2fc = HRSIP(phylo.qSIP,
                design=~Isotope,
                density_windows = windows,
                padj_cutoff = 0.05,
                parallel=FALSE)



\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{incorporators\_taxonomy}{Isotope incorporator list with GTDB taxonomy}{incorporators.Rul.taxonomy}
%
\begin{Description}\relax
This function provides a table with MAGs and their corresponding GTDB taxonomy
as an output. This would be useful in identifying the taxa that have incorporation
\end{Description}
%
\begin{Usage}
\begin{verbatim}
incorporators_taxonomy(taxonomy, bootstrapped_AFE_table)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{taxonomy}] A taxonomy tibble obtained in the markdown. This taxonomy tibble is
typically a concatenated list of archaeal and bacterial taxonomy from GTDB-Tk
Please check GTDB-Tk documentation for running the tool

\item[\code{bootstrapped\_AFE\_table}] A data frame indicating bootstrapped atom fraction excess values
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A tibble with two columns, OTU and Taxonomy, with taxonomy of the incorporator MAGs
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(taxonomy_tibble,df_atomX_boot)



### Making incorporator taxonomy list
incorporator_list = incorporators_taxonomy(taxonomy = taxonomy_tibble,
         bootstrapped_AFE_table = df_atomX_boot)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{mag.table}{MAG abundance table in phyloseq format}{mag.table}
\keyword{datasets}{mag.table}
%
\begin{Description}\relax
MAG abundances in the format of phyloseq object to be used in the qSIP and (MW-)HR-SIP pipeline
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(mag.table)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "phyloseq"
\end{Format}
\inputencoding{utf8}
\HeaderA{phylo.qSIP}{Master phyloseq object}{phylo.qSIP}
\keyword{datasets}{phylo.qSIP}
%
\begin{Description}\relax
Master phyloseq object
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(phylo.qSIP)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "phyloseq"
\end{Format}
\inputencoding{utf8}
\HeaderA{phylo.table}{Master phyloseq object using the MAG phyloseq objects}{phylo.table}
%
\begin{Description}\relax
Creates a phyloseq-style object using processed phyloseq objects for otu
table (here, MAG table), taxa table, and sample table
\end{Description}
%
\begin{Usage}
\begin{verbatim}
phylo.table(mag, taxa, samples)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{mag}] phyloseq-styled MAG table

\item[\code{taxa}] phyloseq-styled taxa table

\item[\code{samples}] sample information table
\end{ldescription}
\end{Arguments}
%
\begin{Value}
phyloseq object for MAGs
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

data(mag.table,taxonomy.object,samples.object,fractions,taxonomy_tibble)
###Making phyloseq table from fractions metadata
samples.object = sample.table(fractions)
taxonomy.object = tax.table(taxonomy_tibble)




### Making master phyloseq table from scaled MAG data, taxa and fractions phyloseq data
phylo.qSIP = phylo.table(mag.table,taxonomy.object,samples.object)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{qSIP\_atom\_excess\_format\_MAGs}{Reformat a phyloseq object of qSIP\_atom\_excess\_MAGs analysis}{qSIP.Rul.atom.Rul.excess.Rul.format.Rul.MAGs}
%
\begin{Description}\relax
Reformat a phyloseq object of qSIP\_atom\_excess\_MAGs analysis
\end{Description}
%
\begin{Usage}
\begin{verbatim}
qSIP_atom_excess_format_MAGs(physeq, control_expr, treatment_rep)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{physeq}] A phyloseq object

\item[\code{control\_expr}] An expression for identifying unlabeled control
samples in the phyloseq object (eg., "Substrate=='12C-Con'")

\item[\code{treatment\_rep}] Which column in the phyloseq sample data designates
replicate treatments
\end{ldescription}
\end{Arguments}
%
\begin{Value}
numeric value: atom fraction excess (A)
\end{Value}
\inputencoding{utf8}
\HeaderA{qSIP\_atom\_excess\_MAGs}{Calculate atom fraction excess using q-SIP method}{qSIP.Rul.atom.Rul.excess.Rul.MAGs}
%
\begin{Description}\relax
Calculate atom fraction excess using q-SIP method
\end{Description}
%
\begin{Usage}
\begin{verbatim}
qSIP_atom_excess_MAGs(
  physeq,
  control_expr,
  treatment_rep = NULL,
  isotope = "13C",
  df_OTU_W = NULL,
  Gi
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{physeq}] A phyloseq object

\item[\code{control\_expr}] Expression used to identify control samples based on sample\_data.

\item[\code{treatment\_rep}] Which column in the phyloseq sample data designates
replicate treatments

\item[\code{isotope}] The isotope for which the DNA is labeled with ('13C' or '18O')

\item[\code{df\_OTU\_W}] Keep NULL

\item[\code{Gi}] GC content of the MAG
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A list of 2 data.frame objects. 'W' contains the weighted mean buoyant density (W) values for each OTU in each treatment/control. 'A' contains the atom fraction excess values for each OTU. For the 'A' table, the 'Z' column is buoyant density shift, and the 'A' column is atom fraction excess.
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(phylo.qSIP,GC_content)
### Making atomx table

## Not run::
### BD shift (Z) & atom excess (A)
atomX = qSIP_atom_excess_MAGs(phylo.qSIP,
                         control_expr='Isotope=="12C"',
                         treatment_rep='Replicate',
                         Gi = GC_content)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{qSIP\_bootstrap\_fcr}{Calculate adjusted bootstrap CI after for multiple testing for atom fraction excess using q-SIP method. Multiple hypothesis tests are corrected by}{qSIP.Rul.bootstrap.Rul.fcr}
%
\begin{Description}\relax
Calculate adjusted bootstrap CI after for multiple testing for atom fraction excess using q-SIP method. Multiple hypothesis tests are corrected by
\end{Description}
%
\begin{Usage}
\begin{verbatim}
qSIP_bootstrap_fcr(
  atomX,
  isotope = "13C",
  n_sample = c(3, 3),
  ci_adjust_method = "fcr",
  n_boot = 10,
  parallel = FALSE,
  a = 0.1
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{atomX}] A list object created by \code{qSIP\_atom\_excess\_MAGs()}

\item[\code{isotope}] The isotope for which the DNA is labeled with ('13C' or '18O')

\item[\code{n\_sample}] A vector of length 2. The sample size for data resampling (with replacement) for 1) control samples and 2) treatment samples.

\item[\code{ci\_adjust\_method}] Confidence interval adjustment method. Please choose 'FCR', 'Bonferroni', or 'none' (if no adjustment is needed). Default is FCR and also provides unadjusted CI.

\item[\code{n\_boot}] Number of bootstrap replicates.

\item[\code{parallel}] Parallel processing. See \code{.parallel} option in \code{dplyr::mdply()} for more details.

\item[\code{a}] A numeric value. The alpha for calculating confidence intervals.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.frame of atom fraction excess values (A) and atom fraction excess confidence intervals adjusted for multiple testing.
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(phylo.qSIP,GC_content)

### BD shift (Z) & atom excess (A)
atomX = qSIP_atom_excess_MAGs(phylo.qSIP,
                        control_expr='Isotope=="12C"',
                        treatment_rep='Replicate', Gi = GC_content)

### Add doParallel::registerDoParallel(num_cores) if parallel bootstrapping is to be done
df_atomX_boot = qSIP_bootstrap_fcr(atomX, n_boot=5, parallel = FALSE)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sample.table}{phyloseq-styled sample table}{sample.table}
%
\begin{Description}\relax
Creates a phyloseq-styled sample table from fractions metadata
containing data on fraction number, number of replicates, buoyant density
calculated from a refractometer, type of isotope, and DNA concentration
of each fraction, and isotope type. See below for information on "fractions" file.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
sample.table(fractions_df)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{fractions\_df}] fractions data frame
A fractions file with the following columns
\begin{itemize}

\item{} Replicate: Depends on how many replicates the study has
\item{} Fractions: Typically in the range of 2-24
\item{} Buoyant\_density: As calculated from the refractometer for each fraction and replicate
\item{} Isotope: "12C", "13C", "14N", "15N" etc.
\item{} DNA\_concentration
\item{} Sample: In the format "'isotope'\emph{rep}\#\emph{fraction}\#".
For instance, "12C\_rep\_1\_fraction\_1"

\end{itemize}

\end{ldescription}
\end{Arguments}
%
\begin{Value}
data  frame: phyloseq-style sample table
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(fractions)



### Making phyloseq table from fractions metadata
samples.object = sample.table(fractions)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{samples.object}{Fractions table in phyloseq format}{samples.object}
\keyword{datasets}{samples.object}
%
\begin{Description}\relax
Fractions metadata in the format of phyloseq object to be used in the qSIP and (MW-)HR-SIP pipeline
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(samples.object)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "phyloseq"
\end{Format}
\inputencoding{utf8}
\HeaderA{scale\_features\_lm}{Scale feature coverage values to estimate their absolute abundance}{scale.Rul.features.Rul.lm}
%
\begin{Description}\relax
Calculates global scaling factors for features (contigs or bins),based on linear regression of sequin coverage. Options include log-transformations of coverage, as well as filtering features based on limit of detection. This function must be called first, before the feature abundance table, feature detection table, and plots are retrieved.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
scale_features_lm(
  f_tibble,
  sequin_meta,
  seq_dilution,
  log_trans = TRUE,
  coe_of_variation = 250,
  lod_limit = 0,
  save_plots = TRUE,
  plot_dir = tempdir(),
  cook_filtering = TRUE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{f\_tibble}] Can be either of
(1) a tibble with first column "Feature" that contains bin IDs, and the rest of the columns represent samples with bins' coverage values.
(2) a tibble as outputted by the program "checkm coverage" from the tool CheckM. Please check CheckM documentation - https://github.com/Ecogenomics/CheckM on the usage for "checkm coverage" program

\item[\code{sequin\_meta}] tibble containing sequin names ("Feature column") and concentrations in attamoles/uL ("Concentration") column.

\item[\code{seq\_dilution}] tibble with first column "Sample" with \strong{same sample names as in f\_tibble}, and a second column "Dilution" showing ratio of sequins added to final sample volume (e.g. a value of 0.01 for a dilution of 1 volume sequin to 99 volumes sample)

\item[\code{log\_trans}] Boolean (TRUE or FALSE), should coverages and sequin concentrations be log-scaled?

\item[\code{coe\_of\_variation}] Acceptable coefficient of variation for coverage and detection (eg. 20 - for 20 \% threshold of coefficient of variation). Coverages above the threshold value will be flagged in the plots.

\item[\code{lod\_limit}] (Decimal range 0-1) Threshold for the percentage of minimum detected sequins per concentration group. Default = 0

\item[\code{save\_plots}] Boolean (TRUE or FALSE), should sequin scaling be saved? Default = TRUE

\item[\code{plot\_dir}] Directory where plots are to be saved. Will create a directory "sequin\_scaling\_plots\_lm" if it does not exist.

\item[\code{cook\_filtering}] Boolean (TRUE or FALSE), should data points be filtered based on Cook's distance metric. Cooks distance can be useful in detecting influential outliers in an ordinary least square’s regression model, which can negatively influence the model. A threshold of Cooks distance of 4/n (where n is the sample size) is chosen, and any data point with Cooks distance > 4/n is filtered out. It is typical to choose 4/n as the threshold in detecting the outliers in the data. Default = TRUE
\end{ldescription}
\end{Arguments}
%
\begin{Value}
a list of tibbles containing
\begin{itemize}

\item{} mag\_tab: a tibble with first column "Feature" that contains bin (or contig IDs), and the rest of the columns represent samples with features' scaled abundances (attamoles/uL)
\item{} mag\_det: a tibble with first column "Feature" that contains bin (or contig IDs),
\item{} plots: linear regression plots for scaling MAG coverage values to absolute abundance
\item{} scale\_fac: a master tibble with all of the intermediate values in above calculations

\end{itemize}

\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(f_tibble, sequins, seq_dil)



### scaling sequins from coverage values
scaled_features_lm = scale_features_lm(f_tibble,sequin_meta, seq_dil)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{scale\_features\_rlm}{Scale feature coverage values to estimate their absolute abundance}{scale.Rul.features.Rul.rlm}
%
\begin{Description}\relax
Calculates global scaling factors for features (contigs or bins),based on linear regression of sequin coverage. Options include log-transformations of coverage, as well as filtering features based on limit of detection. This function must be called first, before the feature abundance table, feature detection table, and plots are retrieved.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
scale_features_rlm(
  f_tibble,
  sequin_meta,
  seq_dilution,
  log_trans = TRUE,
  coe_of_variation = 250,
  lod_limit = 0,
  save_plots = TRUE,
  plot_dir = tempdir()
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{f\_tibble}] Can be either of
(1) a tibble with first column "Feature" that contains bin IDs, and the rest of the columns represent samples with bins' coverage values.
(2) a tibble as outputted by the program "checkm coverage" from the tool CheckM. Please check CheckM documentation - https://github.com/Ecogenomics/CheckM on the usage for "checkm coverage" program

\item[\code{sequin\_meta}] tibble containing sequin names ("Feature column") and concentrations in attamoles/uL ("Concentration") column.

\item[\code{seq\_dilution}] tibble with first column "Sample" with \strong{same sample names as in f\_tibble}, and a second column "Dilution" showing ratio of sequins added to final sample volume (e.g. a value of 0.01 for a dilution of 1 volume sequin to 99 volumes sample)

\item[\code{log\_trans}] Boolean (TRUE or FALSE), should coverages and sequin concentrations be log-scaled? Default = TRUE

\item[\code{coe\_of\_variation}] Acceptable coefficient of variation for coverage and detection (eg. 20 - for 20 \% threshold of coefficient of variation). Coverages above the threshold value will be flagged in the plots. Default = 250

\item[\code{lod\_limit}] (Decimal range 0-1) Threshold for the percentage of minimum detected sequins per concentration group. Default = 0

\item[\code{save\_plots}] Boolean (TRUE or FALSE), should sequin scaling be saved? Default = TRUE

\item[\code{plot\_dir}] Directory where plots are to be saved. Will create a directory "sequin\_scaling\_plots\_rlm" if it does not exist.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
a list of tibbles containing
\begin{itemize}

\item{} mag\_tab: a tibble with first column "Feature" that contains bin (or contig IDs), and the rest of the columns represent samples with features' scaled abundances (attamoles/uL)
\item{} mag\_det: a tibble with first column "Feature" that contains bin (or contig IDs),
\item{} plots: linear regression plots for scaling MAG coverage values to absolute abundance (optional)
\item{} scale\_fac: a master tibble with all of the intermediate values in above calculations

\end{itemize}

\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(f_tibble, sequins, seq_dil)



### scaling sequins from coverage values
scaled_features_rlm = scale_features_rlm(f_tibble,sequins, seq_dil)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sequins}{Sequins table}{sequins}
\keyword{datasets}{sequins}
%
\begin{Description}\relax
Sequins metadata
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(sequins)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "data.frame"
\end{Format}
\inputencoding{utf8}
\HeaderA{seq\_dil}{Sequins dilution table}{seq.Rul.dil}
\keyword{datasets}{seq\_dil}
%
\begin{Description}\relax
Sequins dilution data
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(seq_dil)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "data.frame"
\end{Format}
\inputencoding{utf8}
\HeaderA{tax.table}{phyloseq taxa table from GTDB taxonomy input}{tax.table}
%
\begin{Description}\relax
A MAG table, similar to OTU table in phyloseq, will be generated from a
concantenated GTDB taxa table for bacteria and archaea
\end{Description}
%
\begin{Usage}
\begin{verbatim}
tax.table(taxonomy)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{taxonomy}] GTDB taxonomy data frame.  A taxonomy file in the GTDB output format. Load the bacteria and archaea taxonomy outputs separately.
The markdown requires loading the standard output files from GTDB-Tk separately for bacteria and archaea
\end{ldescription}
\end{Arguments}
%
\begin{Value}
phyloseq-style taxonomy table, but for MAGs
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
data(taxonomy_tibble)



### Making phyloseq table from taxonomy metadata
taxonomy.object = tax.table(taxonomy_tibble)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{taxonomy.object}{Taxonomy table in phyloseq format}{taxonomy.object}
\keyword{datasets}{taxonomy.object}
%
\begin{Description}\relax
Taxonomy table in the format of phyloseq object to be used in the qSIP and (MW-)HR-SIP pipeline
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(taxonomy.object)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "phyloseq"
\end{Format}
\inputencoding{utf8}
\HeaderA{taxonomy\_tibble}{Taxonomy table}{taxonomy.Rul.tibble}
\keyword{datasets}{taxonomy\_tibble}
%
\begin{Description}\relax
Taxonomy table from GTDB-Tk output - combining both bacterial and archaeal taxonomy
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(taxonomy_tibble)
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class "data.frame"
\end{Format}
\printindex{}
\end{document}
